{
 "cells": [
  {
   "attachments": {
    "image.png": {
     "image/png": "iVBORw0KGgoAAAANSUhEUgAAAl0AAACvCAYAAADHR2sQAAAgAElEQVR4Aey932tUyfY3/OnnPX/AuXsuFJLBdB5JchMCDnRkRCYw0x0Gw1xERBxhkA7jTfdcOB5CwAgS5nu8sPvGIY0MqIQhfSEOQ5IRPEjENIwQvGmDrx2ZBBTO+0e8b15W/di7du2q2rs7nV9aAe29d1WtH5+q7r32WmvXyjSbzR34P4+AR8Aj4BHwCHgEPAIegT1F4B9E/R///O+eMvHEPQIeATcCJ4+PYWfHP/+4UfKtHgGPgEfg6CKQyWTwv46u+F5yj4BHwCPgEfAIeAQ8AkcHAW90HZ258pJ6BDwCHgGPgEfAI3CEETAbXVsvcePcGCjkwf6du4PnW1LLl7hB1889wra8tOefgqeUR3xeuP5yH2WQSn7AA4bNHTyXl9r63O34NMw4XjeeaX2f3eHzef1l2LD1KJzryDyHXZKPDmJNJEu1Fz2275Vw8ninc59Sos0qRjMZkCs6+m8U1c2UNNrotjIl+IxW0Q75xHG6HlMroVSsbQrhlU1URzV9Y/2pXR0jya1gimEl2nS+AY6msZLGHn3G9EzJp9NxKcnHunWLX7foxARs5wKtB/t3JVi36ppph7za96D1PWj+Khb+OBUCBqPrAx6UprGIcdTeP8WbF3M4v76EYmlvjaztZ49w455iDJjEvziHNyTT+6d48vAqsDCNr1QDwjTmEFxLpVsX5dy+t4BFDOLEZyrRD3hQWVIvAKC5vovFk3N48v4+prH386wJsGene4V5z9gZDGMJv9z7sGeyc8JFLO/ssDwvyvVi/5aHUM7abyadCbSCx7UcKq0d7KyV0JeaSMK4lSlkshuYkbLvtFBpFpBRDSmF18pUFuWhZUVfU/8ccrkaHoeWGqew8hi1XA45hR4Qx2+5WEOhTcMyQjLNyVG9CfaVsLYzj3waHdU+R05fZd22KtqaURXzxx6BvUHAaHS9WydmPeihj95TuEmGzu/f8vO9kGPrEf516S4WW+mJ95z9Fj9cBLCwgAeBFy79+M57HsN3v5Ph9yO+SEMkplub49PwiPT5gNU/XgMjZ3CmV2l4Vsccm1fl2tZfWFkHhrPH0INjOPPNILC+itV9xVORp1uHMcy7RZi+D58jPwK8+uOv/fey5q+hkmtgo43vSaLmm2/RxBD601tbnKRz3AqmCjUUl9WbeB9K9yvI1W4ZvHXiRnhNveXL/o8Vb9gQJidzaL6N+uNWHtdQnJxMVDV/rYJcYwPdhC+Rqe9wuBBwrtvDJaqX5uNEwGB0CUXX7+JfzvDdKn4NQpAlREJZWy/x4DqFYXh48sK5R4bwZAkX2PgSLpTu4hWxXZhuK3TzxdfjAF7j3d9CZuJ7TuF7XfXOfYjKpHrI1BDb8TFQ2JL/ybCZlJXCSnp4UPa5gxuBzhIP7kmK6qaPB5AKrzuK/JK+EFP9EIYUTh5XjGTh5RoZxLDa9+9thnv/iWPsas8JstIUPNW+dGzFSXZ0rQkljMnC03cUY1liqOIc53fynDom3s7nzYS5wNi6Nkz8bevlGHpP4uCNU/IwjE5hioXkTGE1zSPGPBIyfCfa6Fq2jAZqKGQyMDqhOhlHnicUMaHaULREmDdlDaWYgZfFQK6B8m3NhWXwvvSPTwL1JSUMSgZbERP0U7CrPx7enKoqoV0CRNU/ApAMaaohzxVMBXhq+JPnT4Q5I2RU+o6QWEQOvZ9KY6rKwrTEg4XRIsxInVGzt5HRUNfRFKrUV8g8aoxrd6AvTLgZJo7k0de3a6yCb2bqsYEg+FwG85PBlO74T0tfx5+4Kfw1yKOyKP0YtkFnWn+jqFbDdRJ6hV1tUfJw6aB39ecHgoDB6DqF72cHmTCvKHx3vMTCfrH8rXXgRJU8PnM4j9dYrEgD5yVunJ7G3EIvai94eDKPuyie1vJg1oE8G1/Fb9Wr3Bhg4cOUHiQFrrfvKNQj+OJMwLd/4S6+Erln2/f+jbmF1zj/8CmezA6CdOOGIo2TITYKW46ztgtq+CiQ1SHb+hLw9U94Q2G6kddYvFTCg61j+C5Rt7R4bQE/VA14K0DQoTCkyHsV/DEv1yCmy2eCS3Sw/c7s0uJ4RroKfJNxsq+JcCxbM+tLmNND1hGco/MSHxNtD+cNBswFxpa1EWiq8O+xrhegJ0vfD4dxGhDs8sHKbZQbijHTqAEzFHokjxK/AQ4tq6FImcNkaSOjhoVYeChuXjeSbDQTxwHIDSCbWv0+lNaWUawVgpt8eMPRiPSNYxJ1LElnFxl4xYlUYbGV22U0EvrW6sB9FhIV8lwW54RT4KWjm2ABEFi3Kk0RtsxjPsBTNS5rKDyeECHiImqFhHnRVGY30mwZaea2NVBHucEJMM9eTfUUbmKp3kAxZg3HGAKooY77gcyN8m3F4yj7t6uvDTdJT/uMrG/XWOlZ5Wu/NdBETSPFTvV1GzHUk+g3eQie1gaF+S+r+Y+2+dWFiMq5s1wEgjVFfRsol2lZkR58/YXGrqtN8nHpIPv4z4NGwGB0AT1Xqsz44F6R11icnQ6Ml0DgIHx1DCdG6Ml/m4dbnq1hEcDw7CS+IMdJ7yl8V6bVreXBBOMDirs7kHy/+Tzgy4xHLVy2WCnh1xM/4cn7p7h5FoAYd/7rU8wz1HP2R5Yz9tsVxWhJI+vIVXx/lsYcE/qmvClLuVPjpeGtoSYNKem9YsbopSUMz/6E7yI5XtrApNO2cNJl5CHqJ18Dq9fv4MI5yhlU1ozkHcE5YUwaeSRdiXHC2oiFZAHE1gsF3plHEDAbp5Lpbj+590l6GthngX741ZBdDgPSsmFhE8Ugy0+giCZYJM7V5hKz03Eumta2POaD/C/FAAs8AXJgH7izi1tdLLRoNCLi+N0aaGEnbllKwuyzOCPz2sj7BgTnZOzJ0O7mEuqK8dtXmkGxoRiCEYp0kkNFhk6zA2EeUVp8Xf1YPlsFkjyTRfJnMis5cJrcspv5M4fJceGSVGU2d9au2vRtH7dwfTvGujDQJLOeatjE51QJ6+fntfxHi74xZnyNB0uQfUejncKQfB7XKjk0lFwCVxujkqhDlJc/OxgEjEYXiULGx2/v76M2O869UOt38av+NpxB5vhN39Cpy5fIwJB8X81eDsKaX82+DjwSPVcu4jwzDskLdRlfiVCVHNdlkRi5NDdlyT80krorCU+qH8cPqhEpWEjjQedokkXKqfdNd87fhv3q0gLeZUfxc5W8o0l/7jHtyCP72taGSRLbejH13Ztr8UTwnR3Ve0JclVys1gYaEe8SD9mx32xXm0v4TscRzV3lTgkDLOJdCgXtC0KMIrQY89BRXwU/5n0ChtpOXAt5xo9Uo65g9qwEg5R5Cq4BSIuvo9/m26ZKEQA3FvlFMlBzqIk3DzaX6kDlWiqvYGRtaRySTy36soG7wc08No5BsoTmHmb6AK3HZaAgw/N6KN6lr8ZJDQVn3Oumr38IaL5VQukhLXubTYdwrD86WASsRhcX6xi+uPIjfhbhxjSiyht5GoMjDT1Xn+d/UlCev6Un+Q7P3g/ecJRvOjKPFk7h5u9P8eTFHKYvUsL4EuZ+eRl4LVx8Om0zGS86LSn33uAlkurprUTKozqt5M4FeWuhx4YbJ/pbj1xiKacuf6pz8jStA+cfVnHzyin0yAT/EfGyholIwph25JF97WvDJIB5vZh6Hopr5I2IGDotbDSEJ8zV5hK+03HsCV7xsAQ8eD5PzHnF8lxkyC3oDKjeJeUy+voxRJ6llbdoRgxNtZNyTGGlZQrraXlWSpf2DxWjjnnodIM4BcW0+Dr6sZtvhBXNe3iBeWxYiJFCiwi9V2GXfT7aDW7msXEMOlXJTJ9TUzyxbC0Z1msSWzK4snVM0pvCMoToGMOMyaF+4xvF9jaXDg5mvmnfEIgbXcpeTjyPS964x/ElheOS/s6OMi/Gq9k6T56nJHG2VYHZ28LI9R5HfxJdvX3rEX5ZAHDxIr6jm7jk+8dfImk/mrDO91cqYRWn8N0P9No/f2tPjgty0oT+kZwunbfpnHkCKbdMbs0g8ErSTcrdDl4m/uKaNDC4ESfflOTbbLx5oeTO/c8p4OwkpulNvNYHbMPy1qPkJeTcDU6Lf/IXFLj3zRBelLyUT+sYlzw65hJjy9pQ2AWH1vWi5MKlMaoDgnt9QIYIFEOHJbOLJ3BXm0uuTseBh0aiRg7PN6nlwlBYwJq9lRnfzmGzehllGPojj4liA+VCmawI400poC0P8vNYpjGRXBzZ2OZnLGxHyffqTViEdZPIpsXX1Y8M3EYZ8h2EzeotzetGWNXwuLqEOiYhI4ZJorXXnlZfCtEqa1RN3k9i6MI8EYMk4vSSh0M2JmcXDHbyWCre6ZWpuKerdkvmiq3gdjmaf+dqYxq6dEgBge+yPwiw2osRVmd/xJNZ4F+Ux0VGDf2NjGP6hSOJXHTjH6dw88UcTvyygOLpMXZpeOQqai++dWyxcApfXhzEIr29uDCOGvGS3hCVNmuXFwYxPDuHJ1dOiQucL0rTKJ6+y6+NCFp0duUnTLf+jbnTY5gjg+viVfzMQm7HcPPFVaB0F18d5+OGL86Jtjb2YhoZx4k//42TlyikOajgpeumv9reCV4SA8PnZz3MoCRDivLL3H880f9diV6Y4PNcq9q2BiE5O8Tp7I+oXdxCUc4fraeLwNzCFra2YJ7rxDEueY7F1hOtSevaMIDUY10vwHaLz3F0HzQDkX29xJOaR7MZZBhf2ntrTYSSXG0uITsdRy8qrmGnn/bqyqAsWRSXLTlVlEi/g/6pDLKZoDeQq6Bl2TssP0FJyM22PDf5+WUUMwVkp/otckhBkz554j/bkJV1VbBmN74yytkMNpZ3MC9z7owk0+Lr6qe1FYsoAmGuHwDCqlAoo7i8k85ANcpqudiWvg7cLOTDy66xUQxylQqKqIdDUx056DNP6YaylpX5TkVbdGIPF1kUMjzNP1dZRiVX4FvAiPS54tAGshnxDa60sKaEzo1t8oUSxsKhQzty+r57ikCm2Wzu+ILXu8GYthuYxuLIVTzZy73MUotIHr7LmMNhkSe14Eek497g6wteH5HpP/RiUgj3FgZaSriThbVoo1r1JYxDr8gnJiB5grPYmNlBkGgfIOBqCzr5gyOAAL0QFQ8vHgHBvYguBOQmp+JtUldX39YBAh9AmwcPf/O5sg9aB2T8EI9ANxDQQ19qWFnQpwT6pK0yuiGKp+ER8AgkI+CNrmSMjlwP9uYdlvCfFG+bHjnlDlpgtv2EIz/xoOXz/D8tBPpKuF8BC2fGthVhBlkG2fIQluPuk08LJ6+tR+CQIMDCi0NDQ4dEHC+GR8Aj4BHwCHgEPAIegY8TAZZIT1sr+D+PgEfg4BCgnK7/9//5vw9OAM/ZI+AR8Ah4BLqGwP/63/8nRotekfDhxRgs/oJHwCPgEfAIeAQ8Ah6BzhH4/ywP0d7o6hxTP9Ij4BHwCHgEPAIeAY+AEQGT4RU1urYe4QLtXE7/RKFoSen5dXH9+JgoFB3dfJRtCnqO+miFrSWBI/NJW0CEujIsGB538MBcH/qANNPx77YYZhwuXH/Ja2x2m52n5xHwCHgEPAIegY8MAd3wihpdqrKRQtEv8R+5UaraZ4+Ot589wo17fPfyPWKRTPbinFJOaA7nqWxQ6dGBGhwHgouCw5OHV4GFaXyllBBKAvJAZE4Syrd7BDwCHgGPgEdgnxBQDS+z0TUyiGG8xru/hURb7/GWdnEfGVRElCVm0u5Urwx1HW49wr8u3cViy9Vpv9toV3kqW7OK1YPydsVw2SP8HdD2nP0WPxAOCwvpvH4xmR3EfZNHwCPgEfAIeAQ+UgSk4WU2uk72slqIQd27p6t4hXHkv1HRSBHeorqL50o8XHl8DBeuq56iD3hwXW0jz9YHPCgpRZllqJLoqH3PPRL1FUkeEQY7d0fpUxIhUCGvUw5VpzaOnTRDmW4EchtkCtrGcMGoUwkXWMi2hAsxXHT8Q54hVjrPMHx84fod3OggHPzF1+NAxCB/JOjIsLQMwzrm0rom2sDfd/UIeAQ8Ah4Bj8ARQ8BsdGGUe3YW1vA8KITcA1M5RLu+L3Hj9DTmcAa1F0/x5sUc+hfu4iuRK7Z979+YW3iN8w+f4snsIF4tTOPGM14LkIpRg4W1yIsm6Cz0BnTyuIviaS13bH0L+KGKN+/ncB6vERRmluMtctjlV1q2RHh15AzOMBDcugUj15eAr3/Cm/f3MT3yGouXSsJDlFYnIF+lYtVV/FZVilW/d3gXnTjcxSsqV/T+KX77GlhcDyRt+4AX1CY97mLx5ByjybAPwrCOudzNXLQtqR/gEfAIeAQ8Ah6Bw4GAxegCerIUStzC1rMOy56wnbt5uRRWvLr3FL6fHYyF6BYrJfx64id207551gCKpDM7yQsj957Cd2Xytizhl3tKQerAIDqGEyMUChRlcOT4bz4PxpvkiHGm4swyof70NN5eVGoZpqU5chXfn6Wi08eEzCJkK8en1ikmnf1CChx6aPTZSUwTTrv6O4Wb75/iydfA6vU7uHBuGotET2Kv05Z6tzsXOh1/7hHwCHgEPAIegSOIANsc1SR3z9gZDM/exUplAa8AnD9xDHhn6mm+tv2OJz+9mr2Mk7PRPpQr9t2Vizj/xzQW18kDdBmLI+OYrv6I76JdIen0E/8O/uR4mxxW9x152v7nFLbvlfDV7GsAx4Naex3TBEAeom1wbDrVqQMYdo2jzpPL/hI3yNBaH8T52Yv4+YdR/HpaGF76ACCQoe25MNDylzwCHgGPgEfAI3DUELAaXeg9zvO61sngGMeX5IVqw+jqOUFxuNcYnr2P366YDKZTuPn7U3y/9RKrvyxgbmEJc7+M4rsfohBKOiycxbxG0fakMzneLoebQs+VnzD9x2XMLUzjQpbrshuaZKzwQO1rZoChA53cEptbpcyd4iipPv9zCcAgTnwGgDxX68D5h1VwL6V443SkJzBQ5Tj6lDJ0OhcqLX/sEfAIeAQ8Ah6Bo4aANbwIiDf2SCPLTdSp7NlRnAfw6o+/RNJ7NPGbPEgnj5ewilP47oczoDyu4ewxSGMvoC3pzNY5HUpgr9CNP2XRYTneIkfAx3oQ5ia9IhmoX1qa63fx6zMKgX4IZGbGqxzfjk7CCLaKmdQgef7xF9v2YvtZHXPt5nRtPcIvtHXIxYv4TknwC164uLcQDS/qMisyPGfOvuiaSFLBt3sEPAIeAY+AR+AoI+AwumReF8/LYnlAbWl6CjdfUFI7Jb3Tm22XMYdx1F7wJHDmQboIzFHb6bvAxav4mXnEyNgbZPtBsY1Wtzid6YtbnM7paazgakAnWSS3HMnjAfSKrRKwhCLboyolzZFxnPjz31z39UFMC93JoCVs2tNJxyWV5EqnU7j5cBzD63fx1fEx/OvPHpxPk9MVyW1bBWbn8OR/TnG6Z39ELZirMXz1Ry+m6ZxyAZlRpcucEjdFan/oEfAIeAQ8Ah6BjwWBTLPZ3PnHP//7sehzSPSg7RumsUhvCv7+rTHUdvCCkpfpMubWx1FzvQ158IJ+EhL4gtefxDR7JT0CHoFPGIH/63//H1/w+pOZ/6DEk9hqY4u/lYqLo/jikwHBK+oR8Ah4BDwCHoGDQ8CeSH9wMnnOe4FA77f4eXYb/5pdQvE45cRRhYGrqMlQ4V7w9DQ9Ah4Bj4BHwCPgEQgQYOHFoaGh4II/8Ah4BDwCHgGPgEfAI+AR6D4CzNP15v3T7lP2FD0CHoHUCFBO187OTur+vqNHwCPgEfAIHC0EMpmMz+k6WlPmpfUIeAQ8Ah4Bj4BH4Kgi4Nwy4qgq5eX2CHgEPAIeAY+AR8AjcNgQsBhdctPKMZxk+1KlFZu2ShjDSVHUOj5K0tWKVcc7frRXtp9RjULat4z/u3Dujtg81qbyXmDWHZq0we0Fqn8ZvBkZ6kX6sTabWup1Nl6+VfkIF47vx/rgGNx4pgoijlV5DM2pLhGNtr47FqqbVYxmprBiaY5cTt13BVOZDMjVLf+NVjcjpA7VSWq9uif1ypTAZrSKriBzADpE0GD8w/mW855R15boE1sLNtlXppDJjGK3SyfAmq3HlGs9opx6sonqaAZTpi+MTQ91+G6Pu82DYWzSh+uZIUUZT/M8bFZHEcyxoBXOfQajU7b1TfT1udjEyhTRyyAzOrXred8t1Ed1vNno2voLK3K38oU1vgv7UdVwH+XefvYIN+6JUjgGvqyO46Ul9Jfvg/Lo3ry/jzxtuHq6hAe8HKNhVPxSEp/4iL248hK/zvbih6DEE+33RTrJf3Pon72c3vCSIvZ+i98+hn3DaENdTMNo1EldEz9XMJUto5HYjzqk7Mt+oAvA8g7LIaM8sp2dZQyVs4jdbFPx3YNO+o2rr4S1nXnk94CVmeQKHtdyqLR2sLNWQp+5U3tX910Hk3hFLLP5Dud+uVhDQTMsG+XbKYz8TVRv1VAsDqF822ThmPibrilYtyrImbp069pezIG+Vrslq6Cz+baJXC6H5lvN9F+5jTKKKA5kgb4SZoqN+DxsVnG53EBxmX93Vh7XgOKy8r1vYRJlXI5ZzWRwZVHGALKKPitTWRQwgxatoRmgntWNMqWzP7QiYDS6tp+u4hUGcT6yu7iVRhsNx/Dd73RT5rvStzHw8HfdeoR/XbqLxZZNVDJSXgMX53AzqLco8ahGyupEKcg+ArNEPtHR5jONprmT8+r2vQW8nZ107PHFd8APSic5qX2cjV98PY7FyiNWdqldDfkTagHNYjHxRpS+7yaql8tApYX5iAWTx3yrAqS62baryRHsv/kWTQyhvyvW1uHWP3+tglxjA+HPVhGVShO3YjdiTY/NJdQbRUxcG0Cu9jiFkaaNl6efENZS5XY+WxsA7S7Q2AhnCCCDF1ieGcCAWKT5iSIQmQf+EBZ+1zfxtgnkyEgL/vowPpCL0t6sYooMLnrSG+oPHzg2q7hVK2J5Ps+v5ScwhCZ0WzAg7Q+sCBiMrg9Y/eM1MHIG37OaiK+x8pTqByp/W49wQw2RxcIoq/g1aC8pT/uGsBbVUjxHdRhFuO26vEmZ+lLoSQlbWcdSJRre9+T1R3hwXdInWV4qsquy0RibLKS7DJ3e0ehR2wc8KN3FKzpkZXMM4TEqDg3g/NeihI4CZ3goeZRECJLoqDhY+KSS20ZTcNfm9OS5Ow7vG60RID9mKmQeasNrVMqSQHRd6Mfm2oARdZGhPfkZkFPDgRY6NObcHTG/gv6zO8HaYmtMX6t/hu32cKiFH8kWoa95LKnW5PoqVtvwYgbq9vMnyrVrA8El60HavuJGOVMyWBMxL4AaglSeaMWTfZWFLXi4Kuohc4wbncLUKI0R9PRwB4sJSY9dDQUZutK9CexchsqUsEqibBqCJjp0jXkXib8trDOKapVCa0IGNZZl1AkiBCT0Jh5OLBSdNJHZaYSH1lfVaapqD7WZ6Ipr/aX7mKy7vV2bS3U0ihPIMy9LLW6kuWSUvHWs+faBshWI0MiAhdKolek4BfsaDElIGmyNinHML5dEg7XL+bXhaFirkrUiu7o8mEdarhv5PZBjYp9kKA1hYkJrWLmN+tAE8LiOwBWVv4ZKLpyHlakCarkK7gff9RY2GmRHKd/9zRXcJk/YhHwCW8HU5Tow08JyEcp1YOV2mc+3Jorz1Imh43ci45pbyzinIIerMW50ydDiyePo6f0c+RFetHo7kPslbpy+i8WTc3jy/imePBzHq4XpaBhpHThRJY8W1V587XjaJ1rTmMMZ1F48xZsXc+hfuIuvWE7YMXxXHgewhP+IvBvpgeM3e9fYQFhgYRu9P1Tx5sVVDJMslxYAFt6bw/kROpc3/5T01rcAohfRLSyKTZ4skydv+10bd951IM/w0z2CJj5p5bbRJKyIRjinTLf1JcyVpAGs4EmHbI30olcpeq31EKfHcGLkNd79TadkNE0DD3n48cnsForW3D+qd0lrL5x7xhNX8f3ZBDrrS0BZelNf4salJZwXPN88HAcWFiLG5OKbHraOSWcKh8bDgS5+RH8L07R2Kaz6sFfDTNXfjJDtal9ePFHaOijXU/dtbaCRi4YMFDLKIYUXwhBkq9LUQlA11HGfhymWiwjDUQnjGjVghkJbFO5YwVShhqIMc9KvfO0Wqpvc65YDhcLWENwzAun4jW4oGDeEciTMYZMtICAOLHTI+GRhLh6Ki3oEJY0GymWK0PLQbLFWEKFZm05ynPIZw6LJw5lEc3kI5cu2XBviYesb1ak1UOceC4Wtfmi+mfahNAMUotaCMpRu1kDlGr9Zk5elUV9Sct9cMipkdKzp5z74S8IyxTyT4SOwWosvJAA2GmlxtK3VGgqPJ4LvR60gH1oSvh+B7uKAHpIoxJcdQK75VuBLXq4mJq8Bj5uqN5bmTHwXV6ZQoPD4fSU0zjyKQK0gH1YyyGQLmtc7j/m1NeYFf9vMIXSKcS9ZaJzpgprOXRgm4WCbl6RxJjkO37WY0cUNG+mROYbekwDUp3XFY0NFsHvO/shuOL8FuT1gXrIz7IZMNx0av20OsQhaw998ji+of+8pfD87GPIjTwGAxT8pTyr0wDHaSWMl1lTmhtE+jn66Rh48Ft47hhOyD32mpTdyBql0U2m3exzwSDGwbblNNE/hJhnQXwOr1ynRf5p55azz9vc29+qZSNmuMUNtHF+e5R16rlxM8AIdw5lvBsXcA7Qu8c3n6EmkM4gTn0khuF43BU/ueZNt/PN8WdbGpOLccq0pfRL5SaMSAH0XIrU2+ffn7TvNU6yQP+jDaBKz8JrI0JF4AO4rzaDYqGMpSCvJYXJcPDHTDUEqkWJc+EOex/zOThjmzE+gKOm4PtnNo4jg4ZyNU8McFtl0mol09AHRc5knA+RxrSJDNO3opN7UiHYDQQQpP5+QS2bpu/KYeTeELQQ2bxGxufcu8NBlMrg10MKOybIkz0mTjOAIAX7C+ExCLgHQHDTKiKZ2WWQ0kDNfStsJ0tcAACAASURBVMIyYZ7fVjHKDC6T4S45Wmgk4ijH2z5zgUGKtr4fGj16SFJDfNTMcrkmMQ4KgWsPUOy7UEOhUEOucj/ywMI8k7kKz8eSeX2tZaB82TDH5BVTDTo619ZrUljYhWGK34nOfl80/A7pqWZ0CcMG4c2R8lKAMMTYlscmQWlJ69Xs5SAE9BXlPUHeyPiNEJTMz25+ABloZOwlj01grjV3m55GHj0nEt1C+pBU592Rm4dcv7q0gHfZUfxcJQ9lwt9IT4pC3lTfUTWCqASRfMNRGHYONj1jZzD85j22mcGthjNddDQPnAwzM75unj3ZQYs0Nn5k1M0Bl6ROYwZPmZkkvVQhQ+rtvSFsppfqKt0AIvk7QH5eJlUva0aPeoMuoBZhoP4gRxqY94DCcvzGnjBODT9k9L46XXEe89ZlMZBTbvBpc7ES6Vj4Gy739Q8B0hORWicVQzIwlgHFC2F1MsHel5Kuo3+EjXpFSaQXSeuRcJPaFX0o3Z9EPWpJUWyPJdCjUUZWm+faY5lQb5cxwiLpxImlil+cUI2iZCxNUcoU7wPLWknG0URLveaSzfW9UmkALImeJcr3Y4h9b8XLCzMl9FGy1+R4mHPFhvIHAETCipxmi8cWo/378pgoNlAPn6Z4ZzKYKHQsxTEYWEF4WfbRPpMxdOHQHfw0kQ7NabT2ojBsKKQn6/NJSV/98Re2r3wrjAcyjHb/xw2R1xievY+Ip0whzYy+hSX8Uhpkyf3TIo8ocWwb0Txil0hPkamjQ+a1W2Kem5tnXXld7VHvitzkLVsHzj+sgnuFxBuYNsPqsx4MC+8lGcDWP+aF60WN7E02H/SGox4ylW0GKvQG4MkSVp9RJsQZ/NwuHTK4Tq8i/+IpfmM2L+VmLRgY8UvbrdcYzlKe2nutj0Vu1kt40+iY8rsu3cGXJh01ij1XqnhzJbx40i5W2Gm3R33jmMyVcat6DXljuEVlQDdowxuDJs+HOoyFBVOMoxtqto7J1g7WmNOMcjVuRSgZTxTDkfvaDE/hxoHaxW7RITOEjJ2hCfR1qhMTTXh26JiFxaYwYcLf1ZeMv8gfz+OJXJInFNpb3kCmMIqBlsUbRPlayGBqqSJHAcJLEVsbTPdbqF7LCw9LO/qE5IOjXWEJFGdKyOf7UcyoMgXUnQfMiI70cOAY6ZfmxPK9MgwlQ2logq9yZjwT9qjgfp6WSA1DE/OxUTQmN3k/alxBhAdnAjNKjOPXJQ9JjBt71+Qp0NePIWyE59jEUr2BYoxe2CUZQwsOnf6+hKwP/VHE0yVDi2QEha/+38c0CxGKhGAZ8pNvZYlEYnsSsgMDQYsMuufspsyTpU+q+zSdnWT8X63z5H4e2qNQDg89Osc6WMeadkuvV4QvY4TlBRE6XaBtBGS4SeqrJWDLIaZPnc9u5VZ48DAuQG8mUtK/NbyojLEf8nyqYfmGYyxHK91+XF983Yu5S3d5aJGYtUOHhUFDz9fz63FPV/B24dYj/LIA9J/QXg5w8WNeNNfcfcDWGwNNO2h73ELeC3r817eH4LkSgTeLGWc1BI4LugEmJv0CaGcceZoUTwNL/I1or4YMlQZ2A1Bko6dyhY7S0324Szq1WzLnSklGTtTJIhLDV0uIt3R17cmkh/k2q7c0D6VGND+PZdpqwJo/BtDbjc1yuG2JOQeM5p5vW8C8Ju3oo4kUnHaKZUCADvKYd+bHRTqHJ1q4NBHHtG/xtfP9YIaSDOn1oX+oice36xgiL1ekLRSbHk1pV4i499L0YLKJlepllFGBDEdLSszYUxPuQR5TmaRvHyfHs08Xhm3hoFDtdJxC4jAcKp4uGVoc1N5K47k1c+s8xPjdlVO4+eIqULqLr47fZToMX5zDzyynSxoTaVUjWnNAaRrF05wWRsZRe6F6Q0L+Yf4N0U8zNq0c3aBHodBBLNLbiwtCBy2iSN6NJyfu4F+XLoNS5ehvWOqr9RXNho84n2QMDWTUS2d/RO3iFopMdsp7G8f0RWBugb95yHLi1P7MELmL/zz7EV/IfCmDd5QS2H8L2mmbijm+eS6jNYjpF1XHlhOCoTAqvwxyBtugwwz2y4HXdnh2DtMj0zyxX+B9/uQ2vjo+xpjRwwbz9EW8pA5+tKfYw22cPD2GOaNOenhV6LTLD8rDolwcc3JwAnH2luI424cnU1b60v49a/JJuA+ltWW+gSrrQntWrYXhBmVY9LCNcextqywKGW7q5SrLqOQKPK8pzz1y5WwGG8s7mFffcqcbaauC0WwGmbZki0rKbsi7oFMc2mDhNaKaq7TAoaM3yCw6RXTQZJFep2wGfEoceDv7atgUiyxkHObRaXzJLJlfRjFTQHaqHzuKcyPo2VfC/UodWSaY3FNLrpOgFzughPpC4TZWSvPci5ZGnyiJ8My1PlxYhhT4EdG5lbXrp/dn523gyAyBMsxrVSfexvcDZCgNQTi6WM5frVlBizm39DbBh4UBc5jU8WEPJg0gmA/eP5erYFlNtmeXyfuVw0BkLdDD2jI2LmdBvxm5YgWtxP3rXBi2g4OKYafjVBoHf5xpNps7//jnfw9eEi/BkUKAcpL+hZ+sYeEjpcxeCUte4EoPnkSS683MKL/LF7w2Y3O4rpJXMIuNGeUlgMMloCIND9law4dKT3/oQsDj6EInXZvHkHCiXNdIeDEdeL6XRwCgtw/7Z+u+WoFjMTz/cwlR76yjs2/yCOwWAT2s12nodbdyHPXxHsfdz6DH0IqhN7qs0PgGNwKUo7aFX6j2ov+LI0A5Ym9oX7F4k7/iEdgTBFgokFL2xNujbMsEw0sNe8L8IyLqcdz9ZHoMrRj68KIVGt/gEdg/BHx4cf+w9pw8Ah4Bj8BBIODDiweBuufpEfAIeAQ8Ah4Bj8AniYAPL36S0+6V9gh4BDwCHgGPgEdgvxHwRtd+I+75eQQ8Ah4Bj4BHwCPwSSIQNboiJVPC0iYnj7s2gGwTN8ZDFpnWx9KO4V3kpZOnc5W/emzq26Vrz68LLF0Fntvh5ZLb1dYOj6Pe96PCgV63Trl5pm3e2NtEsvCurVM3r3dB5m6Kc1RosXmSZZQyyIzKTVhJAdquQra55pL3s5cSSglGmjWjybtrnilFM3bTZAlrTCpYiT6jelFJm65UHcD13aNx6hzpMnQLEJWuys8IRJcu2jDpEvlkMrv8DXHJ72pLFmxXPaJGFyNFJU+eKjvSP8Wbh72YO90lY4g2lExRJmVXWqUdnFaWXd3AX+I/C7QR6FOtGHJaIdvsl1anNsnuS/fd4Lybsapy3aKj0jwMx2xT1EPyJtsB/uAZp+IwyUM7sdNGtbIosbIJ5cpUFuUh3rZcrKHQrRu6EZQVTGXDneiNXcD7DC2L+p3LRdQKu3w4MDNq46pSX1JgyLDSDJVG+TZcVRk5Q1HrsDiEcqwGpeixVA9rIJKBlt3AjJy7nRYqzQIyu54njjMqLbYuWpN1ZHdNsw1Id9P1IL9b6m/eQcqh4WcwurQedMp29pZFqA3t/pIdga33eIuwFI29o2/xCHgEPnUEVh7XkDNuIS92gxc1W6g8T672OIXh0D6im9VRZDIFNItFROpl66RoH7CcUkaG7SKvFh/XBxzMOcMqUui9iEqliVu6t0sXT9SZnLg2YMGaahCSzUX1EVcwVaihuKw+2PCyW7naLSSx0llHztlO80XMiHqpfaUZFHdLM8LAn+wnAumMLlUi8gScu4Mb5yhkJsKEzDsgw5HCIxbzGPA6gzfuaTX3RO1GemX+5PU1lRMACjdKuraQpBii0mG0RNFmjWLkVJcxQkN69l7ixum7eMXK3MhrESr8xIZBMHYMN54ZxkV4EgaK3BGahvHK2IC2rpMJw1gfMTdMPhvm6vyJOSFZVRlV2U18SX3B+8G9EticHx8Dr9tpwVnR0R7mtowlfsr4ACPtOl97hLuDjmHq9vQSC2uIUNLUY40Vud0tYSZ1nBoW0Z/02LmkX2VhK/bwLPpV2Y2Xt0dDMSl5x2SWKkgPSg2FmHxSpwRviSp7JoPIQ39E/4ziZaCQ2yiqVQoXSb3J12GQx4iVCFFR2+gUpliYT4atHJhItVN98hIs/CauDWA33iEEJfFY7UhLfUptKCugnckgOo96J+W8fwatnR2sXRtQLhoO8/PYUTxxYKVrDP3kpci8JcxxZB61viqdKWXtSj4pPvtL9zFZd3u7NpfqaBQnkGd1JWXtQZV4CxuYBLe5qAZoERN6hSTmbVEKikf0UtenKHQu16b63VBZpjo2rEd9TYtQNfvuuGSS/PTx+rmRhuG7xegZ5JN8VDrW3xAtfC7WQ/A7QDTIsxnIaJMjinkwXsqyR5/pjK5ndcytj+NLudHj+hJQphAk1UjkN6r+hyIkyUKRd/A8Vij4L6zgKr7nZe6EOrwoMtXoowLbT7JbvNAya6Wb/DQg22a3ULTmREXpvHk4Diws4EGkhl4SgkRji4cBKbxKepQeYZvVeLyKYVDYtYrvjDUSbRh8i99eyLFPeV2/iBguuTWaL67i7SXV8FxC8c9RHgZ+OI7FSJtkYsHQNjdnLf0lOQCLfwA/s/DzHM5TrcaSOCc9A8yT6CxhBT8Fsr9iO9vzmp5RnG1zogjEDk1jqcGGkQ13Gx2d316fy6dmHrZpDTSVwsX0g1MAREinVWmiEIROaFwTlZYM9wxZihnzHyEZFmoN1FFuqDrVUMd9HuJaLiIMxSTxpid9k8wqbV6TLQcKA8mbUVSeHSpSnJUGjTqWjrW+rQqaBdk3itvOchGIeAQaKJcJOpJxGcVaAaPVLKvlGJVH56mdN2rADNEgr4YLE20c+42XhqVqLK6gylwhVFMPqF+WfTRjI04u+QrdgMSaSFursy+fB/lu2v1jhaFVz1eEgDZvSXNsXcdROvG1G2HKTsxFuvtQmoEjREuFzIGK9CxOFNGoL2FTJU+evqH+EKvcAPSyh2r30Bsmv5/q+nR8d7Xi7LwAt82jaFmPsWLRS6izYtdJ35moBuYzGw3Td90iHyMcpRP93VM592F8MofaYxEgbm0AuRyab/nsbL5tIjc5Hs6LqNca/47XUHg8EfzO1YLfEZVX948NRtcSioF3SXg0mDGiFqEexInPhDAsfKYYZKxAMRVK5oWqF//knpvtp6vAN5+jR9Xh2RoWR8Jdu6m0zHnZvvUXVhRDj7Wtr2LVaEidws33ilEjiiRLUuk/lRDq2R/T52BZMUji7JBbwwaxXK1BTP9wijP4rAfDJlZWDC1zY+0fEg/L2hzDiRGEZW6YISfwS6SjFFW3yR6w7HBO2HgbRg7cA74HeKCFbVg4QYojQx7iiZq1NepYCu4Gyo9xzBMhiLjosy45ETIBkB0IQ0wu3ok0pQKGTxE+CbwE+QkUYfHiaHyg5m3Qj+uOUhOR0YnyC8M/eVyr5NDYaEU7pDrLIYgAujDRaFHY7vGEuOEyY1EaV4/RT6EjhgMweV/02ZnBhtX41IibTt9WMcqMF2ncmjp15xrpli0PYTni+VJotzPHbJhlHWvzH/lusHHkQZW48k8qDr8zr7ugqNr3NVSaltAf4yO8WESX1lKjDDW1i0LBxWDRKrpaD5PWp0VnYTQ0C1yfy5hBxRb3ta7HqKFCXjwwwyRJJqsySkMbNKzy0RNJNFwdn9uQZd/4JHLNt8wIXnncxOTMJMC+yxTybWAocAmHY+JHucCojvzOxTt29YrB6DIk0sc8PEqO0t/beDXSoxhTdDPmN8qesTMYfvMe2/iA1T+A/NixiPDb74wWlNJHNQCnFS+Y0kUeqmGu4wl95ZjIJ92I54BLMpxpCOdF+isnDgyUXuZDi9zJ2ChzYKYsrpoxtM+Nub+ThbHRRSet7LuYEyaTg48Fd6Mq+3yRntTcf+qNpaB4wejHbxkQP84URjO5zOP0sxiI/IgrYayYIGbecZqxgfYLlDwe8RKQPMoNSBmZyEeEGngIUcVGISIO+/qHAPHDHW91XdHxMWOiU+grrSG49zNjURpXIg9ID0UxAhbjUyduOK/VgZkKrEnghiEdXeIGF1BpqflMGqk25hjMeDav4/j862tXSaRvVdgDg/0GTDlXk6irlhQTmyfQo1FGNjDg+FoKvCugHDstnBjJG9P0l6fW9Znw3VXWy1oJ2Ggohr+kHXya12NoqKi5aPRSbBWjmp4BqbQHbdEwyxefWwdz8v6xh00KyQ+hP9+PIZbjSN5ibV6sZPTvsbVjVxsMRleb9MlTsb6N7WDYB7xbF56w3m/xw8lVrD6j0OIZnNFCcz0ntAsBDXmgG4CW8B7dQE+vIk9vCMrQlyTR1if3gDAa1pCdgaALA0P34JJD7mRsAioJBxYMrXNj6Z/AJd7cLTodzklcoPCKA/ew08EdMWPAyV65sbBQmerJ4E+d7O039jaZDL2FBOP0eVgr7OE6MvOO03TR0NrImxa5YZE85puKkw/98GfrmJThVQohaqzUU/Yjr4aH1Ma2js2YtEUiqbOew6XneBnGF2dKyO9x0nXg4QpCxQZB6FIbc8wpmNdxfP4da5cMlaQ3KilfCwVMLSlyC28MD0NLw3gHO2TEyXA1eWUo30sOY17VGmTES15mIUX58JO4Ps06h7TEUeLcW9Yj6TpUx9IKhRaFFy9Rphj3+IW2aZjli89tnFV4JY+JYgMbS6QLhXXpvIm3VW1ewgGH5mj3RlfvcfRjCf+RSeIUFlPe1vvi617MXbobDy0SBBQGXL+LX8XY7XsLoTcrlnekJeCrEJKnSeH5/HoHni7m+XAkyoNCpipT5TgBA6Vn9NAlt4YNT0B3yRclzc4SMIzNTUJ/AwfzpV3RUXBOnBOdvTJWb1LPXbizfinpqDS7eayFMngOh2AQy82gp1RhWNGPX5oEXBd9lx4u3m3TVDw4Ws4KCzPA8hSq8eFP6SL3ibwpyriVqbinq3ZL7ntFOTsNJTxkl4clVNtwcWFiG2O7zpKIFSOZbq45GeLiNxm5dYE5T8lEOI95yp+6LPU29enw2mYVlynvyeXhkqTbmWPXOtbmP/LdkLzUz/w8losNp/70dmOzHG6PYcWWJdQ3UF/aBBns0dAiD1dHt8zg+UvBG56u9enSmfIYle+1VT7SO2E95ieGUC6Uw20uXDKpOGrzF/lOJNJQv1vjmMwpxinTW6z5Nuc2P1FEsy7DpMyuR73etLz9S8oocqi67fPx7o0ukWj+Vobl9Pwvll81jh+uREOLXE+euCzH/gtnwpwuHMN3vyvhPubJUvPKFKTYlhZhOOuX7BymRYhT6eU+pJwp9hKAIY+NGRGvMXfaFnKM6nFSx8DG2Sm3RvP0XfQ/tHj6bPSTMIzNTRuYW3lSQ4d0dJxdc6Lz18fq7eq5C/d26Kg0u3rME1DDHI5JxWPTh9KaEnphnh0lPMUSlEVOC8vnMYV8ovSzG0OgzQGCPCWrLg7esbwTVWaNILsxNFDOyvBnVB6e+G2Sm+hofbNlDC0LTx/bsiAMXdwaWEZFC1MWhzZEyKiAZqXFw30GeSjfqxbk0Dh0gQsTTe+k0/w82IsRMtRDez4pOVL5eZ78T6HTQq2I5SBWmUCYcEE52NtpZaqNNxk10upYdvMHn8fgjVDp1dHGxebNujbJcCAPFb1MYVrH0flPs3YZbhQqNMXaSc6+Eu4HSVLRrTl0NehG3yhfxuU6Yt8XCh/zl0CE3Bmxr5qcQ9f6TNJZwaPQrKBlnfuE9UiGDcLtJ1heW879neEYSKNS5pUp3wmnXmRkqd91l3zRub0MhYc+EXSeHQAaYf4WecoaDbmFhzYg9h3X2vfxNNNsNnf+8c//7iNLz8oj4BHQEaBtKygkeDB/9CR9CwMtNUx5MJLsHVfyOmSxMaMk2u8dM0953xD4FNbuvoHpGe0xAvRw0gVP1x5L6cl7BDwC3UVAD2VQfooSlusuM0/NI9BFBPza7SKYntRBIOCNroNA3fP0CBwkAiycAkv45iAF87w9AgkI+LWbAJBvPuwI+PDiYZ8hL98ngcDBhhc/CYi9kh4Bj4BH4EAR8OHFA4XfM/cIeAQ8Ah4Bj4BH4FNCwIcXP6XZ9rp6BDwCHgGPgEfAI3BgCHij68Cg94w9Ah4Bj4BHwCPgEfiUEGjf6GIbVqqFl/cDrpe4cTzFxqCqbOrxfogY8Egpa9BfO1Dlth1rQ5ynKg1nxw4b26HfTt8OxfHDUiKgbkqYckhit72gqW0MmSiD73BEEKAtPOQ+bZrI6jpSj7VukdO0/WhQO30jTAwn3aRlIB9eUjZI3Teegvt+8wuVTnGk4JKi92Ho8o/DIMSeyMAKRO8J5b0l2m25u02vHe3JyDq9jR/eWza1bYeW77s7BOiHkzbb3LFtOro78mw0qw/XBTqexO4R2I/51qXcC557saa6SVOltRf66xjTucrT1L7ba7oee81vt/IesfHte7qOmIJeXI+AR8Aj4BHwCHgEPAKHAQHF6PqAB+eUMjcsFKScP7uDk+cehYWt6fw4L5lzQ9ZdZBpReE2U0jmuhCFFaOnBvVIw7sK9D3YMFPonr69p/Sw81F4ylCU/gzZVTwsdGnPuDm6cIz2kDpa+RNcpq2Ss8gV4LUUDvjF55fiET1UGNRSr02PnYn6uPwrnXPSzzk+E/hhOXn+ZINBL3Dh9F69A5Zm00LBCK9XaSeD0UTazOnyynIioLQiAyrBktHImVHSYXaMn1MwUqnQuSsmMVjdpFKayVFuOyn2EtBhuCp8oWXLbS/5KPUAapIzJqPQE/xU5IbZ+nHE6+lOPJTXzZ6c8WNhS6DdVDUNdug76uTpO1rwkyazYC7FZu8RTD6t1gLUZDXHVMt8RGbR1EKPXrkwWnpH5ia9dPJ7S1qomiI5/hJ5BB6U9up4VuipNcRz/zij96VAdw5pEeLQqa58a9Kcxo1OYGqV5V+ujynVgkF9lq+iSUb8Huizs3LCWGa0uzKOLn+H7n4il1FHVj35r5IQJflY66jgVF0n3kH8qRtcxnPlmEIt/ipvp39vAyCDevuOG0fa7LQx/8zl6mEJLKP45ijfvn+LNw3EsXpKGCRkW08DDp6ztyewWiqqhhiWs4Kdg3KvZOp4bAXqJG5eWcF7SyW6FhbCRxEMjyGrpKQW5t/7CCq7i+7MJdNaXgDLpQaExV1+XrKos7eCrjktzTDJsYfoFx/0N1ZAsKQZyQIIbQv0BrquYWw8aAev8RHWkOcfCAh7YCoAzkrx25DDGUXuv1ozsdO2ocn7sxyuYYrXpdlhpIFbPTRQspuK8udpjBIYNNrFUVws311DHfTGO6sTdxoqoV5hDEcs7aqmfGgqPJ4K+tYI0rnihXixz/qwe4KgsmGyXLTorrn5J9GsoSt4DTdSihJWzTnlw/lR7kUov7UxsoNxQyFoPXXLTIBP2dJ3fkIeETjutCppdxVoXmNewi863JgOr5SfnWx/v0tOGuYVnIZzLneUiULsF9hwgWNaaA2jRHOwsY6ichbzv6hLxcxtv2du2nmW77dM2b0r/WCHpJdRRwbVx2cekP4BGDZgh/Sis384ckK4hdi3r9yBKszVQV9Zyt+ZR6kifUX681qS6jlJgychF9YuvDRud6Dg7LqrMh+tYMbqAnrEzGH7znnmznv+5hXz5DNAio+sDVv94jf4Tsmj1IKZ/OMU1+awHw1InMmjWx/HlWX6h58pFnF9fxWpwcx5EfkzQUMfJ8fLz2RoWR8gwUujItkQesqP8jBo7209XATIeE+kM4sRngoarr0tWKYL4TI+vNjDV6Wu8+1t0PPsj3vz+rTCQlcGJstrm5xRuvn+Km2I+wAplK3TbOux07bTF5CPo3MBGS6iRn8eOLJpr+vFvFDGRlyrnMDnex0+yA8jJy8bPHCrXxEC17+YS6grNvtIMio06lshpxv4sssnm4NPSz0WfShLlKpBiMd4BPdNBBzwE/5mSwIkV7DXR1q655GZdLdhrOvGcHJFbl0jTop8mWuLp5ls0oawTVvi4ibfBnCoUuiZTHvM7Sq1LxlPhA6A4UwKfhTwmyCZ7HD5ORHvKMxcelvUsh1o/LfMW6d+H8clcIN/mUp0qKwvZIx21E6WQfDtzoK0Z6/fA1a9r86iolKhDGiyJXtLasNBx6auIeZgPI0YXeo+jnxlJH7D1phe9Z4+jf2ENz/EB7xRjCuhFb69NLQonyfDitOKhov6ucSE98qq5/1w84iNDY4eMR4SGHwt9pZXVzDNZVkWe1PgqY1IdklE0B1ySuihhS2V8XNZjODGidHDNjxqWPK7Pq0oj6di1BswYJ1H8+NrpB2kZKMgwhBqOMvz4V64hsLnaqqE4hH5hd8QxpFCk5F9QvE0u2VQqSf3M9DffNlUiCced8UggmtBslpsPMuOZrJONZpJ+CaKqza0NNHIDyAbXshjIKQZMcF0edEkmNfSVUdeR5BN+Zgfcjwj8Rm37XhAdM/4hB9tRunF945PINd9ik3mXyeayfnkURgrtNuYgec1wFvF+NK8Ke5ZSsJvvsUoLQKIOir7a0Nipc22Y6cT1jVE99BeiRhdO4cuLr/HuKYXgetDDzrewdW8NixdH8UUqdSicJMJc7FMNLaUigJ4TVotOEGiTR++3+OHkKlafkV5ncCYg3w4dc99kWVWdu4GvSk895t6oeMg37BOXlYzpsN16RAbX6VXkZfjy/RzOWzvvpsGM8W4oHt2xdMOV4cUiwtAfvbw0gyILMVJoMe2Pf7tIUChS8GefaljSLluUi6ufmX5f/1CUROJZ+zwSSTo7mOV2DUnWyUXTpZ+Lq9ZGnszGBqTzFGhho6F4YbTuYKHoXc4/3VSzdUy2JJ1lFGN8wgutjQZyA6FZGLaoR13CQyWZ9rivhJmhOpZWKLQ4iVQ2l0q7jTlIXjOccLwfzavKtMtrqw0dVClix22uDTk+rq9sOTqfmtEFfPH1ON7+IUJwAHqywMofWxjOytCiQ7lY/tQjXAgS0R3j9CYKX63fxa8iQX/7eKUP6AAAIABJREFU3kLoMeuQxxdf92Lu0l0eWiR+7dBx9XXJquuFXeJroMcuMS+Ulqxu6tumrAGJv7fxSvGCPb/ejqdrC1tJjkti5MI4EOQTOWBPgK4kWwrF1PC42u6PvyWcpMMaC2HKZGGZUOySTRBz6eCiTyGoRhm3RZRps3pL8bJpgnbKQ/C/JZOLVm6HeTB9/RhCDTLKRWGk4B7mklsTLXKq6cSTsgWGLpou/SIMbCfKfGt6gcI0Ns/QrmRSeJJXROGxMhX3dNVuiVzBzSpu1YAhu+tVJLOnWHs2OLpwPT8xhHKh7AgtKvrr/NqZA23NWL8Hrn7dmkdVj3Z0UMfpxynWhj6Enbv0NQ44fBfj+3R91gOsL6G/zI0s8o68Wt/CdDWF0YVj+O73Of72ItN1ENMvqik9ZCo4PAn7wukxnAQwPHsV57EqOnTIQ+QhfXlF6tEOHVdfl6yqTuJ4V/ga6NEl2ovr4TZOnh7DHOtiwz0qKy6OM49VkLtmIY+zk5geuYzi8SXWY3h2DtMj0zyHTOa9mcYyQ+ou5k6P4d3Dp7jp6tu1tWMS5Ihdo31xljeQyWZQZqLnUGmtKSFEID9RRKFQZgnnaYIcYD/AZZSzGWws72De6VDoQ2ltmb9dqPNPIRsb4uznoC+S/kezGWQA5CoVFFE3T2DHPDj/jUwWGQI4V0QxCMnkca2SQ7aQYcYe578h+LvkNovIr/Ika6kTXaMXBXhKmYOmUz/+JuutgRbWZG6aKoI+33ldhviaCod3KFOM5zVUclkUMvxViFxlGZVcgecqikVbHNpANkMzTXPdwjzFyU15ZtQhAY9Q/j08ops+gIk0mMe+Y+3MQbSv/XsQ7YdikcnHHYZdmseIHho/uNaRYx5YHqVlbUT46TSi/O246OMOz3mm2Wzu/OOf/z08EnlJ9hEB2gZjASdetB8C3kchPwlWtP0KvUmX+o+8IHu92WlqYY56R3rLK4uNGSXp+yiotFnF1NI45k0GwFGQ38u4BwjQFhG3MNBS0wH2gI0n2RECtJVPLLzYESU/6GggoIch6W1GJWx4NJTwUhICLOxVnIh4vzwynxYCm0sbGGg7sejTwuij11YPQbvCxh89GEdDwXh48WjI7aXsBIHeb/Hz7Cq+SgxDdkLcj9kXBJiHizY6pQTZ8J3FfeHtmRwqBPpK8ygdKom8MPuOQF8J9yt1ZB2pCPsuk2foRMCHF53w+EaPwP4g0HZ4cX/E8lw8Ah4Bj4BHoEsI+PBil4D0ZDwCHgGPgEfAI+AR8AgkIeBzupIQ8u0eAY+AR8Aj4BHwCHgEuoCAN7q6AKIn4RHwCHgEPAIeAY+ARyAJgc6NLvYmnCh0rR4ncTxM7fsiNxXLNpflOUxQdFUWhmtYkojylcJ/JTx41uGmuV0V8hATY28kqUVkVVnplfAubxCp8VuZEmVDgiLXKv/OjlWaVUmflRmy6anw0eRTWg7+kMkmy6yon4peos+o3IxVSm3Ta2UKmW7PseR5WD5tuh8W+TqR42PUqRMc/BgnAp0bXbQh5/sfO9j41CLPvhhAFt7+cjIC7cwPWxuiFNSLqxiGWt6niu+cm6Qmi/LR96BNIHdEQeS9UFa/OUT4reBxjTY83AmLbO9aBpVmPzYkfVZeKIWeEfk6FEbXuUMy5mF6qZUdLBdrKGhGa6N8G0nlnGln0OqtGorFIZTltvxmpu6re6qvm3Wq1m7MaSpG+9jpY9RpH+HbFavDvt4V5To3uhQi/tAj4BH4SBDYfIumUrqlK1qpNNXjrhA/nETy1yparcMiKpUmgtJDNrE3l1BvFDFxbQA5Vl/T1tFf9wh4BI4iAprRJUJh9yj8I0JC118CariIzunP6fmgnc5lSEmEICU6z+5EQ02sLt9L3Dh9F6+whOJxpYagyjd2/Q4e3CsFtC7c+yA5AHDwV3oFh4pMN0S9R9amXGfhMak7NUbaFJkDotrBn6He6WW16CGwt+uv8m5jTtkwE0/L/KhsOjlWMIzg3u78dcL7MI/Rn9pYuEmErqYea5JTuFGGtfSQ1hSq1VHQa8r0j4e3VjCVpX2+aijIEJbkR59Bm+QX9c1sEr2p6LVQIIMsOk2F/lRVqelIRJgcUpcMAjZSvoCRgU8wPqXOAS3lQMVZYiObVdmmqqiOKvLJPgmf/aX7mKy7vV3BprdUXLlYixtpLhkD/oY5pjZVB12/YKzsF19Lwaw7ZbDMjUo/kEXwEPMbX6v6IHGu6hGbCwv/JB4qzQg2VK0gA75Wle+E2l8uVMEjwEntk0lYLx1jquirYmGURZlTKOMyynUxzjoXTp0sNE3T6KITwSKj/N7QXIyiWqXwuzIXpEvwuyJSL1z6U9voFKZGiYbUvQ3ZTfq0cU0zuvjIxT+An99TeGgO5xemcbIkzilUtLCAB84CxnSTnwYe8vDSk9ktFM89wjYj/RI3Lm1h+oUIPT3sxVyJ2nhNQB6GkiVp+I2+X9B5Q31PqwbcElbwE96QnA/H8Wq2jueMh4u/CZklFP8cDegsXpI8SNYlnA/4jwOB7jY9TPT5tcU3PXgiMO2fvQxuZLhkdbURTZv+ZhnSzamNp2l+zHzSX7XhbpMhPeWPq+cKpgo1VquPygS1BppKAWj6ESoAyzushFCr0tRCWjXUcZ+17SwXwcNbvHZZjm2uqpUKofBIqwLetoMdOo54WzaxVG+gOGHalNUii4km472D+XF1pvgP55DQhXg3C/IHUe1n4RN0aVPnYBzh3ORhVQp7Lg+hfFkUYxY/6lK21kA9LJAdjI8erNwuoxGrGNCH0gxQkDfp6BAAK7hdBirXOL5UX7NRX1JKEbpkVImZ5ljDl/TLmvBV6ZiOXTIkzY2JnrxmmjfZpn5G9YjORRJ/G48oTTb3Gja1OnCfhcOXUawVkLksztl35Bb0dD2aSzIE5Jqxr2fSrVNMub7NSot/xyc2EtclR7JLOEW+o0k07XMYxYawCH/v6HcLNRXfBspl+smj3zw+F6PVLOaD3y3tN01lqx43asAM0aD0hnZkV4l0dmw0us6Xv0UPo3cMJ0aA4JwVMH7NCx3b+G39hZX1cXx5lnfouXIR59dXsRoYasr4sz/ize+Sl0Zw6z3eIqQDVrB6C1sBnUHkx0Tx6s96MCyHJ/KXHeXnIKZ/OMVPVDpkCL5/iptCD85fjqHPlHqIIQGGOIUvLwKLf5IH0YGVq43RtOiviqgch/wdc5rIUyG460ML7vsqw66V2HsCVNYjV4G4D6OvNMMK2jLGMhQlbCDW1qhjKSgYnMOkLBOTHUBQ0zmt1KyAcQ2P5aO7xi9CRmuLyxLpHT/R9GTFjU15bYl8dqNzgxdjJuny82FOmyZbZA6YJuQ1DD109BROhah3WPVmTVUq9NtUbyJKO+MzCTlloOLKjTKiqV0WGRUyxkMW1i0isJdZ4eYm3gZrxTjKctEiQ+LcWMixyynnzTUXifwtPFJgU5wpgdfozmIgBwTn7Dui4CFV1OS0rmfZHwoNde25dBJtM7L2JisiHRC0H7hoslEWnFw6JdJUxHHRoYL3O0oNVFFgXBmN4rLMA+XF6RsbLbU55XEOvCg4q6nGQ/rW39GUJFN2MxpdKcc6ulGYUIYXp7EY9CRDZg64JNscb/X9vY1XIz3C+CMCZCwoho6zZqCNfyCIctCL3l7lVD2MhDc71EOlJ457soPKVZesrjaH3Ar19g9dPNunZh/hkn+/ZLBLd1haNt82E0RRb/gFxQtGw4bQz+8UCTRszX0Yn8yhJqwuCn2hcs1R79Eli40Hv56spzrexadTnenHfhkohMaTdEjFZeM33lAiJZGePXEDQ1bg+1C6P4l61JKiX36WQI9GGdnAgOPzKfEHuyGZZQxlsRy1NtDIDSAbNJMOyo0+uJ50YMeJj3TNjYt2unlLngsXfwuPrmET6heXM2yLH+0VpnFO4ZX2cUrWyUUz5JxIJxJ61H/TQjp01Nc/BDTfKt7gaLv9TF8L6WS300vfske1F+ltNdubjcKDRDJSTs+lO/jS1Je8TuvbLCzJvW4f8G59ECdSvfnm4p8SHDK4Tq8i/+IpfmNGGeU6LSiDU+qhjJCH263XGM6Sl+49wN7sM2DFPHoWPQJvn6TYzc+D4KnLb5FB7/YJnLMfFaeedMOXT35Kx448GMp4cci8OpnHWJnP4m0dmLzvsuI6lyVZT1W2zvmoVOLH4imbGiivpDCFCcKWftgjfy1sNCIXwhMKpy5vIFMYxUDLEuqgfC1kMLVUCccJT0FsLukGlL2F6rU8uEPDImNIyXxEns7GBsgnwGeQdFCe9s2jLFddMljmxkKp3cvxdaLPhYW/6/vQVWy4RnE5kzTtAFOXTknsWIi/vd+NZJ0s2GuyOOmw9V7HZGsHa2yhUq7VLY1CeMoMuKEJ9OFteLGjo3Syd0RaG9R9TxcLQS7hPzIhXSR9s3wrduxKOlfCh73H0Q+FzrM1LDq9W0IzF39NeecpedoUfs+vK56uRD3ilBcrIq9t6xF+WQD6TxwDXLK62uLku3MlkacyP93hGKeSKEN8yEd9RQsxbVZvhd6sWPhPS0x3ApM2tJTHRLGGx9Ul1KGEvnTau5KFwnlaKI097Rr2I9sVH4fONn6kpyZbZA50HFj/eSwXG0pOWLwTvd3YLNMLDfzPnANGFhIl1DdQp5ixS8Y4CwCKvn39GIISKqYQj80TqvVlyf2SvkuGXc2NZJDw6ZqLTvlr+sKFTYJ4QbMmp3PuOsVU6Bu8EbtyO8zp0nSKzGGnOLl0aoemiw55HZV1uTIV93TVboW5lrfLao6pfb1H9A8mSRy0I7s+toPzPfB0HcN3v8/xtxeZQIOYflHl+3nR/k0Pt3Hy9BjmYm2fIz9yF3Onx/DuIeVS8eTtC6fHcFLv61TUwd85Tms8O4npkcsoHl9iDcOzc5gemeb5bGcdemhk5On5k9v46viYoHVf5Iq5ZHW1Sard/nTwZMaQOj/A8+tj+CV7H79dEbl1XRHHIUNX6B81IjwpejSbQQZArlJBEXWhRB9Ka8v87UV2hfbXWnOE/+SwcUzmyihnM9hY3sF8GHMygkMJ3YVCmSXz2/1cHcoScIzqSZeLyzvcuxN5ou+QD/thVXTW3wWQHqpsBmUmk4qlJluxyPLqgpyQQIfwID+/jGKmgOxUP3auhdeDo74S7lfqyDJmch8zXSjem+N/Gyulee5FM8oYUOYHMX01HaDqp40Fz5XJFjLMwOdrbkPQFZ48owwdzo3O3nmu6RGZi075azSd2DiFUxp1msp6VnqxQ+fac+nE2zYyWWRoHeWKKAaJm445hIumLpx67tKpHZoOOn3XUMllUcjUGONcZRmVXIHnWoofn+LQBgvBU4dcpYU19rXRftPyLv1VnejYLTtt7Ew5mmsyd04f3uZ5ptls7vzjn/9tc5jv7hHwCHQTAdqShN5OPJR/9CSe3cCMKYx5KAXea6F4yMMaPtxr9p6+goCfixAMegsvi40ZJRE9bPwIjo6+fvSSTffDix/B1HoVPAIegRAB5pqPbX8Qtn/0R3r4pxvhp48etD1S0M/FHgHrye4XAt7o2i+kPR+PwFFDgN3gMsiWh7Bs2v7gqOnTqbwsFAgWjmWbMrL9vAxJyJ3S9+PSI+DnIj1WvuehRMCHFw/ltHihPjUEDnV48VObDK+vR8Aj4BHYAwR8eHEPQPUkPQIeAY+AR8Aj4BHwCJgQ8OFFEyr+mkfAI+AR8Ah4BDwCHoEuI+CNri4D6sl5BDwCHgGPgEfAI+ARMCHQHaNL3QBV5+Jq0/t+zOfdwqFbdFSs950m7e5v2ySXCl47ykOpcruOVZ3UY9eYXbW5dNoFYZbMnlCYmHZPD0rHGDYU3QV7NpTR3wO6u5WrnfFpcLTQo316GL6jclNGS0d/OT0Caecjbb/0nOM994NHnKv9ymGTxy6pb+kAge4YXS7GtCGqqcyPa8xhaevmzVrFYTd0VTqHBafDLkdazHYzL3uCwQqmsuGu5UYW9APN3qbbYft87SwPoZxNMNKMhGwXeU3AYnEI5Vi9QNuYQ3idNqDsaJ8xuXHpTlgA+xCqFxPpIG7c7fBMOx9p+8UAcFxoR04Hma416fLshc5dE9YT2i0Ce2907VZCP94j8AkisFkdRSZTQLNYRLDJtAkH9gOt1PijEhtq+RfTmHauiZqAE9cGkKs9xko7Yz+Gvptv0VTKknwMKnkdPAIegYNDIG50URHq42Phv+svQ+mYJyBsuyHrK8oeytigLeY9oDCMpHEHYU1GccxoqSEmQ3+1z71HuCDpkayqjKrssNAR8j24Vwp0vnDvA0D9T9/FKyyhaA2FIcrvuBIWI7rn7uDGOdL1Dp4HONjoOuQz0hGgK5izeYvoLCfG1NcQ3lNoBfNHQ1VMVSwCnQR9/Vxlr9A+eX1NbTEf/xmuQz4fspsFJ9ls+tTlUmUJ9GlzXoiPSieNTibZbNf6Z9Da2cHatQFbD+N1Vhcw56iRyJ6qRbgskxwyDDZGZTUAawjqvBm5Q9QHFPSnqqiOZjAVWGq0e7jkrXjjxJN+lRmavH20qtT+scpMO1RnMFWlmpOS54omg2CuexMiNFUZFcWoD/M01lDICD6jU5gaJV5C/khoN4NMqKwmh4pFG3IzcdrFTXpISW7HHLswiLQpNNj1KcTnysLTho+gw2bHSlOuJ4G1qx/hxNrlOlDxVuYUFjmpiyKrOo2gMXJ9yXlXSUaOLX2tshvkEX0DbPQ11448TkzalVUqahknm9VPBdNMbC266Cht6u+Iig3x0c9t2Ih+8XUrhFVxou968JtF7YosifOvKm8+1oyul7hxaQnnHz7Fm/dP8ebhOLCwgAdbNJjflPpl24ureHtJNZSWUPxzNBi3GGmTzMmYmgYEjSezWyiee4TtWJHjv7CCq/j+rKW/JAdg8Q/gZ5L1/RzOL0zjZEmcv7iK4UD2JDpLWMFPgeyvZut4Dl77cRjjqL2v4rtehWlwmIDJ+hJQJtl+5LUn2TgT3QT5jHSImGu+AiHFAfXdwvQLObe9mCuJItysh23+NB0f9mLutDrvOh/TeVTOJ9ktLJq6KdcW3/TgiZjX/tnL4EZgAk7KePuhDYd256V9newyxVv68nnY6xzG+8sbDtXyq9wvWcbyH/ih5bShyBXcJnrXeE1AqgHYqC9BMYc0QaL0WwP1sAAvyNAoAIJ3q9JEIZIjVUMd90WItIhG+bbwqkVpmsKntTpwf4d0WkaxVkDmsjhvVZCr3YJqv3GBNZqtCpoFxQiUWpEXkWigiOWdHcyPA2jUgBniRZujrmCqUGM1IqmE085yEQj4RXlEseAM0sndCW68th2XW/GCSr3YZ1S+nQgGWlssZG2aKxNPFz4RYQCYaOp96NzWLyqzCW9OzSQnp1t4PBGsv1qwHpLwV2VM6muS3SaPQjey5pJ4KONofWbLkN/3KCZJdEyyEu2kcRp/PfXhssyLdNHhbc1Ki8/HxIbyO6LS149dNKmvTacoTtHvQhJNXYbkc83oOoWb76nYtBh4dhTnJY1na1gcIUNIXIjlyQxi+odTvPGzHgzLcern1l9YWR/Hl4JGz5WLOL++itWtYzjzzSAW/+Rete2nq8A3n6PH2j8ker78LXrY6TGcGAGCc2bIveYFqhPpDCI/Joo222QPWYZHKTA58VnY3XqUQj4zHcd8GZkJPKjt7I9487vEji5Y5m/rPd4inDOwNbGFLWaIG5nEL2o4sXmP94pcCeYRp/DlRfC1kYhThITjxIWDMszFrwOdFMrdP2RhRrrxO3K6WKisiAlZVzkpFEnlblSvGfVvlGFN7WL9KxA2GvpKM6wwNFNWhikFb9bWqGMpsOBymBwXZmZ2IAypppC5OCONzCwGckBwzoo+N3ixXBVxTU60lUOTQ1joOo95MsYieApGGo8IFqJLICcccneKm6qv6ViTL4JBIuaWuYrxceAT65uWpqWfpo8J7xjLyIVc8HCByPpbQr0Rfmfi61YhsldzBWXNJfJQ5HFhkkjHgnPiOIU/O1S+f/n5MC/SRUe0zcgC03kqgq3TNZy7aLLuFp00nKLfhTbm3yCS6ZJmdOmhpOnAI7H9Luku24teozdIZ0vhOhleDOn3jJ3B8Jv32MYHrP6B0Ahi4b14f51q8rmZLx+XVvYol+5hQnQ7lC8S+gvxjEpKZ2SgzQGXJJZKKJR1tmDw9zZejfQIw5Y6knGrGC1xRrEryTjFhkQu9GQHlXMXTko362ESDvpAM7/d6qRz6dq5y5BqbaCRG0A2YEY3e+VHMbhOBzyBHo0yskFopYAaPS8+jvjeg1Gbb5vBMT/gxkR4kYfp+JuWnFbYNoR+k2uvLZlDaq6juJyu3nqbJmckLBHqFOehY6HTdZ13gJuLHM1ubK6UAYmYaxgoQ2OHFnxi/VLnzZl5x/VpF28zXS6nC39dE1dfFw+djnquj3PxCMclY+Kio/MM6ZLHiMLt5u+x2o+M7mWgIPvqYbu0dFSaSccummad4jjpPFw09b7J5/+IdKEb+OlV5F88xW/MgKL8mQXWpedEKosqQs58QuE6Ndwme32LH06WsPqMIqhn8DOxY3aerb8cl/bTQifJlnSQ7x4mxKQD+RzzZRZbeMaokfKRLt3Bl8a5UEaT5299G9uAMLw+4N36IMyeN2WccrhbnLZbrzGcJU/keztOCr/kw3ZwsMxL174PydJ2rQc9wTc20AJE+LGFjYbyFK0yEk+Ny/obf3QTzd5C9Voe8kFUDuvrH5KH4pPoq5coTGeoWRh4u9S+4rgdmQ3DTZficpp6pbjGsKhjsrWDNWYwUu7HLTYwzkPHIgX9oEsHuAVjzQdx+ZR+3cLcgY/CrSuHcX12g7cukgV/vRs7t/R1rXEjHddFCw9tSDImFjqJslrGafz5qfB20gnldxWmMBH8BljoJPI3MhIX26cZx0mnb6Gpd0t5HvV0kVcDocfj+XXFc0JhpfW7+FUmzzMPiyEZ28U4lrtFSfBhftAXX/di7tJdHlokOgn9Xawibbui4wil7QoThW6n8rnmKwKA9GC2OV9sDo6jH0v4j5x3CqvJNdIbbaOw8CudL51rOG3fWwg8qKbudG2xIvLNth7hlwWg/8Sx7qyHxHWbcl460Mmm666us0RVJR+JQkNqSFAl3tePIdQQOKrIrW7xMKzcLqNRnICMnAVkWEJ9A/UwLhg0QQs/sqR+2cpCfQpv5gFR5Jb99M82ZNaHWs81OXk+nJIsbh2oNZBHSMFvZSr0dDmx0Mg4TzvFjRFt4q3tBubCYFeYKzxd+DiV7qBR0yey9ozkFDmN7eJiO/i30zfGcw/kcWHSqaztjGPfccv3ykVHtAUv7azcDnO6tLXJXvSRWLpoyj6mTw2nyO9BpzRNfMS1qKfr7CSmRy6jeHyJNQ/PzmF6ZJrnRfXyROMLp8dwUgymhHuWYJ7aW3QM3/0+x99eZDQGMf2iGiaZ040MwJdXRH4VEvoLOZI/OqTDjKG7mDs9hncPlVy3gGGHmMTodiifc74CIfkB5eA93MbJ02OYY1c07LXu4amuozruFL6fHcRXl8aYETU8exXnmU8sHM2PojR4v1W9U+T8/MltfHV8jF0bnr0v8gw7xEml7MKhrXlpXydVjN0c02adtwZaWCNXU34ercooCwFymvypzBSpA3jS7mg2gwzrnEOltRY3rCD3poqZXGwUJdQXCrexUtK9Vhr9YpHldPEcqD6U1pb5W2BO3lyL8H+NJmwyhyOSj3SaYMnwuucukQ7LNcmikKGgK5CrLKOSK/Acsj6NRwSLRMpKhw5xYzeLMsrZDDaWlbyzgLImH1QM9LaUmMd4Ui6OBZ8wxh1ItLsDTWYX3rqcTlnawb+dvoq2eyaPC5MOZUUb4yhXcnkDmWwGZaauuo5cdHjbRiaLDA3MFVEMcrryuFbJIVvIsFSHXKWCIjYEmC6aCt6xQw2nyHehU5oxJsGFTLPZ3PnHP/8bXPAHHgGPwP4jQNt90BtwH9cfD7cNtGxv0H1c2rq18Vi48el2q8c7juhRxYTeIMxiY8b08BDX8jBfoTy4aHjxMEvrZfMIeAQONwJ6OMERvjzcinRBOo9FF0Bsg4THOw6WxySOySG44o2uQzAJXgSPwEeBQF8J9ytgIS32ZhPbo0cPQX4UmiYr4bFIxqibPTzecTQ9JnFMDsEVH148BJPgRfAIfJzhRT+vHgGPgEfAIyAR8OFFiYT/9Ah4BDwCHgGPgEfAI7DHCPjw4h4D7Ml7BDwCHgGPgEfAI+ARIAS80eXXgUfAI+AR8Ah4BDwCHoF9QKB9o4ttLhluaBqRsdO2CBHTCRU6DsvXUP7Lyeu8TmOsN+20ftywCSiTTaPByhEZ+saIHrILLpypCLZJ/xQqPL8u8KEi5Cn6G7s4ZTOO8BfbRID26WKJ6pGC0QYi7O2lFBuQGoZGLjE6xNNEi15Dl230ancGGaNcvN8oVZ8O6IXlQZg+GctGikY9BK+Ad0Tij/CE8GsHn48Qgt2oZFxDuyFoGsvX5JS5SpZpQHevdUPHbtBI1MqxlhPHHv0O7RtdaqHrdm6w6rg2cXt+/TLmTs7hzfun4t99TL+ZNhheH/CgsoTzF3sx94vJKKOSLpKG+HzYi7nTFsOrHf3a1GlX3XeBpZ3vS/xngTY+faoVwraPYC2HFaMEsbsq+75iIDcu3QmLx0pd9/QHM4dcTtlRXvJkxWJzokA130iQimJfJuNK+WM7hOcquB/sQEqbuO6wvclofzL2z1isewVT2TIi1YRYRZEsykPLbNxysYbCgd3pFCUP6lAt2L2na2CfFfyYdNkP6NR1sNf8/Nx0jHD7RlfHrDodKIyBH04pBI7hu+pVDC+s4blyFVt/YWV9HF/+0BNvU/upx2xX9/YKOKvDP5rjrfd4K8v7fDRKfYSKUJkfpfTM/mk4hMnJHJpaXZmVxzUUJycVMfKYXy6iUb6N4IHlrisPAAAX90lEQVR/s4rLZaByvyRqPird1UMqx4GwHMpmdRSZTAHNYlEYdbKzMDyv8R3z89cqyNUeh/xkN//pEfAIeAQOGQKK0cVDeDdkjT329D6G4JzCdhR2Cp7qX+LG6bt4hSUU9XAWC/HxUFUwPhgn6wDewYN7JbBQ4fExXLj3wQLNMZwYeR33XBm8Paz238VRfNH7LX64uIRfrDQtrCKXLfoJXLjcFg8Zo0N4lvDgHoU7RdhOhkSJxrk7uMFCpiJUa6KrYhbQHMONe9Galax4dcBjLaIFWLhRhlUdYeFgLg30VTkiso/hZDBOw8K0BphkFP40yBOh65AzGKusTaLLZJR0FVmE7PG1ZplfC17bbK3qdGndPrJ/D7SZaOuUPUnK8JsIK9E15vXhVe+jzh3pEaI2LQzF6jNyWrExLDwnw4NuCfvHJ4H6EkIfFhk/RUyMa+Py8wi9T5uocosrViBbGwWQ16w4E/brn0FrZwdr1waiXXXDk9VjC421aGdEw5lTVRYCDXGQ4VENA/EkX2WGH8eOhUYD4pZxQbtyoODPwqiSeRIPddzUY4WgdijorMCyBlQ6+tqIkKLw2Ciq1Skevqa1oco6OoUpCh/LMDPjy7HJROiKMFu1ilG5voiO2l/SJf7q9YCORRfSUdKUckgaunwR3cSJgoUqAivGHNBV9KZhypioniYGAB6H+KVfMw69MlOwr0NFhmAdxHWNyW3EXKElDyO6S1zanBuipdJxrWXJ9yP+VIyuYzjzzSAW/xRhub+3gZFBvH3HjaHtd1sY/uZz9ARg8Npzw6CQXZXXYGRtSyj+OcrDgA/HsXjJcgPFElbwU9Dv1Ww96rUK+PB6e+cXpuPGS9CHDl7i11lgWnjEvvh6HK/++Cs5N+lZHXPkHTsbIQbApB+/Ufc/VEOTNv2I3mvMzUKENOdAOgTG5foSUCY6P+ILutGfvosY3Vgh7L+wgqv4npckFAK/xI1LS6A6mBR+fZLdUopJk+E3Dci22S0UTflaZMC+uAo+l09xM0JfxwVAIPtTvAnGpVkDCfIEdAkT/U/D6MVVvA3WltbGQsbqvJjWmml+7fL1XKmidjE0/p//cheYvY/frnyLm0YMdPnbOec/akPLWtiNwgetCnLgobn5SGlEXj+Mt6lld2ooPJ4Q4bsiagWZl0U3xgIgeLQqTRSMuViK3FQjDnUEta6ZkWQoig0qCbmMYq2A0dHLKEMNK0p63HDkuVzixl2oIccLNbJOffm82zMmSTk/o1i2Buph8VwkYVBDHfcD7ELvXdI4VaAVTBVqrLYjD6EWgdothNFXG4/ouNZAk9WaUynHj01rgOg0UWkpa+lyVTGcdSoNlMu0LKi/mEMpbKMGzNB12vA2iuuOITRcqwP3FTqZy+Kc1nCAgY2OSZcE3CPy6XrRue27EMV6Z1mdo3bxA2rNAfawQPgNlbPgxp1Ldlcbl9u8Dk06ymsuuW2Yy7Hy04ZLu3MTpZNuLUsZPr5PxegCesbOYPjNe2aoPP9zC/nyGaBFRtcHrP7xGv0nZCFqFxCDgeGDz/7/9s7vpY6r6+NfIX9A7w14glEk5kaEFFKJhAitkdKHXigvwQRC8SXexF68aTkUNFCkfXsRe2N5ghQakWIvyhuCMQFLMCTCExBvTBCt5Bww0P/Cl7Vn9pw9e2bvmTnnGDV+hXLmx97rx2emndW11sxuQ49zaDcGB0J53nEi4ALuRr1YQfCSaKZ/9hKLvf3oL4UKZfHs9Vn8qjN36rBk5XQ2JPwdraD8Iu0hn2K4KsEZAZpaoLuCimfB75F5LTtYHHpD8RTZ3Wg/E+pwyo0HwpLJQyzwBaD8HsfNMGhs++qaWjRcSdblVvPc+ipWPfameJ1yyLA95az4poPf2D2QaY9HruUnzEynk582Lue9lmHfpR+DXsLJb+5hbGscP0QLs2s9TfpVmZwx/EsHVVbZrZiWi5gJy3DoOFcr0/29hD/WajrO3v4OY2tGQJWq5CyCZFeQ61KlxchIe0JYZlxzlRVTerp25HP2N4yAxJZZx77qOZuBRqD81GIyGVzE8FC4dHjd7Abx731jzTh1LbUB8uvQ4bPbnJ5rey1YhFvGDv472QtoyRh7rFcRCBYXXnuzE464iCgmznGPjn2ny8kdOHcRiPbVAs+hTTnkRObluF6RfdEkc8Px74IsBO+9RgX5RX4P4l8Sv/2fZPk8/775zinzHfeI6VrqtsPu3MyzuBhKfT409V42dB7TzVjQhdJpdKqH8jtUtkooXT6NTtU39Q67qdmgNK9LKOnAJ+10dCzvuGhCuBEGYJJdWFjAAxVABA30WJ/Fp1FQVVYZnyhzp2anNNLHsnS2Lmv/bRUbvW1Gti8ofe6+tcY5dtvaS0AY1MLsn/LIrQXCEviiFqiGOiQD6f8zA82AiX98nrNZ18533mePe57XTw+/wBu33KS3PvtacX1iCIsLSxiZ+NK4D5JSGjqy8wZrF8+hIxIiDy3jP6DR8Twb59EZxg3J0Wa26WqOTApwNioxhqVFHRgmhSMI8nz6rUmqTFivn5ascPfv7U3rRBAA1A76GPhs982rSVdbsVKOzTldR9JuS2buXXlwPgau6jJgS5h5ySfgbOd5YHM7zIwZtjbrHi0sx8fdsC/VPc955zVqjF+HRJvRn8923zmP3ZFse8NjdxHmTi62PtlP96F593KazuN3LB504QKuXHuN3RUpY0lwIfsVVOZeYlF6pQ7DP9UbZJaKQiNU6S1sgA8zFIk3E2OBWROMl4zcetUoWUow6snOWCpV4NB1Ovmw9smV/rSuVaw+k2tiZPJC2SqQs/TEd+1A0ywDxke+n7367PH66eNX2CmffVLKraA8b5Y2CyvIniBZlbU30PkFYAdv1owsQ7aEnCPsbJNZlnSIkMBIMmLL29iMBYaO8YUP5/DT7uFS/+ee/mBSQUPMBmFpHqiDgZqec548tDr+wLAu70nJzlTv2E7a7RiY63CQsdDlzVqJOXuyemCe70yWeZt1jxaWk5N7tmu1EZnXqH5+O2/WjJK5z3bfuZqpxbYcdudlnsnFtibdh+bey7bO47dvBV2A9EJtP6qVsdo6gOVHFfR0uEqL/vJaw0jU24VLiV6k6tz/Ylr6my4D0l+zkRYUSsBy7TWWV1xN+nmsM/yTTCCW8JcuWUrJy8xYpYhbnNHfvJKes9cY+cx8CzOckCH30mclTI/OJkuLMt0qo1bnFmo9XYmeMKsBP8VedciyR72g4BqrjhuMfOPqtUdkWn4GjfNhY7tlryq5ZlyXmpmG7V77gn6v7ak7uH75S/yQ6I8z5NSE17elggrj8wySns/9xqKnody0RpV4DB3q/2h1v5c50N6WkskaJq5OAMNDyYexPbzAfvBZiWHoip57amjDT8H7kcs/TWBtLL23DFLOW5tAOBRKhxZcL4Mi8ySrYFy75f+2M13aGOvXZ7c1NLlr3APqulovViQnxI7c/173fC3jp4k1jKWVkBu6Rw11mXIMX4pwN1RkbvquUSP8/v4Z398Hzkuq2We771ym8Y4BPrszmYcyfVzUkJzXpqF72eHfMT6cCLqkBwfrtf4tyTBsrCfLWsrnMNs03We9SdZUINJIv4L7XWbpcACfPurH04dS4kn7pETNANVQ72zSr41L3Ur4FzRfb4/m7wcb6aqGJc8ytqd+w91Ew75ozpCreseGcCu1hyg+91v013q6ELyEAG1v3yoGc/WvBf1ni+G8uEyLVIKRdT62W689KYzUiwc6axdn0JW3Ty9hu9s+9a041Pq4VO+clLPljdSEHEA+NBu9NBFjkGcnaFTd1CUh1Qit+2w889V/vNcw0ZGnhBR8UysqO6lsTA4d0hYkjSpmL5LHJPcpsxQRlL46Js7j8UvdB+SeKWd0o7404l+9P4bH8bcKjMlxlh1vzkM+QBH0/tTLoMC8wf/BzMWar9+fe4yZXKXiuN03MJwrQxY83I17QF6+UE3u+mUFaar3X+ex82/Qod7ku4rNmR2ko43b15L3HjWuTLDpkZO4nwtwT+jxHPBdo0b4dch3UjQ/n+2+cx67fae8dnuYmzK9XIYwfNG4z+DzIa7PvpflY8/xtzxNIz687ZbNzc39Ux/98+F5dugeSWbkBnYnVhyB1qEbSAMOkkDlT0yufIy7qYFyUrG8GCLlH/4dNAF5Nf97nNvJUUo9aFOOnHx5i64Db74zGv+PnI00iASOLwH5n8Rkpuv4+kPLSeDIEKiuVNGu3849MladQEPsMkuhUu0J5EWXSYAEDpTAqQOVTuEkcEIJtH31Na6fUN+PlNtnb+O3mT/Q0dGCCWXYRczsvITvpcsjZT+NIQES+KAIsLz4QV1OOnNcCbC8eFyvHO0mARIggXwEWF7Mx4mjSIAESIAESIAESKBhAuzpahghBZAACZAACZAACZBANgEGXdmMOIIESIAESIAESIAEGiZgBV3ymYOD/OZWw/YeLwGVnB8jrdsrz/VqUHd17nbwnakG5eR27X3pyW0QB5IACZAACZBAcwlYQVedwo/bA/Og7LXlmosy14n2cKbJ1/NLwcdYD8oHm9XhOEqtJEACJEACJPDeCDQn6Hpv5lLR+yAgSwltTw0fzlqb78NB6iABEiABEiCBQyCQHnQ9uQd5hV3+iZYysTMT0f4rTPbNYgNLGDsdroWX5ohauDpcPkdky/Ip8idyvriHyS/knF7Y+hUmQ/21YylCQxsezN1O2qtlR3K0bQ3am5Cry7EpciNGoe1qXzPQ9oQMTt+D0w8XO40k7Xrpc9FvTqZ4h9VHxrJPtg+RPL2RV64eL78prPRpw9dJvcalOpeuR5bbie6lUIaURu1jWjwM+eoeb8Z9GAnnBgmQAAmQAAm4CaQGXYtbbXi6t4KtvWl0Tt1A/OFnCwvWvevBEO7v6bXw7DGvMDm6hJF5kbmCrfkhYGEBDyrhuPUlYELOfY1LCBYWRjj2aWJhYVv2EpZxJ5K7Ea2zGDzYOyOdJUz3SVDXqL2W3Bfj2B7NI9eaN6/t0f54/PCxA5B9vQowrfwHy+sllEraLt9vAbkxMa5rsISxJ59E13JRcZWJbj2Xbo2jZ+ElnkfyJWh0LCwuwZ6PZUP3YWQAN0iABEiABEgglUBq0DUyIQtJy98FXLkGLD4Js1KpIvIcvIC7e8YahGoBZ3NeN9rPhPvqoT+EK+HC0MHCwqtY1QGaOU1td2NQL7dypg09+nxlD9uoyYHSWUHFKUdPlF+Pvc9eYrF3HDf1wtV5e54y7XH44bMlNDnzehVh+raKDROFb7uIXJ+c6Fw3yrcuBHuxaymBYO1axu4Jtdj0Ev7SWTHLpki02vBcV3W+kfswrol7JEACJEACJGATyFwGqK2jG9ixp9WxL2UqVYbUc7tR1puwMytSqlyKzgLmWOOw2rTnhucleOhtC4NHOdaK9t7X2H0LQAd4tihz32FvdVeitiAkNYdnbmfa4/BDBDtsSdPpvl4FmMa4pWkxjxWQa05L3fYwUOXrtHuiFf2fd2P6ySvcvXwB1ZVVYOqOux/Ny9LW30zfUh3mQRIgARIggRNEIDPoqu68Rk9HK4C9+rGoB90qBl+s4HdVtpL+nAWPPClVSqmxgT/JlKxXUY1CpHfYXTcyGT7RHnvb2nPV3ZLS67XHY0tSCeC+XjmZJuxM02IeyynXnFLXtluPynydfonnP7aiIv1oP8v9mvJXkCVUybzB+zDFDB4iARIgARI4mQRSy4uLM3+qYEUyLL8sAJ3trUDpNDpRK+NIRiFehvKU7lTJqpZFeP5NGYsu3olyUZ3furLshZQFYxm1Ou2VMuX6LH6Nyllin9EUD4fcTHscQHKwS71eprhmMTVlynbDch2sCuuRMvgS/pr7D5bRj35XXJyDZaS6Yd8iSdwgARIgARIgAUUgNega6ariU3nrr28WmPoNd1X/0gXcnOrG4mjw9t236MeIhqgeUK8x3aff5NMnwt/Lwyj3SqkmmPtLxzTKutRnDZUy4PWH00Cop6tPMmT1ZBuCZu1tLWe0grKW05C9lty+WXTOhy8QeOVa80x7EgyMAznYpV8vQ0YRpnawYYoBIG8LRm+0ZsiNj7UEeVlZYzP0yOhLnw1hcWoWnVE/oi0DQA6WtVnNug9rErlFAiRAAiRwsgm0bG5u7p/66J+TTYHexwjIJxe+xR38/pWjTBcbfUR2VOmwiluNlqUPyR35fMX+/v4haadaEiABEiCBgybQ0tKC1EzXQSum/KNNQHqkOqNPbxxtW7V1qtx97ZPG+gC1MP6SAAmQAAmQwAEQYNB1AFCPv0gpJVfwy9y7o++KZLhOD+DTqRLu/xh+buLoW00LSYAESIAETiABlhdP4EWny0ePAMuLR++a0CISIAESaCYBlhebSZOySIAESIAESIAESMBDgOVFDxyeIgESIAESIAESIIFmEWDQ1SySlEMCJEACJEACJEACHgIHE3Sp5mZZBNrx5zvvO+cQdywOm36Z28fC+BQjxYdvGl2TM0UuD5EACZAACZDAB0qgOUGXHURkLQJtni86931eCNu2RnSbPjciR89tpm1aZpHf0pe4hTIm9Zf5i8zlWBIgARIgARI4gQSaE3SdQHB0OfwKvF4yikBIgARIgARIgAS8BJJB17N7kNfXo3/MEpLKrtTOBVmOV5jsm8UGZJmfcA1CnYXRv5EJ7/DgiwFMzun1FD1zozmyOLbWaZUsY7aa6x9Gk4ON0I4Hc7cjv2pL2QCyxqR86ynwWctJsc0SC5j+hPOFlylP80uwMIUV9dFhm6lXXwtRE+p2+o+i+kPb1TqUq1itmL5wmwRIgARIgARIII2AFXS9wuToEkbmV7C1t4Kt+SFgYQEP1EM1eNB36nMvxrE9KkFQsKZgD4Zwfy9cg1Brstfxq8iCxOO4OaAHeOaqIRLUlIFQ59OpCsa+CBfjlkBBrV+obS1h+rY+p+Wbv0tYxp3Ir43oi+uWX/MlTPdl+GWKBbD4CPhBeO1NY2ShjK7b4f6LcfRE/KxJ0W49PqZxc/mhFbn8r0e/ltmKducamnoMf0mABEiABEiABISAFXRdwN29lXCBa1kg+JPaotbPXmKxdxw31eLXAHL1KLWi//NuLD4JGq5lqRZ8/jHa8rKXIG19CFdCnbI8zci6mVl5jd23obDLX2Pr4Zce2d0YHAjXEjzThh5tQ2UP26jpCHyuoFIgezMSLbIsQQgQ7YeLOkc2ap3mb7N8zPTD5X8jjFtR6gK2d4/Bl+tN5twmARIgARIggUMgcCqhU0pRqlyoz3SjDKC6K1FI7nBJT0bbQD96Hu2hilasPgIGfy66iLKULZcieUBgDyTDtjcdlB7Ds5Khu6uDQmNGsFlCqZQ4CLytYqO3zfDMyN6cSRl/IIea4GOmHw7/lT9N0H8gXCiUBEiABEiABD4cAvGgSwVcqxh8sYLfVYAivT4Lytu29rSIJQcIecut6zZWnwHL6McPIqZAFgmqbPm1YyHjMDMnZkh/1+g9XNlzjXXYKlmv9SqqUUj5Drvr3Wh/bwGX2CWlWZfdOX1syI8m6Hfg5WESIAESIAESIIGAQLy8KNkS1DIiz78pY1GTUk3Ts/hVfyJANWfrpnMZ5C7JXfqshOnRWU9p0TE30ROmG/B1c7ipXxta8Ld0Gp1Ywl/aLymjGgx8fhXUlD68IR8Nbpl+pKtHQ/rfobIFdLYXzV46bOFhEiABEiABEviACcQzXZeHUe69EZXzeqamUe4tB31TpaB5+7/6BtAVApFy3nWVAPsYg72zmO4bwK6U+OwsUdgbduWrlIezeui75rbi+kOzhNiN8oufg6yX9JTNV9HVN4BpZY9xrtAFs/0y5Ni2OUuXhRRag+v0MWGbxw9LY3y3Tv1KyGFkBePWc48ESIAESIAEjguBls3Nzf1TH/1zXOylnUeJgJR0Z9rw1PsCw1Ey+OjaIp8r2d/fP7oG0jISIAESIIGGCLS0tCCe6WpIHCefNALPnyxhZGLFeAnhpBGgvyRAAiRAAiSQn0C8pyv/PI486QQqf+KXLeMTIiedB/0nARIgARIggQwCzHRlAOJpBwHpqXvoOMfDJEACJEACJEACCQLMdCWQ8AAJkAAJkAAJkAAJNJ8Ag67mM6VEEiABEiABEiABEkgQYNCVQMIDJEACJEACJEACJNB8AqqnS15X5x8JkMDhEpDXiflHAiRAAiTw4RL4f6HrO9SvZYkDAAAAAElFTkSuQmCC"
    }
   },
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <center> Cas9 enrichment apple RedFlesh locus + Oxford Nanopore MinION device </center> <br>\n",
    "🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂🍂\n",
    "\n",
    "\n",
    "**Author: Elena Lopez-Girona **\n",
    "\n",
    "Discovery Science project\n",
    "\n",
    "   - Tracker ID\t2030\n",
    "   - Title\tCloning of gene loci in the absence of a reference genome\n",
    "   - Project Leader\tElena Lopez-Girona\n",
    "   - Start date\t1-Jul-18\n",
    "   - End Date\t30-Jun-20\n",
    "   - Funding 2019/20 exc GST\t182,200\n",
    "   - Job Code(s)\tP/150008/01 DS2030 CLONING GENE LOCI\n",
    "   - Portfolio\tNew Cultivar Innovation\n",
    "   - **Objective ** in Project Plan Development of a referenceless method to detect structural and epigenetic variants, to fine map genes important to PFR breeding programs. This approach makes use of very long read sequencing Oxford Nanopore technologies and CRISPR-mediated targeted capture of megabase-sized genomic regions.\n",
    "   \n",
    "   FIRST MILESTONE: \n",
    "   \n",
    "- Proof of CISMR methology in the red flesh apple\n",
    "\t\n",
    "- 1.1 Generation of pairs sgRNAs flanking the MYB10 gene designed to capture ~50 k. 1.2 o sgRNA/Cas9 evaluated with synthesised sequences, to establish in vitro digestion and identify efficient sgRNAs to use in genomic sequence capture. 1.3 o A genomic fragment containing the entire gene sequence of the MYB10 gene and its promoter has been isolated.\n",
    "![image.png](attachment:image.png)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Run Albacore2 basecalling\n",
    "* Run pyQC\n",
    "* Trim with porechop\n",
    "* Map ON reads to GDv1.1 genome using minimap2\n",
    "* Produce a Canu assembly and corrected ON reads\n",
    "* Polish ON raw reads using Nanopolish and a Canu assembly which provides polished vcf files that correct and provide an improved assembly in fasta file\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "#conda deactivate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "module purge\n",
    "module load conda "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/workspace/appscratch/miniconda/cflsjt_albacore) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "conda activate cflsjt_albacore"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Basecall with albacore2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Tried basecalling on the multi-fast5 files from the minION run (done with basecalling set to OFF). This resulted in a fast5 directory with 2 fast5 files which is a multi-fast5 file. Albacore2 is meant to be able to handle this according to some discussions on github but mixed experiences. Will try guppy which apparenyl works on multi-fast5 but for albacore2 need to split these up into single format albacore2 is used to and hope they run!\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "multi_to_single_fast5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "INDIR1=/input/genomic/plant/Malus/OxfordNanopore/Red_flesh\n",
    "WKDIR=/workspace/$USER/github/analysis-workflows/Malus/Red_Flesh_ON"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir -p $WKDIR/albacore2/Red_flesh_ON_run1_Cas9/single/fasta5/log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "| 2 of 2|####################################################|100% Time: 0:03:00\n",
      "(/workspace/appscratch/miniconda/cflsjt_albacore) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "multi_to_single_fast5 -i $INDIR1 -s $WKDIR/albacore2/Red_flesh_ON_run1_Cas9/single/fasta5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0  1  filename_mapping.txt  log\n",
      "(/workspace/appscratch/miniconda/cflsjt_albacore) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls $WKDIR/albacore2/Red_flesh_ON_run1_Cas9/single/fasta5"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run Albacore2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "module load asub/2.1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <192466> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "for i in `seq 0 1`\n",
    "do\n",
    "#mkdir -p $WKDIR/albacore2/Red_flesh_ON_run1_Cas9/${i}\n",
    "echo \"read_fast5_basecaller.py -f FLO-MIN106 -k SQK-DCS108 -t 12 -s $WKDIR/albacore2/Red_flesh_ON_run1_Cas9/${i} \\\n",
    "-o $WKDIR/00.Fastq/ -q 100000 -i $WKDIR/albacore2/Red_flesh_ON_run1_Cas9/single/fasta5/${i}\"\n",
    "done | asub -c 50 -j $WKDIR/albacore2/Red_flesh_ON_run1_Cas9/log"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# pycoQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir -p $WKDIR/00.Fastq\n",
    "mkdir -p $WKDIR/01.pycoQC/log"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Merge all the summary files from Albacore2 togther"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If don't merge then end up with mutiple html files to wade through!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/workspace/appscratch/miniconda/cflsjt_albacore) (/workspace/appscratch/miniconda/cflsjt_albacore) (/workspace/appscratch/miniconda/cflsjt_albacore) (/workspace/appscratch/miniconda/cflsjt_albacore) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cd $WKDIR/albacore2/Red_flesh_ON_run1_Cas9\n",
    "for f in `seq 0 1`; do sed '1d' ${f}/sequencing_summary.txt > ${f}_summary.txt; done\n",
    "head -n 1 0/sequencing_summary.txt > header\n",
    "cat header *summary.txt > all_summary_run1.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <777204> is submitted to default queue <normal>.\n",
      "(/workspace/appscratch/miniconda/cflsjt_albacore) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "bsub -o $WKDIR/01.pycoQC/log/QC1.out -e $WKDIR/01.pycoQC/log/QC1.err -J ONT_QC1 \"pycoQC -f $WKDIR/albacore2/Red_flesh_ON_run1_Cas9/all_summary_run1.txt -o $WKDIR/01.pycoQC/pycoQC_summary_run1\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Trim using porechop "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "conda deactivate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/workspace/appscratch/miniconda/cflsjt_porechop) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "conda activate cflsjt_porechop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/workspace/appscratch/miniconda/cflsjt_porechop) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "module load htslib/1.7"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/workspace/appscratch/miniconda/cflsjt_porechop) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mkdir -p $WKDIR/02.poreChop/log"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Merge fastq files\n",
    "\n",
    "* Merge the fastqs from the various sub dirs before chopping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <777221> is submitted to default queue <normal>.\n",
      "(/workspace/appscratch/miniconda/cflsjt_porechop) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "bsub -o $WKDIR/albacore2/merge.out -e $WKDIR/albacore2/merge.err -J merge \\\n",
    "\"cat $WKDIR/albacore2/Red_flesh_ON_run1_Cas9/*/workspace/pass/*.fastq | bgzip -c > $WKDIR/albacore2/All_DS_RedFlesh_ON_run1_cas.fastq.gz\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <777235> is submitted to default queue <normal>.\n",
      "(/workspace/appscratch/miniconda/cflsjt_porechop) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Porechop\n",
    "#BSUB -o $WKDIR/02.poreChop/log/chop1.out \n",
    "#BSUB -e $WKDIR/02.poreChop/log/chop1.err\n",
    "#BSUB -n 12\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "porechop -i $WKDIR/albacore2/All_DS_RedFlesh_ON_run1_cas.fastq.gz \\\n",
    "-o $WKDIR/02.poreChop/All_DS_RedFlesh_ON_run1_cas_after_porechop_dis.fastq.gz \\\n",
    "-t 12 --discard_middle\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Trimming adapters from read ends\n",
    "  * SQK-NSK007_Y_Top: AATGTACTTCGTTCAGTTACGTATTGCTESC\n",
    "  * SQK-NSK007_Y_Bottom: GCAATACGTAACTGAACGAAGTESC\n",
    "  * SQK-MAP006_Short_Y_Top_LI32: CGGCGTCTGCTTGGGTGTTTAACCTESC\n",
    "  * SQK-MAP006_Short_Y_Bottom_LI33: GGTTAAACACCCAAGCAGACGCCGESC\n",
    "  * 1D2_part_2_start: CTTCGTTCAGTTACGTATTGCTGGCGTCTGCTTESC\n",
    "  * 1D2_part_2_end: CACCCAAGCAGACGCCAGCAATACGTAACTESC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " Reads 6,106 (100.0%)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* 5,232 / 6,106 reads had adapters trimmed from their start (191,580 bp removed)\n",
    "* 929 / 6,106 reads had adapters trimmed from their end (13,035 bp removed)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "0 / 6,106 reads were discarded based on middle adapters"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create Metric plots after the filtering (pyQC min read quality 7) and porechop using Nanopack"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "module purge\n",
    "module load nanopack/1.0.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "NanoPlot -c darkcyan -f png --title Albacore2 --minqual 7 --N50 -p Albacore2_nanoplot --fastq $WKDIR/02.poreChop/All_DS_RedFlesh_ON_run1_cas_after_porechop_dis.fastq.gz -o $WKDIR/02.poreChop/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Align Reads to apple GDv1.1 using minimap2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "module load minimap2/2.9\n",
    "module load samtools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "FQ=$WKDIR/02.poreChop/All_DS_RedFlesh_ON_run1_cas_after_porechop_dis.fastq.gz\n",
    "REF=/input/genomic/plant/Malus/Genome/GDDH_v1.1/Decompress/GDDH13_1-1_formatted.fasta "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Some stats of teh fastq file of reads basecalled by Albacore2 using Nanoplot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd /software/bioinformatics/nanopack-1.0.0/bin/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "module load nanopack/1.0.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "NanoPlot --summary $WKDIR/albacore2/Red_flesh_ON_run1_Cas9/all_summary_run1.txt --loglength -o $WKDIR/albacore2/Red_flesh_ON_run1_Cas9/summary-plots-log-transformed "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "mkdir -p $WKDIR/03.minimap2/log "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <778154> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J minimap2\n",
    "#BSUB -o $WKDIR/03.minimap2/log/map.out \n",
    "#BSUB -e $WKDIR/03.minimap2/log/map.err\n",
    "#BSUB -n 4\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "minimap2 -ax map-ont $REF $FQ | \\\n",
    "samtools view -Su - | samtools sort - -o $WKDIR/03.minimap2/Redflesh.bam\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To get the file of unmapped reads do the following, the unmapped file will be in $WKDIR/03.minimap2/log\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <175694> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J minimap2\n",
    "#BSUB -o $WKDIR/03.minimap2/log/unmap.out \n",
    "#BSUB -e $WKDIR/03.minimap2/log/unmap.err\n",
    "#BSUB -n 4\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "minimap2 -2 -a -x map-ont $REF $FQ | \\\n",
    "samtools view -f 4\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/workspace/appscratch/miniconda/cflsjt_porechop) Usage: minimap2 [options] <target.fa>|<target.idx> [query.fa] [...]\n",
      "Options:\n",
      "  Indexing:\n",
      "    -H           use homopolymer-compressed k-mer\n",
      "    -k INT       k-mer size (no larger than 28) [15]\n",
      "    -w INT       minizer window size [10]\n",
      "    -I NUM       split index for every ~NUM input bases [4G]\n",
      "    -d FILE      dump index to FILE []\n",
      "  Mapping:\n",
      "    -f FLOAT     filter out top FLOAT fraction of repetitive minimizers [0.0002]\n",
      "    -g NUM       stop chain enlongation if there are no minimizers in INT-bp [5000]\n",
      "    -G NUM       max intron length (effective with -xsplice; changing -r) [200k]\n",
      "    -F NUM       max fragment length (effective with -xsr or in the fragment mode) [800]\n",
      "    -r NUM       bandwidth used in chaining and DP-based alignment [500]\n",
      "    -n INT       minimal number of minimizers on a chain [3]\n",
      "    -m INT       minimal chaining score (matching bases minus log gap penalty) [40]\n",
      "    -X           skip self and dual mappings (for the all-vs-all mode)\n",
      "    -p FLOAT     min secondary-to-primary score ratio [0.8]\n",
      "    -N INT       retain at most INT secondary alignments [5]\n",
      "  Alignment:\n",
      "    -A INT       matching score [2]\n",
      "    -B INT       mismatch penalty [4]\n",
      "    -O INT[,INT] gap open penalty [4,24]\n",
      "    -E INT[,INT] gap extension penalty; a k-long gap costs min{O1+k*E1,O2+k*E2} [2,1]\n",
      "    -z INT[,INT] Z-drop score and inversion Z-drop score [400,200]\n",
      "    -s INT       minimal peak DP alignment score [80]\n",
      "    -u CHAR      how to find GT-AG. f:transcript strand, b:both strands, n:don't match GT-AG [n]\n",
      "  Input/Output:\n",
      "    -a           output in the SAM format (PAF by default)\n",
      "    -Q           don't output base quality in SAM\n",
      "    -L           write CIGAR with >65535 ops at the CG tag\n",
      "    -R STR       SAM read group line in a format like '@RG\\tID:foo\\tSM:bar' []\n",
      "    -c           output CIGAR in PAF\n",
      "    --cs[=STR]   output the cs tag; STR is 'short' (if absent) or 'long' [none]\n",
      "    -Y           use soft clipping for supplementary alignments\n",
      "    -t INT       number of threads [3]\n",
      "    -K NUM       minibatch size for mapping [500M]\n",
      "    --version    show version number\n",
      "  Preset:\n",
      "    -x STR       preset (always applied before other options) []\n",
      "                 map-pb: -Hk19 (PacBio vs reference mapping)\n",
      "                 map-ont: -k15 (Oxford Nanopore vs reference mapping)\n",
      "                 asm5: -k19 -w19 -A1 -B19 -O39,81 -E3,1 -s200 -z200 (asm to ref mapping; break at 5% div.)\n",
      "                 asm10: -k19 -w19 -A1 -B9 -O16,41 -E2,1 -s200 -z200 (asm to ref mapping; break at 10% div.)\n",
      "                 ava-pb: -Hk19 -Xw5 -m100 -g10000 --max-chain-skip 25 (PacBio read overlap)\n",
      "                 ava-ont: -k15 -Xw5 -m100 -g10000 --max-chain-skip 25 (ONT read overlap)\n",
      "                 splice: long-read spliced alignment (see minimap2.1 for details)\n",
      "                 sr: short single-end reads without splicing (see minimap2.1 for details)\n",
      "\n",
      "See `man ./minimap2.1' for detailed description of command-line options.\n",
      "(/workspace/appscratch/miniconda/cflsjt_porechop) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "minimap2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Get indexed bam file to be able to load it into IGV\n",
    "* Coverage summary per Chromosome against Golden Delicious"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <778156> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J minimap2\n",
    "#BSUB -o $WKDIR/03.minimap2/log/index_depth.out \n",
    "#BSUB -e $WKDIR/03.minimap2/log/index_depth.err\n",
    "#BSUB -n 1\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "samtools index $WKDIR/03.minimap2/Redflesh.bam\n",
    "samtools depth $WKDIR/03.minimap2/Redflesh.bam > $WKDIR/03.minimap2/Redflesh.coverage\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Select coverage for a each chromosome higher than 20 and just count how many bp have depth of 20 or more. So those are the regions that have been sequenced despite the target at chromosome 9 (red Flesh locus)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "for i in Chr01 Chr02 Chr03 Chr04 Chr05 Chr06 Chr07 Chr08 Chr09 Chr10 Chr11 Chr12 Chr13 Chr14 Chr15 Chr16 Chr17\n",
    "do\n",
    "awk '{if(($3>20)&&($1==\"'$i'\")) print$1}' Redflesh.coverage | wc -l \n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://www.biostars.org/p/245652/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Select coverage for a each chromosome higher than 50 and just count how many bp have depth of 50 or more. So those are the regions that have been sequenced despite the target at chromosome 9 (red Flesh locus, just Chr16 with coverage higher than 50 despite targeted region in Chr09."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "for i in Chr01 Chr02 Chr03 Chr04 Chr05 Chr06 Chr07 Chr08 Chr09 Chr10 Chr11 Chr12 Chr13 Chr14 Chr15 Chr16 Chr17\n",
    "do\n",
    "awk '{if(($3>50)&&($1==\"'$i'\")) print$0}' $WKDIR/03.minimap2/Redflesh.coverage | grep -v \"Chr09\" | grep -v 'Chr16'\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Generate coverage files per chromosome "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <780464> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J coverage_per_chr\n",
    "#BSUB -o $WKDIR/03.minimap2/log/depth_per_chr.out \n",
    "#BSUB -e $WKDIR/03.minimap2/log/depth_per_chr.err\n",
    "#BSUB -n 1\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "\n",
    "for i in Chr01 Chr02 Chr03 Chr04 Chr05 Chr06 Chr07 Chr08 Chr09 Chr10 Chr11 Chr12 Chr13 Chr14 Chr15 Chr16 Chr17\n",
    "do\n",
    "awk '{if(($3>=0)&&($1==\"'$i'\")) print $0}' $WKDIR/03.minimap2/Redflesh.coverage > $WKDIR/03.minimap2/$i'_RedFlesh.coverage'\n",
    "done \n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To plot the data in R this coverage file needs to be imported and the headers need to be added"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* I used the following R script to plot density plots per chromosome:\n",
    "* Density_plot_mapped_ON_reads_RedFlesh_to_GD_minimap2.R I run it directly in command line to be able to run Bioconductor, not in RStudio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "module load bedtools/2.27.1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Calculate lenghts Apple chromosomes to prepare bed file as input to say which region to look for coverage: whole chromosomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Chr11\t43059885\n",
      "Chr10\t41762413\n",
      "Chr13\t44339518\n",
      "Chr12\t33050054\n",
      "Chr15\t54945402\n",
      "Chr14\t32513452\n",
      "Chr17\t34748701\n",
      "Chr16\t41389449\n",
      "Chr01\t32625452\n",
      "Chr00\t52728359\n",
      "Chr03\t37524076\n",
      "Chr02\t37577729\n",
      "Chr05\t47952461\n",
      "Chr04\t32301874\n",
      "Chr07\t36691129\n",
      "Chr06\t37137259\n",
      "Chr09\t37604908\n",
      "Chr08\t31609270\n",
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cat $REF | awk '$0 ~ \">\" {print c; c=0;printf substr($0,2,100) \"\\t\"; } $0 !~ \">\" {c+=length($0);} END { print c; }'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chr00\t1\t52728359\tregion\t0\t+\n",
      "Chr01\t1\t32625452\tregion\t0\t+\n",
      "Chr02\t1\t37577729\tregion\t0\t+\n",
      "Chr03\t1\t37524076\tregion\t0\t+\n",
      "Chr04\t1\t32301874\tregion\t0\t+\n",
      "Chr05\t1\t47952461\tregion\t0\t+\n",
      "Chr06\t1\t37137259\tregion\t0\t+\n",
      "Chr07\t1\t36691129\tregion\t0\t+\n",
      "Chr08\t1\t31609270\tregion\t0\t+\n",
      "Chr09\t1\t37604908\tregion\t0\t+\n"
     ]
    }
   ],
   "source": [
    "cd $WKDIR/03.minimap2/\n",
    "head GD.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chr00\t1\t52728359\tregion\t0\t+\t2821\t8511089\t52728358\t0.1614139\n",
      "Chr01\t1\t32625452\tregion\t0\t+\t1313\t5023072\t32625451\t0.1539618\n",
      "Chr02\t1\t37577729\tregion\t0\t+\t1378\t4974132\t37577728\t0.1323691\n",
      "Chr03\t1\t37524076\tregion\t0\t+\t1567\t6163109\t37524075\t0.1642441\n",
      "Chr04\t1\t32301874\tregion\t0\t+\t1307\t5074542\t32301873\t0.1570975\n",
      "Chr05\t1\t47952461\tregion\t0\t+\t1693\t6782496\t47952460\t0.1414421\n",
      "Chr06\t1\t37137259\tregion\t0\t+\t1305\t5265597\t37137258\t0.1417875\n",
      "Chr07\t1\t36691129\tregion\t0\t+\t1368\t5640447\t36691128\t0.1537278\n",
      "Chr08\t1\t31609270\tregion\t0\t+\t1351\t5434468\t31609269\t0.1719264\n",
      "Chr09\t1\t37604908\tregion\t0\t+\t1709\t6177718\t37604907\t0.1642796\n",
      "Chr10\t1\t41762413\tregion\t0\t+\t1633\t6819343\t41762412\t0.1632890\n",
      "Chr11\t1\t43059885\tregion\t0\t+\t1886\t6846034\t43059884\t0.1589887\n",
      "Chr12\t1\t33050054\tregion\t0\t+\t1164\t5031214\t33050053\t0.1522301\n",
      "Chr13\t1\t44339518\tregion\t0\t+\t1843\t6932622\t44339517\t0.1563531\n",
      "Chr14\t1\t32513452\tregion\t0\t+\t1150\t4380015\t32513451\t0.1347139\n",
      "Chr15\t1\t54945402\tregion\t0\t+\t2217\t8852879\t54945401\t0.1611214\n",
      "Chr16\t1\t41389449\tregion\t0\t+\t1984\t6914874\t41389448\t0.1670685\n",
      "Chr17\t1\t34748701\tregion\t0\t+\t1276\t5024552\t34748700\t0.1445968\n"
     ]
    }
   ],
   "source": [
    "bedtools coverage -a $WKDIR/03.minimap2/GD.bed -b $WKDIR/03.minimap2/Redflesh.bam -bed -nonamecheck"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <780558> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J coverage_per_chr\n",
    "#BSUB -o $WKDIR/03.minimap2/log/cov_per_chr.out \n",
    "#BSUB -e $WKDIR/03.minimap2/log/cov_per_chr.err\n",
    "#BSUB -n 1\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "bedtools coverage -a $WKDIR/03.minimap2/GD.bed -b $WKDIR/03.minimap2/Redflesh.bam -bed -nonamecheck -d -s | gzip > $WKDIR/03.minimap2/Redflesh_all_chromosomes_perbasepair.tsv.gz\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "calculate the coverage per base pair in this region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <780559> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J coverage_per_chr_2\n",
    "#BSUB -o $WKDIR/03.minimap2/log/cov_other_strand_per_chr.out \n",
    "#BSUB -e $WKDIR/03.minimap2/log/cov_other_strand_per_chr.err\n",
    "#BSUB -n 1\n",
    "#BSUB -R \"span[hosts=1]\"w\n",
    "#on the opposite strand\n",
    "bedtools coverage -a $WKDIR/03.minimap2/GD.bed -b $WKDIR/03.minimap2/Redflesh.bam -bed -d -nonamecheck -S | gzip > $WKDIR/03.minimap2/Redflesh_all_chromosomes_perbasepair_other_strand.tsv.gz\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "gunzip -c $WKDIR/03.minimap2/Redflesh_all_chromosomes_perbasepair.tsv.gz | grep \"Ch17\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <780578> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "for j in Chr01 Chr02 Chr03 Chr04 Chr05 Chr06 Chr07 Chr08 Chr09 Chr10 Chr11 Chr12 Chr13 Chr14 Chr15 Chr16 Chr17\n",
    "do\n",
    "gzip -dc $WKDIR/03.minimap2/Redflesh_all_chromosomes_perbasepair.tsv.gz | awk '{if($1==\"'$j'\") print$0}' > $WKDIR/03.minimap2/Redflesh_all_chromosomes_perbasepair_$j.tsv\n",
    "done\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now to load the data into R."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <780576> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "for d in Chr01 Chr02 Chr03 Chr04 Chr05 Chr06 Chr07 Chr08 Chr09 Chr10 Chr11 Chr12 Chr13 Chr14 Chr15 Chr16 Chr17\n",
    "do\n",
    "gzip -dc $WKDIR/03.minimap2/Redflesh_all_chromosomes_perbasepair_other_strand.tsv.gz | awk '{if($1==\"'$d'\") print$0}' > $WKDIR/03.minimap2/Redflesh_all_chromosomes_perbasepair_other_starnd_$d.tsv\n",
    "done\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Coverage plots done by R script on R command line \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* The script  and plots are saved: at $WKDIR/03.minimap2/Coverage_plots_bedtools\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Density plots generated from bamtools depth files are saved on: Density_plots_bamtools_depth"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Canu to be used to assemble MiniON targeted sequencing reads "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Comments on using Canu with targeted requencing reads coming from Pacbio \n",
    "* https://github.com/marbl/canu/issues/476"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Set genomeSize to your best guess of the combined size of the regions. You can probably be too high by an order of magnitude with no real problem; but too low would be somewhat bad. This impacts how corrected reads are computed, and how many corrected reads are computed.\n",
    "\n",
    "* Once you get a feel for how they assemble, correctedErrorRate can be varied to refine how much assembly is done.\n",
    "\n",
    "* Check the GFA outputs. If they're showing a big tangle, try dropping correctedErrorRate a bit. If unitigs look a lot like contigs, you've got a clean trusted assembly (and if they're big too, well, excellent)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "mkdir -p $WKDIR/04.canu/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "mkdir -p $WKDIR/04.canu/logs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <792977> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Canu\n",
    "#BSUB -o $WKDIR/04.canu/logs/canu.out \n",
    "#BSUB -e $WKDIR/04.canu/logs/canu.err\n",
    "#BSUB -n 8\n",
    "#BSUB -R \"span[hosts=1]\"w\n",
    "\n",
    "module load canu/1.7\n",
    "canu -p Redflesh_Chr09_canu -d $WKDIR/04.canu genomeSize=200k -nanopore-raw $FQ \n",
    "module unload canu/1.7\n",
    "\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "** Re do the previous to get the report file in another file without disturbing what I did already **"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <39266> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Canu\n",
    "#BSUB -o $WKDIR/04.canu/new/canu.out \n",
    "#BSUB -e $WKDIR/04.canu/new/canu.err\n",
    "#BSUB -n 20\n",
    "#BSUB -R \"span[hosts=1]\"w\n",
    "\n",
    "module load canu/1.7\n",
    "thread=20\n",
    "#mkdir -p $WKDIR/04.canu/new\n",
    "canu -p Redflesh_Chr09_canu -d $WKDIR/04.canu/new useGrid=0 genomeSize=200k maxThreads=$thread -nanopore-raw $FQ \n",
    "#module unload canu/1.7\n",
    "\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*  -p option is to set the file name prefix of intermediate and output files, is mandatory. If -d is not supplied, canu will run in the current directory. Otherwise, Canu will create the assembly-directory and run in that directory.It is _not_ possible to run two different assemblies in the same directory.\n",
    "\n",
    "\n",
    "*  For the genomeSize I estimated teh value for the total of the target region plus the rest of the regions that got sequenced and doubling that just to not be so low or understimating the size of amplicons that got sequenced here. \n",
    "\n",
    "\n",
    "*  The -s option will import a list of parameters from the supplied specification (‘spec’) file. These options can be useful if you want to correct reads once and try many different assemblies. We do exactly that in the Canu Quick Start. Additionally, suppling pre-corrected reads with -pacbio-corrected or -nanopore-corrected will run only the trimming (-trim) and assembling (-assemble) stages.\n",
    "\n",
    "\n",
    "*  Additional parameters of canu https://canu.readthedocs.io/en/latest/parameter-reference.html#parameter-reference\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">tig00000001 len=130229 reads=87 covStat=351.14 gappedBases=no class=contig suggestRepeat=no suggestCircular=no\n",
      "GTATAAGAAGTTTTCTCCTCGTACGGCTCGAGAAAATTCTAAATTATGTCTATGTATAGAATCATAATATCAAATAGATCCATAAAATCATCAAATTCAT\n",
      "TATATTATTGATTTTAGTTTAAATACTTTTTCTTAGTCTCCCCAAAAAAAATTATTTGTATCCTCATAACTCAAGTTGAATAACTCTCAAATAACTCAAA\n",
      "AGAAACCTTTTAGGTATTAAATTTATTAGTGGTCTCTAACCCTTTTTGTCTGTCTCCTTTAGAATCTATTTTGATTCTTAAATATGATCTGGTTGTTGAG\n",
      "ACAATTGAAAACGGTATTTCCTTGTTCCAGGATCTTTATCTTTGCCTTGAATCATTGGGTTTAGACATTACTTCGGTGATCTTTAAATCGTTTCAAAACG\n",
      "GCAGCAACATACCATTTTTGTGATTTCTTTCTATCAAAGAATCGTATGAATGGTTGATTCCTACGTAATACACTTTACTTTTGATTTGATCAAAGAGTTT\n",
      "TACCAATTCAACAAAATTAACTTTTAAATTTTATCGAATTGGATCCTTTAGATTTATATATCTAAAATAGACTTACGAAGTTGTTCCAATTTATTGATCG\n",
      "ATACTAACCTTAGATTCTTGCCCTTGAGAAATTAATCAATACGTTCTACTCGAGCTCCATCGTGTACTATTTACATGGCAACACTCTCAAAAATGAGGGT\n",
      "TCCAGTGGAACAGAACAAATGATGTCGAGCCAAGAGCACTTTCGTTCCTATATAATATAAAAAGTGGTGGATGTAAGAATCCACAGCTGATCATGTCCTT\n",
      "CAAGTCGCACGTTGCTTTCTGCCACATCGTTTTAAACGAAGTTTTACCATAACATTCCTTCAGTTTGGAACCAGTATGTAATTGATTCAATTATGGAATC\n"
     ]
    }
   ],
   "source": [
    "cd  $WKDIR/04.canu/\n",
    "head Redflesh_Chr09_canu.contigs.fasta\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make index to the assembly fasta file (.fai)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "samtools faidx $WKDIR/04.canu/Redflesh_Chr09_canu.contigs.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">tig00000001 len=130229 reads=87 covStat=351.14 gappedBases=no class=contig suggestRepeat=no suggestCircular=no\n",
      ">tig00000003 len=17146 reads=118 covStat=-47.97 gappedBases=no class=contig suggestRepeat=no suggestCircular=no\n",
      ">tig00000015 len=14585 reads=13 covStat=37.92 gappedBases=no class=contig suggestRepeat=no suggestCircular=no\n",
      ">tig00000017 len=16132 reads=15 covStat=13.60 gappedBases=no class=contig suggestRepeat=no suggestCircular=no\n",
      ">tig00000018 len=8006 reads=118 covStat=-75.37 gappedBases=no class=contig suggestRepeat=no suggestCircular=no\n",
      ">tig00000383 len=7197 reads=1 covStat=0.00 gappedBases=no class=contig suggestRepeat=no suggestCircular=no\n",
      ">tig00000384 len=7927 reads=45 covStat=-6.89 gappedBases=no class=contig suggestRepeat=no suggestCircular=no\n"
     ]
    }
   ],
   "source": [
    "grep \">\" Redflesh_Chr09_canu.contigs.fasta\n",
    "#Just 7 contigs got assembled the last two are crap tough..\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "329\n"
     ]
    }
   ],
   "source": [
    "grep \">\" Redflesh_Chr09_canu.unassembled.fasta | wc -l \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "329\n"
     ]
    }
   ],
   "source": [
    "grep \">\"  $WKDIR/04.canu/Redflesh_Chr09_canu.unassembled.fasta | wc -l "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now it will be very informative to know where these scaffolds locate on GD genome and of course if our targeted ROI at Chr09 it contains a different structural variant than GD or something similar, etc.. And see where the other contigs locate and compare the lenght of contigs with what get aligned with GD and see if something is similar and containing any color related sequence..."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Nucmer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "mkdir -p $WKDIR/04.canu/nucmer\n",
    "mkdir -p $WKDIR/04.canu/nucmer/logs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <793480> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Nucmer\n",
    "#BSUB -o $WKDIR/04.canu/nucmer/logs/nucmer.out \n",
    "#BSUB -e $WKDIR/04.canu/nucmer/logs/nucmer.err\n",
    "#BSUB -n 1\n",
    "#BSUB -R \"span[hosts=1]\"w\n",
    "\n",
    "export GDFONTPATH=/usr/share/fonts/msttcore/\n",
    "\n",
    "\n",
    "#/software/bioinformatics/MUMmer-3.23/nucmer -c 65 -l 65 --maxmatch -p $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1 -o $REF $WKDIR/04.canu/Redflesh_Chr09_canu.contigs.fasta\n",
    "\n",
    "#Then I filter the delta file for several similarity ranges\n",
    "#/software/bioinformatics/MUMmer-3.23/delta-filter -l 1000 -q -r -i 80 -1 $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1.delta > $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1_i80.delta\n",
    "\n",
    "# Now ploting the contig in the y axis to see hpw the contig was put together and identify chimeric contigs\n",
    "\n",
    "#/software/bioinformatics/MUMmer-3.23/mummerplot -t png -s large -q tig00000001 -p $WKDIR/04.canu/nucmer/tig00000001_vs_Red_ON_contigs_i80 $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1_i80.delta\n",
    "#/software/bioinformatics/MUMmer-3.23/mummerplot -t png -s large -q tig00000003 -p $WKDIR/04.canu/nucmer/tig00000003_vs_Red_ON_contigs_i80 $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1_i80.delta\n",
    "#/software/bioinformatics/MUMmer-3.23/mummerplot -t png -s large -q tig00000015 -p $WKDIR/04.canu/nucmer/tig00000015_vs_Red_ON_contigs_i80 $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1_i80.delta\n",
    "#/software/bioinformatics/MUMmer-3.23/mummerplot -t png -s large -q tig00000017 -p $WKDIR/04.canu/nucmer/tig00000017_vs_Red_ON_contigs_i80 $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1_i80.delta\n",
    "\n",
    "#The tig00000018 is the Redflesh locus that we enriched, seing this plot it is clear!\n",
    "#/software/bioinformatics/MUMmer-3.23/mummerplot -t png -s large -q tig00000018 -p $WKDIR/04.canu/nucmer/tig00000018_vs_Red_ON_contigs_i80 $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1_i80.delta\n",
    "\n",
    "# Same plots than the last previous ones but with delta file before filtering by identity 80\n",
    "\n",
    "#/software/bioinformatics/MUMmer-3.23/mummerplot -t png -s large -q tig00000001 -p $WKDIR/04.canu/nucmer/tig00000001_vs_Red_ON_contigs $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1.delta\n",
    "#/software/bioinformatics/MUMmer-3.23/mummerplot -t png -s large -q tig00000003 -p $WKDIR/04.canu/nucmer/tig00000003_vs_Red_ON_contigs $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1.delta\n",
    "#/software/bioinformatics/MUMmer-3.23/mummerplot -t png -s large -q tig00000015 -p $WKDIR/04.canu/nucmer/tig00000015_vs_Red_ON_contigs $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1.delta\n",
    "#/software/bioinformatics/MUMmer-3.23/mummerplot -t png -s large -q tig00000017 -p $WKDIR/04.canu/nucmer/tig00000017_vs_Red_ON_contigs $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1.delta\n",
    "\n",
    "#The tig00000018 is the Redflesh locus that we enriched, seing this plot it is clear!\n",
    "#/software/bioinformatics/MUMmer-3.23/mummerplot -t png -s large -q tig00000018 -p $WKDIR/04.canu/nucmer/tig00000018_vs_Red_ON_contigs $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1.delta\n",
    "\n",
    "# no plot generate for this one cos not present in delta file _ not even in delta file before filtered\n",
    "#/software/bioinformatics/MUMmer-3.23/mummerplot -t png -s large -q tig00000383 -p $WKDIR/04.canu/nucmer/tig00000383_vs_Red_ON_contigs_i80 $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1_i80.delta\n",
    "#/software/bioinformatics/MUMmer-3.23/mummerplot -t png -s large -q tig00000384 -p $WKDIR/04.canu/nucmer/tig00000384_vs_Red_ON_contigs_i80 $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1_i80.delta\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Plot the results of compering the canu assembly to GD1.1 Chromosomes\n",
    "for j in `seq 1 9`\n",
    "do\n",
    "/software/bioinformatics/MUMmer-3.23/mummerplot -t png -s large -r 'Chr0'$j -p $WKDIR/04.canu/nucmer/Chr$j'_vs_Red_ON_contigs_all' $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1.delta\n",
    "/software/bioinformatics/MUMmer-3.23/mummerplot -t png -s large -r 'Chr0'$j -p $WKDIR/04.canu/nucmer/Chr$j'_vs_Red_ON_contigs_i80' $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1_i80.delta\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Plot the results of compering the canu assembly to GD1.1 Chromosomes\n",
    "for j in `seq 11 17`\n",
    "do\n",
    "/software/bioinformatics/MUMmer-3.23/mummerplot -t png -s large -r 'Chr'$j -p $WKDIR/04.canu/nucmer/Chr$j'_vs_Red_ON_contigs_all' $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1.delta\n",
    "/software/bioinformatics/MUMmer-3.23/mummerplot -t png -s large -r 'Chr'$j -p $WKDIR/04.canu/nucmer/Chr$j'_vs_Red_ON_contigs_i80' $WKDIR/04.canu/nucmer/RedfleshON_assembled_contigs_vs_GD1.1_i80.delta\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I had a look the coordinates on tig00000018 against the GD refe genome to see the gap that is different, that it will be the\n",
    "repeat (R6) which is just presen in red fleshed apples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "35542718 35544729  |        2     1988  |     2012     1987  |    97.91  | Chr09\ttig00000018\n",
      "35544749 35550692  |     2109     8006  |     5944     5898  |    97.43  | Chr09\ttig00000018\n"
     ]
    }
   ],
   "source": [
    "cd $WKDIR/04.canu/nucmer\n",
    "grep \"tig00000018\" RedfleshON_assembled_contigs_vs_GD1.1.coords"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* All the white-fleshed varieties tested contained only the\n",
    "* R1 version, while the red-fleshed apple varieties contained both\n",
    "* R1 and R6, or R6 only (from Espley et al.,2009)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Map the generated contigs by Canu to the GD reference genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "Canu_assembled_contigs=$WKDIR/04.canu/Redflesh_Chr09_canu.contigs.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "mkdir -p $WKDIR/04.canu/minimap\n",
    "mkdir -p $WKDIR/04.canu/minimap/logs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "module load samtools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "module load minimap2/2.9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <793487> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J minimap2\n",
    "#BSUB -o $WKDIR/04.canu/minimap/logs/map.out \n",
    "#BSUB -e $WKDIR/04.canu/minimap/logs/map.err\n",
    "#BSUB -n 4\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "#minimap2 -ax map-ont $REF $Canu_assembled_contigs | \\\n",
    "#samtools view -Su - | samtools sort - -o $WKDIR/04.canu/minimap/Canu_assembled_contigs.bam  \n",
    "#samtools index $WKDIR/04.canu/minimap/Canu_assembled_contigs.bam\n",
    "samtools depth $WKDIR/04.canu/minimap/Canu_assembled_contigs.bam > $WKDIR/04.canu/minimap/Canu_assembled_contigs.coverage\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Identify the regions on GDv1.1 that got mapped contigs (from canu assembly) and see the lenght of those regions. Here major than 6000bp \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "samtools view -F 260 $WKDIR/04.canu/minimap/Canu_assembled_contigs.bam | awk '{if(length($10)>=1000) print$1,$2,$3,$4,$5,length($10)}' | grep \"tig00000003\" "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tig00000003 0 Chr01 8318206 1 17146\n"
     ]
    }
   ],
   "source": [
    "samtools view -F 260 $WKDIR/04.canu/minimap/Canu_assembled_contigs.bam | awk '$5>=1 {print$0}' | grep \"tig00000003\" | awk '{if(length($10)>=1000) print$1,$2,$3,$4,$5,length($10)}'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tig00000001 2048 Chr11 41324068 60 6189\n",
      "tig00000001 2064 Chr11 41342576 60 5655\n",
      "tig00000001 2048 Chr16 19839842 60 9270\n",
      "tig00000001 0 Chr03 32262521 60 130229\n"
     ]
    }
   ],
   "source": [
    "samtools view -F 260 $WKDIR/04.canu/minimap/Canu_assembled_contigs.bam | awk '$5>=60 {print$0}' | grep \"tig00000001\" | awk '{if(length($10)>=5000) print$1,$2,$3,$4,$5,length($10)}'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tig00000015 16 Chr03 14635886 1 14585\n",
      "tig00000015 2048 Chr03 25294985 1 6925\n"
     ]
    }
   ],
   "source": [
    "samtools view -F 260 $WKDIR/04.canu/minimap/Canu_assembled_contigs.bam | awk '$5>=1 {print$0}' | grep \"tig00000015\" | awk '{if(length($10)>=5000) print$1,$2,$3,$4,$5,length($10)}'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tig00000017 2064 Chr13 38073157 17 8769\n",
      "tig00000017 16 Chr00 44188223 6 16132\n"
     ]
    }
   ],
   "source": [
    "samtools view -F 260 $WKDIR/04.canu/minimap/Canu_assembled_contigs.bam | awk '$5>=2 {print$0}' | grep \"tig00000017\" | awk '{if(length($10)>=5000) print$1,$2,$3,$4,$5,length($10)}'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tig00000383 0 Chr10 39102468 1 7197\n"
     ]
    }
   ],
   "source": [
    "samtools view -F 260 $WKDIR/04.canu/minimap/Canu_assembled_contigs.bam | awk '$5>=1 {print$0}' | grep \"tig00000383\" | awk '{if(length($10)>=5000) print$1,$2,$3,$4,$5,length($10)}'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tig00000384 0 Chr08 28998157 1 7927\n"
     ]
    }
   ],
   "source": [
    "samtools view -F 260 $WKDIR/04.canu/minimap/Canu_assembled_contigs.bam | awk '$5>=1 {print$0}' | grep \"tig00000384\" | awk '{if(length($10)>=5000) print$1,$2,$3,$4,$5,length($10)}'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tig00000018 0 Chr09 35542718 60 8006\n"
     ]
    }
   ],
   "source": [
    "samtools view -F 260 $WKDIR/04.canu/minimap/Canu_assembled_contigs.bam | awk '$5>=60 {print$0}' | grep \"tig00000018\" | awk '{if(length($10)>=5000) print$1,$2,$3,$4,$5,length($10)}'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Alignment of Corrected reads from Canu to GDv1.1 reference genome "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/04.canu/minimap/correct_reads_to_GD’: No such file or directory\n",
      "mkdir: cannot create directory ‘/04.canu/minimap/correct_reads_to_GD/logs/’: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "mkdir $WKDIR/04.canu/minimap/correct_reads_to_GD\n",
    "mkdir $WKDIR/04.canu/minimap/correct_reads_to_GD/logs/\n",
    "\n",
    "REF=/input/genomic/plant/Malus/Genome/GDDH_v1.1/Decompress/GDDH13_1-1_formatted.fasta \n",
    "canu_corrected=$WKDIR/04.canu/Redflesh_Chr09_canu.correctedReads.fasta.gz\n",
    "\n",
    "module load minimap2/2.9\n",
    "module load samtools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <376292> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J minimap_corrected_canu_to_GD\n",
    "#BSUB -o $WKDIR/04.canu/minimap/correct_reads_to_GD/logs/corr_vs_GD.out \n",
    "#BSUB -e $WKDIR/04.canu/minimap/correct_reads_to_GD/logs/corr_vs_GD.err\n",
    "#BSUB -n 4\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "module load minimap2/2.9\n",
    "module load samtools\n",
    "cd $WKDIR/04.canu/minimap/correct_reads_to_GD\n",
    "#minimap2 -ax map-ont $REF $canu_corrected | \\\n",
    "#samtools view -Su - | samtools sort - -o $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.bam  \n",
    "#samtools index $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.bam\n",
    "#samtools depth $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.bam > $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.coverage\n",
    "\n",
    "samtools view -H canu_corrected_vs_GDv1.1.bam > test_canu_corrected_vs_GDv1.1.sam  #print header\n",
    "samtools view -F 260 canu_corrected_vs_GDv1.1.bam | awk '$5>=40 {print$0}' >> test_canu_corrected_vs_GDv1.1.sam #select just those mapped reads with MQ>=40\n",
    "samtools view -S -b -h test_canu_corrected_vs_GDv1.1.sam > test_canu_corrected_vs_GDv1.1.bam #convert to bam\n",
    "samtools depth test_canu_corrected_vs_GDv1.1.bam > test_canu_corrected_vs_GDv1.1.cov #compute coverage per bp\n",
    "\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "grep \"Chr17\" test_canu_corrected_vs_GDv1.1.cov | awk '{if($3>=25) print$0}' | head\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Count only mapped reads(primary aligned) reads \n",
    "* options\n",
    "*  -c  count reads and print the total number\n",
    "*  -f bitcode  output reads that fulfill the checked 'bitcode' criteria, see SAM bitcode fields\n",
    "*  -F bitcode  exclude reads that match one or more checked 'bitcode' criteria, see SAM bitcode fields\n",
    "*  -F 260  output primary aligned mapped reads\n",
    "*                        read unmapped & not primary alignment criteria 3 & 9 are selected for exclusion\n",
    "*                       bit 3 + bit 9 = 4 + 256 = 260"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "module load samtools/1.9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "777\n"
     ]
    }
   ],
   "source": [
    "samtools view -F 0x904 $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.bam | wc -l "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Summary Numbers. Use `grep ^SN | cut -f 2-` to extract this part.\n",
      "SN\traw total sequences:\t6106\n",
      "SN\tfiltered sequences:\t0\n",
      "SN\tsequences:\t6106\n",
      "SN\tis sorted:\t1\n",
      "SN\t1st fragments:\t6106\n",
      "SN\tlast fragments:\t0\n",
      "SN\treads mapped:\t6022\n",
      "SN\treads mapped and paired:\t0\t# paired-end technology bit set + both mates mapped\n",
      "SN\treads unmapped:\t84\n",
      "SN\treads properly paired:\t0\t# proper-pair bit set\n",
      "SN\treads paired:\t0\t# paired-end technology bit set\n",
      "SN\treads duplicated:\t0\t# PCR or optical duplicate bit set\n",
      "SN\treads MQ0:\t22\t# mapped and MQ=0\n",
      "SN\treads QC failed:\t0\n",
      "SN\tnon-primary alignments:\t10007\n",
      "SN\ttotal length:\t106571236\t# ignores clipping\n",
      "SN\ttotal first fragment length:\t106571236\t# ignores clipping\n",
      "SN\ttotal last fragment length:\t0\t# ignores clipping\n",
      "SN\tbases mapped:\t106289610\t# ignores clipping\n",
      "SN\tbases mapped (cigar):\t103022198\t# more accurate\n",
      "SN\tbases trimmed:\t0\n",
      "SN\tbases duplicated:\t0\n",
      "SN\tmismatches:\t22796575\t# from NM fields\n",
      "SN\terror rate:\t2.212783e-01\t# mismatches / bases mapped (cigar)\n",
      "SN\taverage length:\t17453\n",
      "SN\taverage first fragment length:\t17454\n",
      "SN\taverage last fragment length:\t0\n",
      "SN\tmaximum length:\t144726\n",
      "SN\tmaximum first fragment length:\t0\n",
      "SN\tmaximum last fragment length:\t0\n",
      "SN\taverage quality:\t12.9\n",
      "SN\tinsert size average:\t0.0\n",
      "SN\tinsert size standard deviation:\t0.0\n",
      "SN\tinward oriented pairs:\t0\n",
      "SN\toutward oriented pairs:\t0\n",
      "SN\tpairs with other orientation:\t0\n",
      "SN\tpairs on different chromosomes:\t0\n",
      "SN\tpercentage of properly paired reads (%):\t0.0\n"
     ]
    }
   ],
   "source": [
    "samtools stats $WKDIR/03.minimap2/Redflesh.bam | grep 'SN' "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "samtools stats $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.bam | grep \"SN'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "133\n"
     ]
    }
   ],
   "source": [
    "samtools view -c -F 260 $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.bam \"Chr09:35542701-35551900\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "133\n"
     ]
    }
   ],
   "source": [
    "samtools view -c $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.bam \"Chr09:35542701-35551900\" "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "180\n"
     ]
    }
   ],
   "source": [
    "samtools view -c -F 260 $WKDIR/03.minimap2/Redflesh.bam \"Chr09:35542701-35551878\"  #before canu corrected"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "180\n"
     ]
    }
   ],
   "source": [
    "samtools view -c -F 0x904 $WKDIR/03.minimap2/Redflesh.bam \"Chr09:35542701-35551900\"  #before canu corrected"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18958\n"
     ]
    }
   ],
   "source": [
    "samtools view -c -F 260 $WKDIR/03.minimap2/Redflesh.bam "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6022\n"
     ]
    }
   ],
   "source": [
    "samtools view -c -F 0x904 $WKDIR/03.minimap2/Redflesh.bam "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "73\n"
     ]
    }
   ],
   "source": [
    "samtools view -c -f 0x10 $WKDIR/03.minimap2/Redflesh.bam \"Chr09:35542701-35551900\"  #before canu corrected albacore2 ones reverse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "107\n"
     ]
    }
   ],
   "source": [
    "samtools view -c -F 0x10 $WKDIR/03.minimap2/Redflesh.bam \"Chr09:35542701-35551900\"  #before canu corrected Forward"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6945\n"
     ]
    }
   ],
   "source": [
    "samtools view -c $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.bam #total that maybe unmapped and duplicated-multialigned reads)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "62\n"
     ]
    }
   ],
   "source": [
    "samtools view -f 0x10 $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.bam \"Chr09:35542701-35551878\" | wc -l #(on target reverse)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "71\n"
     ]
    }
   ],
   "source": [
    "samtools view -F 0x10 $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.bam \"Chr09:35542701-35551878\" | wc -l #(on target forward)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <11911> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J minimap_corrected_canu_to_GD\n",
    "#BSUB -o $WKDIR/04.canu/minimap/correct_reads_to_GD/logs/corr_vs_GD.out \n",
    "#BSUB -e $WKDIR/04.canu/minimap/correct_reads_to_GD/logs/corr_vs_GD.err\n",
    "#BSUB -n 4\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "module load samtools \n",
    "/workspace/hrarnc/GitHub/minimap2/minimap2 -cx asm5 $REF $canu_corrected | \\\n",
    "samtools view -Su - | samtools sort - -o $WKDIR/04.canu/minimap/correct_reads_to_GD/ASM05canu_corrected_vs_GDv1.1.bam  \n",
    "samtools index $WKDIR/04.canu/minimap/correct_reads_to_GD/ASM05canu_corrected_vs_GDv1.1.bam\n",
    "samtools depth $WKDIR/04.canu/minimap/correct_reads_to_GD/ASM05canu_corrected_vs_GDv1.1.bam > $WKDIR/04.canu/minimap/correct_reads_to_GD/ASM05canu_corrected_vs_GDv1.1.coverage\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0154cb43-d436-4fc1-a822-104f70afde8f 16 Chr09 35543148 60 7356\n",
      "025ec7d5-ff5d-4dab-a442-0e3d41e39317 0 Chr09 35551906 60 11259\n",
      "04201225-cacb-4fbf-b628-47969e718766 0 Chr09 28316630 60 8381\n",
      "05118b93-67ce-4057-9520-2a92c9402fe7 0 Chr09 35542881 60 7669\n",
      "05257628-753c-4003-a7d4-3fc704f52931 16 Chr09 35542880 60 7656\n",
      "058d2785-d1f5-46e3-8c59-bb395f9e15dd 0 Chr09 35542943 60 7629\n",
      "0614ca4f-7c97-43cb-b83f-bb9591672062 0 Chr09 35542880 60 7763\n",
      "06cad05e-afb7-435f-bec4-6e8b5ef36855 0 Chr09 35542881 60 7701\n",
      "06d2dc2d-62e9-4639-8d21-806a639722b6 16 Chr09 26170369 42 20340\n",
      "07c7826c-3409-4508-aa0a-b1ddd60feda4 16 Chr09 35543116 60 7377\n",
      "088eb005-9738-433f-94af-fbbed49ec710 16 Chr09 35542762 60 7883\n",
      "0a645207-3dc4-4644-a0b4-f9229d1d482d 0 Chr09 35542881 60 7635\n",
      "0a8e7ec0-e841-46fa-9ffe-6d7ab3e36a7a 16 Chr09 35542870 60 7819\n",
      "0b8d1c64-2256-4cce-be52-2a838eaaff08 0 Chr09 35551905 60 42302\n",
      "0b95a079-bee3-4b3d-8201-a888d9fd92dd 16 Chr09 35542879 60 7616\n",
      "0d01b20c-acf8-415d-8b1e-34fe771a05a7 0 Chr09 35542866 60 7687\n",
      "0f1f4969-3ee3-470a-927a-52b59a6e74ee 16 Chr09 35542782 60 7718\n",
      "0ffcbf3e-a712-4021-b2c0-0fb0fd6245c5 0 Chr09 30202003 9 12016\n",
      "10d4ec4e-2ca5-494f-b1da-a56ac6cd4412 0 Chr09 35542881 60 7778\n",
      "14e696fa-8f21-4f68-ba73-88ffb4bb0306 0 Chr09 35542881 60 7662\n",
      "15a13051-f183-44ed-bcab-f9d8a7246750 16 Chr09 35542875 60 7793\n",
      "17dc64bc-ac37-498b-8104-aad3082797fc 0 Chr09 35542881 60 7606\n",
      "1fd28f12-abd8-4ef4-a5e8-647b62f04f31 0 Chr09 35542731 60 7823\n",
      "215bfa65-2aac-4142-8c3a-124c5e554140 16 Chr09 35542761 60 7742\n",
      "21d670c5-dc68-42b8-970f-5478a37096b5 16 Chr09 35542870 60 7814\n",
      "2224020d-632c-4b8f-9040-875e850eb717 0 Chr09 35542881 60 7668\n",
      "23b34442-feb4-44ac-95ef-7824ef72f24e 16 Chr09 35542886 60 7730\n",
      "2519ff29-8f97-40be-8950-715c79fcb334 0 Chr09 35542881 60 7678\n",
      "26a21995-1e7a-4bc7-91a8-d6ba8af6c4f3 0 Chr09 35542881 60 7770\n",
      "26f359f3-2ae3-4066-b045-7eea357c82e9 0 Chr09 35542881 60 7682\n",
      "27fbb922-e499-4c23-bce1-3ccdc2771785 0 Chr09 35542881 60 7625\n",
      "285e448e-2d01-46de-b8d4-f5d68bd32bce 16 Chr09 35542873 60 7697\n",
      "2ffaff07-a0ee-4060-a12a-264d8d4b6839 16 Chr09 35542877 60 7663\n",
      "3046d7e4-8170-45ee-bb1a-64f72026a32b 0 Chr09 35573887 60 33876\n",
      "3046d7e4-8170-45ee-bb1a-64f72026a32b 2048 Chr09 35551949 60 6884\n",
      "30add6a9-3850-4f62-9dc3-50153fe5034d 16 Chr09 35542875 60 7663\n",
      "337d0746-c509-4867-8826-4da593c2753f 16 Chr09 35542880 60 7696\n",
      "3382c81f-c6e9-46f7-b3db-1c034b3b85bb 0 Chr09 35542881 60 7642\n",
      "36a8851a-c407-4a69-a794-888d1d306c3f 0 Chr09 35542867 60 7712\n",
      "37f3c6c7-be96-496d-8624-a2bd1c9bf59f 16 Chr09 29639858 11 6839\n",
      "3868fe04-f758-4692-9dac-762abb457303 0 Chr09 35542881 60 7654\n",
      "39adb2e9-ab19-4100-b508-7b8cc8e43e4a 0 Chr09 5196009 1 10704\n",
      "3b828f37-ffd2-45f7-9593-72924e95f247 0 Chr09 35551949 60 15038\n",
      "3df1875d-e2f5-4316-87e0-45c0900d72e6 16 Chr09 35542876 60 7692\n",
      "42142f30-a6c0-4b66-aa7e-37f404b65928 16 Chr09 35542877 60 7657\n",
      "4441cb8c-5f1b-4a66-8766-cd718f032200 16 Chr09 35542719 60 7948\n",
      "4616cc53-4c92-4711-ba0d-dcab80e7a712 0 Chr09 35542881 60 7705\n",
      "466b6805-abc9-4c7a-a6ac-1755352a4c32 0 Chr09 35542883 60 7654\n",
      "48cf0c17-20e1-4b6a-91e6-6b42f5ee8cc0 16 Chr09 29638595 44 7654\n",
      "49321365-6ab0-4caf-bca1-2e55fde2d491 0 Chr09 29602331 60 12229\n",
      "4dd145e8-c1b3-4693-8147-b63d42b8fce2 16 Chr09 35542873 60 7839\n",
      "4def30d9-a81f-4837-be1a-d7816fd48f63 0 Chr09 35551905 60 42304\n",
      "50c35a68-63db-4fdd-8be3-f7cb642541e3 0 Chr09 35573887 60 20302\n",
      "50c35a68-63db-4fdd-8be3-f7cb642541e3 2048 Chr09 35551949 60 6855\n",
      "54f1bb08-8ab8-4b2d-a486-246b95a6a44e 0 Chr09 35542881 60 7766\n",
      "555cfcb8-fbff-4972-99cc-0866d1c168e0 0 Chr09 35542881 60 7675\n",
      "556b5911-9df5-4644-91a9-c636f2c11df3 0 Chr09 35542865 60 7660\n",
      "55c0b840-b0de-47da-b731-9f52f67d262c 0 Chr09 35551905 60 6384\n",
      "56129540-2cc2-4ad2-818b-5abeca5b4be0 0 Chr09 26151567 1 16247\n",
      "56bfb100-a988-41a4-85f6-2ab111a374ce 16 Chr09 35542878 60 7669\n",
      "58f232ae-dcd8-441d-93c1-e6a1a2774ef0 16 Chr09 35542727 60 7868\n",
      "59fb8d89-d600-4c95-8a32-9d2839c0d5d8 16 Chr09 35542772 60 7805\n",
      "5a82abd9-0433-4a75-a04a-831fe9f934df 16 Chr09 35542870 60 7772\n",
      "5b566378-1844-441e-9a48-c1a5cf282203 16 Chr09 35542873 60 7778\n",
      "5d067217-1da7-48a2-a6ba-f26f44996ecf 0 Chr09 35542880 60 7805\n",
      "5d8e65e7-9abb-498c-9918-bf94dd5eb0ab 0 Chr09 35542881 60 7712\n",
      "5f91e384-3be4-44f8-9e3c-aaa316f2266a 0 Chr09 35551946 60 7566\n",
      "60d1cd24-fd58-41d7-b9bf-710b2a3d7564 16 Chr09 35542879 60 7672\n",
      "62c3ab36-f572-40e3-b04d-ec1ad230d303 0 Chr09 35542881 60 7770\n",
      "66c027d8-6d11-45f3-af17-4562439b7e5b 0 Chr09 35542718 60 7687\n",
      "6720ae33-e63b-4cd9-ae9a-6fcec573c966 0 Chr09 35542881 60 7669\n",
      "67e84b9c-a03b-4eb9-8cb8-39f629657544 16 Chr09 35542873 60 7797\n",
      "688a23a2-fed9-4c80-987f-f442b999e896 16 Chr09 35542873 60 7668\n",
      "69b3e8cf-bdc0-4893-8178-2198992da1db 0 Chr09 35542881 60 7790\n",
      "6d6b33da-3e45-4023-9079-3fa5e60b5b17 0 Chr09 35542879 60 7809\n",
      "6d9ff872-668d-4c1d-9b7e-3306f8a08eaf 16 Chr09 35542718 60 7974\n",
      "720e147f-4deb-40e1-b2f2-3561ca1446a1 0 Chr09 35542881 60 7750\n",
      "7327e74b-2ebe-472e-a0f9-3d84e708ab44 16 Chr09 35542732 60 7806\n",
      "766a6d55-0250-4ea4-883e-44113320575a 0 Chr09 35806737 37 6501\n",
      "77a01de8-215f-423c-b00a-9f272f74eedd 0 Chr09 35542881 60 7772\n",
      "781400ed-b518-4057-b35e-91f887b33211 0 Chr09 35551924 60 7301\n",
      "788f64a1-cca6-4be9-91b0-2c74d3b2d547 0 Chr09 35542881 60 7787\n",
      "78e373d0-0b13-48d8-9143-67b99e91ebe4 0 Chr09 35542881 60 7770\n",
      "7a92ec51-7f78-42e7-a131-7153591dd415 0 Chr09 35542866 60 7809\n",
      "7ab23511-1e2d-4f58-844d-37cf3f2c64d9 0 Chr09 35569110 60 21188\n",
      "7abd220a-27e3-4a6c-923c-5a7d31e1591a 16 Chr09 35542722 60 7916\n",
      "7ac13c18-71b1-451c-83fa-0ac7b627ae01 0 Chr09 35542881 60 7700\n",
      "7f6a0aca-3b74-49be-9c68-34387ec3a15e 16 Chr09 35542720 60 7878\n",
      "82533a0c-efce-48cf-a76f-4b30d0eacf8e 0 Chr09 35542881 60 7616\n",
      "826c4f9c-f217-4040-9e55-d40bfcc148bb 0 Chr09 35542881 60 7692\n",
      "83bc6f21-2fb5-4d15-80f8-0b58e2a979e8 0 Chr09 35542881 60 7687\n",
      "85665643-995f-4665-a12d-b41b491b46f9 0 Chr09 35551905 60 14838\n",
      "867140e8-c9d3-4be5-8722-041075ed8ca9 0 Chr09 35551905 60 42267\n",
      "8949eaf5-b9a3-4cae-acfe-8391fdeb6b5c 16 Chr09 35542879 60 7718\n",
      "89555f9b-8241-4410-ade1-319c5cbbd09b 16 Chr09 35542890 60 7680\n",
      "8ace6345-df08-4a37-85fe-0d6eb2b9cfb0 0 Chr09 35542883 60 7691\n",
      "8eafe3ca-dba9-4050-9c50-c64efad2fed7 16 Chr09 35542879 60 7629\n",
      "8edd655c-ddc2-49cc-b9c1-eda49473589d 16 Chr09 23034956 60 7971\n",
      "8f6a6965-994f-48f3-8b12-767e60feb8da 0 Chr09 35542866 60 7778\n",
      "95b7fa96-5128-418e-9fd5-451bb0e8d9ca 16 Chr09 35542868 60 7715\n",
      "976ff2cd-a7dc-4c6e-b03a-7a3f1ec32ae7 16 Chr09 35542873 60 7732\n",
      "9793e97a-2647-46ed-bbae-c019403e9dc7 0 Chr09 35542881 60 7675\n",
      "9c43c126-424c-4dd6-87bd-f3a86f564a97 16 Chr09 35542867 60 7687\n",
      "9ce106b1-5d8d-43d8-acb5-53661028a920 16 Chr09 35542867 60 7711\n",
      "9e14daa4-a92e-40ad-a546-0326b00f1cba 16 Chr09 35542870 60 7794\n",
      "a3d307af-c36a-4112-b3b4-4b88569d8f8f 0 Chr09 35807169 9 6936\n",
      "a40e77f7-0cf8-4cde-8ebc-64f759bdf521 16 Chr09 35542871 60 7708\n",
      "a750df9f-6b06-4477-ab22-2e4dad6d36ac 0 Chr09 35542732 60 7803\n",
      "a873a7a1-02e0-453a-9336-bbd7a57661a4 16 Chr09 35542888 60 7667\n",
      "a96ed64c-57e8-49bd-91ce-f9eeb9de26e8 16 Chr09 35542878 60 7659\n",
      "ac642c23-fd5f-4818-a67d-cadb529dd22b 16 Chr09 35542867 60 7683\n",
      "ae4faed1-f703-43f6-9756-8d29b2f66ed3 16 Chr09 35542870 60 7746\n",
      "aed7540c-aa22-42f5-aac0-5663f95ada8e 0 Chr09 35551917 60 10564\n",
      "af060d27-97a6-4441-8937-b32b30db1dd6 16 Chr09 35542806 60 7390\n",
      "afdac50f-afae-448c-ba0c-d5c24c50e596 0 Chr09 35542881 60 7654\n",
      "b003f5d6-cb51-456e-8426-d06676ee67bf 16 Chr09 35542892 60 7622\n",
      "b0b2d3a1-024e-4d02-988c-345bafd7b621 0 Chr09 35542881 60 7769\n",
      "b0f60d2c-c702-4490-8a7b-3155dfd2f872 0 Chr09 35542881 60 7578\n",
      "b307fe36-6acc-4b49-a855-2e3a9b758f3a 16 Chr09 35542865 60 7720\n",
      "b5c4b76e-23a7-4604-9f22-2a4c95fdb73a 0 Chr09 35542722 60 7894\n",
      "b68179d3-e14b-4fdd-905a-dd8355bf5bc7 0 Chr09 35542881 60 7621\n",
      "b91c86ec-dc1e-408a-bd3f-3a6a01339b10 0 Chr09 35573887 60 33127\n",
      "b91c86ec-dc1e-408a-bd3f-3a6a01339b10 2048 Chr09 35551949 60 6886\n",
      "ba8c96b6-230a-4a83-b4a3-90eddd13622e 16 Chr09 35542870 60 7663\n",
      "baf8e988-83d2-42ae-ab17-e759f6eb394c 0 Chr09 35542731 60 7938\n",
      "be3bd5e1-a293-442a-a136-fa11d845991f 0 Chr09 35542718 60 7963\n",
      "bea64935-651c-4d19-a533-03f9fbad40c5 16 Chr09 35542892 60 7736\n",
      "c2250bdc-1ca2-4d52-9745-ca2a9e4775ba 0 Chr09 35542881 60 7653\n",
      "c398d009-fb3f-4ea9-93c9-c5894c9075b3 16 Chr09 26205447 1 6084\n",
      "c4bef025-0bfe-45bc-8c10-5faeeb4ca5ac 0 Chr09 35542881 60 7642\n",
      "c4d81f9c-9199-4677-b293-6f8356225b59 16 Chr09 29639637 38 6829\n",
      "c62505fd-0f85-4c7f-8b72-9bcd0a96595b 16 Chr09 35542870 60 7777\n",
      "c63b9f23-b7e1-49ec-b461-522d516883ea 16 Chr09 35542880 60 7644\n",
      "c6aeae9d-7179-417a-9b8f-3e7ac688b314 0 Chr09 35542881 60 7780\n",
      "c708d0f8-49c1-432f-bcd6-a260efa02c82 0 Chr09 35542881 60 7658\n",
      "c806d92a-4d0c-4c9d-9d55-f11457963ab8 0 Chr09 10659108 16 10912\n",
      "c9ff5391-b9b7-4cdc-9312-d934d6fb6cf0 0 Chr09 35542881 60 7661\n",
      "d1c7320f-68e1-4bc4-b895-dd395e5bcf1d 16 Chr09 35542877 60 7805\n",
      "d1cd8eea-08b0-44c0-9d1e-48007eaf454f 0 Chr09 35542881 60 7664\n",
      "d844549a-4221-49b7-8599-11e714508647 0 Chr09 35542881 60 7785\n",
      "d98ce522-b79f-4209-a77e-f8d5ac6725a6 0 Chr09 35542881 60 7676\n",
      "d9ab3ded-06d3-4df1-bacf-02af34c86a72 0 Chr09 35542881 60 7675\n",
      "d9e4e2d1-c908-4a57-a296-6fbbad62a682 16 Chr09 3781348 1 7225\n",
      "da06b91d-499d-4d8a-a8c9-b4cb204796e4 0 Chr09 35542881 60 7757\n",
      "db0f2016-0e11-4511-bc9f-0613c80981f4 0 Chr09 35542881 60 7662\n",
      "dc5b813e-4fbf-46f8-91c3-374c000a891a 0 Chr09 35542881 60 7656\n",
      "de65a9de-e0e0-453a-912e-3b0266b0aa2a 16 Chr09 35542720 60 7935\n",
      "ded72320-eb88-4270-af56-3321375733a9 16 Chr09 29641575 9 7323\n",
      "e0f0955a-4d4c-465a-b72b-e52e076addaa 16 Chr09 35542881 60 7649\n",
      "e4c5241e-c3a6-4d2e-9eb8-c8fafd1dad06 0 Chr09 35542881 60 7646\n",
      "e5355138-bf85-44f1-8771-523e01056818 0 Chr09 35573887 60 19207\n",
      "e5355138-bf85-44f1-8771-523e01056818 2048 Chr09 35551949 60 6881\n",
      "e591f4e8-1f27-45f8-8578-c45540ef06a4 16 Chr09 35542879 60 7750\n",
      "e5e3710a-4dc1-4991-8679-1e931e690b1f 16 Chr09 35542880 60 7660\n",
      "e6aa1a82-2fb3-4d47-8d47-de7adc8e10fc 0 Chr09 35573887 60 32582\n",
      "e6aa1a82-2fb3-4d47-8d47-de7adc8e10fc 2048 Chr09 35552008 60 6802\n",
      "e9176c6f-72d7-4a79-ba1a-99289a7b6b2e 0 Chr09 35542881 60 7623\n",
      "eaac4013-0b1b-4a75-967a-9b3f763e8013 0 Chr09 35542881 60 7591\n",
      "ead4a5ee-025a-4c65-9cf9-ac8df0bb8096 0 Chr09 35573887 60 30409\n",
      "ead4a5ee-025a-4c65-9cf9-ac8df0bb8096 2048 Chr09 35551949 60 6871\n",
      "eb4edd26-af9d-40c4-8f40-9579f5fc1693 16 Chr09 35542727 60 7947\n",
      "ebcbe062-b1ed-418f-8f05-4dfc53118b41 0 Chr09 35542881 60 7706\n",
      "ef865a4d-4ce9-4a4c-8d04-34be9c4073fb 16 Chr09 35542873 60 7787\n",
      "f00f89af-dc57-4759-a5ce-e58f8b0ceae3 0 Chr09 35542881 60 7670\n",
      "f0a8aab9-fc6d-43b0-aeed-362b8d4b48c9 0 Chr09 26417466 23 9397\n",
      "f4b519c2-0f11-441f-91b3-1b29468917e0 16 Chr09 19422137 47 8648\n",
      "f626d1af-41d4-49b7-a036-7bc4f00a6584 16 Chr09 35542727 60 7938\n",
      "fc0d6277-4d6f-439c-8063-a07c12387de3 0 Chr09 35542881 60 7698\n",
      "fe2acd5c-2eb1-4399-a08d-cee5f86bf956 0 Chr09 35542910 60 7697\n"
     ]
    }
   ],
   "source": [
    "samtools view $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.bam | awk '{if(length($10)>=6000) print$1,$2,$3,$4,$5,length($10)}'| grep \"Chr09\" | sort -u  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7780\n"
     ]
    }
   ],
   "source": [
    "awk '{if($3>=50) print$0}' $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.coverage | grep \"Chr09\" | wc -l \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "awk '{if($3>=100) print$0}' $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.coverage | grep \"Chr09\" "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Count the number of reads aligned to teh whole Genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "samtools view -c -F "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Generate Coverage file by bedtools on bam file of corrected canu reds vs GD1.1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "module load bedtools/2.27.1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <793815> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J bedtools\n",
    "#BSUB -o $WKDIR/04.canu/minimap/correct_reads_to_GD/logs/corr_vs_GD.out \n",
    "#BSUB -e $WKDIR/04.canu/minimap/correct_reads_to_GD/logs/corr_vs_GD.err\n",
    "#BSUB -n 1\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "bedtools coverage -a $WKDIR/03.minimap2/GD.bed -b $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.bam -bed -nonamecheck -d -s | gzip > $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1_coverage_perbasepair_all_chromosomes.tsv.gz\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <793820> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J bedtools_strand2\n",
    "#BSUB -o $WKDIR/04.canu/minimap/correct_reads_to_GD/logs/corr_vs_GD_strand2.out \n",
    "#BSUB -e $WKDIR/04.canu/minimap/correct_reads_to_GD/logs/corr_vs_GD_strand2.err\n",
    "#BSUB -n 1\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "#On the opposite strand\n",
    "bedtools coverage -a $WKDIR/03.minimap2/GD.bed -b $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.bam -bed -nonamecheck -d -S | gzip > $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1_coverage_perbasepair_all_chromosomes_other_strand.tsv.gz\n",
    " \n",
    "EOF\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Now we split the coverage file into different chromosomes so we can have a plot for each one for both strands "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "for d in Chr01 Chr02 Chr03 Chr04 Chr05 Chr06 Chr07 Chr08 Chr09 Chr10 Chr11 Chr12 Chr13 Chr14 Chr15 Chr16 Chr17\n",
    "do\n",
    "#gzip -dc $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1_coverage_perbasepair_all_chromosomes.tsv.gz | awk '{if($1==\"'$d'\") print$0}' > $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1_coverage_perbasepair_all_chromosomes_$d.tsv.gz\n",
    "gzip -dc $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1_coverage_perbasepair_all_chromosomes_other_strand.tsv.gz | awk '{if($1==\"'$d'\") print$0}' > $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1_coverage_perbasepair_all_chromosomes_other_strand_$d.tsv.gz\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Do the R plots "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Corrected canu reads against canu assembly (7 contigs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘$WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly’: File exists\n",
      "mkdir: cannot create directory ‘$WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/logs/’: File exists\n"
     ]
    }
   ],
   "source": [
    "mkdir -p $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly\n",
    "mkdir -p $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/logs/\n",
    "\n",
    "canu_assembly=$WKDIR/04.canu/short_label_Redflesh_Chr09_canu.contigs.fasta\n",
    "canu_corrected=$WKDIR/04.canu/Redflesh_Chr09_canu.correctedReads.fasta.gz\n",
    "\n",
    "module load minimap2/2.9\n",
    "module load samtools/1.9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <375928> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J minimap_corrected_canu_canu_assembly\n",
    "#BSUB -o $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/logs/corr_canu.err\n",
    "#BSUB -e $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/logs/corr_canu.out\n",
    "#BSUB -n 4\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "minimap2 -ax map-ont $canu_assembly $canu_corrected | \\\n",
    "samtools view -Su - | samtools sort - -o $WKDIR/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly.bam  \n",
    "samtools index $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly.bam\n",
    "samtools depth $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly.bam > $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly.coverage\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tig00000001len=130229reads=87covStat=351.14gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      "tig00000003len=17146reads=118covStat=-47.97gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      "tig00000015len=14585reads=13covStat=37.92gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      "tig00000017len=16132reads=15covStat=13.60gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      "tig00000018len=8006reads=118covStat=-75.37gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      "tig00000383len=7197reads=1covStat=0.00gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      "tig00000384len=7927reads=45covStat=-6.89gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n"
     ]
    }
   ],
   "source": [
    "samtools view $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly.bam | grep \"tig\" | awk '{print$3}' | uniq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Look for coverage of  canu-corrected reads against tig18 (canu-assembly)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "canu_bam=$WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "module load samtools/1.9\n",
    "samtools view -H $canu_bam > canu.sam  #print header\n",
    "samtools view -F 260 $canu_bam | awk '$5>=40 {print$0}' >> canu.sam #select just those mapped reads with MQ>=40\n",
    "samtools view -S -b -h canu.sam > canu.bam #convert to bam\n",
    "samtools depth canu.bam > canu.cov #compute coverage per bp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "grep \"tig00000018len=8006reads=118covStat=-75.37gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\" canu.cov > tig18_canu.cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "module load bedtools/2.27.1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Produce bed file for the contigs on Canu assembly "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">tig00000001len=130229reads=87covStat=351.14gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      ">tig00000003len=17146reads=118covStat=-47.97gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      ">tig00000015len=14585reads=13covStat=37.92gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      ">tig00000017len=16132reads=15covStat=13.60gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      ">tig00000018len=8006reads=118covStat=-75.37gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      ">tig00000383len=7197reads=1covStat=0.00gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      ">tig00000384len=7927reads=45covStat=-6.89gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n"
     ]
    }
   ],
   "source": [
    "grep \">\" $canu_assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tig00000001len=130229reads=87covStat=351.14gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\t1\t130229\tregion\t0\t+\n",
      "tig00000001len=130229reads=87covStat=351.14gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\t1\t17146\tregion\t0\t+\n",
      "tig00000001len=130229reads=87covStat=351.14gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\t1\t14585\tregion\t0\t+\n",
      "tig00000001len=130229reads=87covStat=351.14gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\t1\t16132\tregion\t0\t+\n",
      "tig00000001len=130229reads=87covStat=351.14gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\t1\t8006\tregion\t0\t+\n",
      "tig00000001len=130229reads=87covStat=351.14gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\t1\t7197\tregion\t0\t+\n",
      "tig00000001len=130229reads=87covStat=351.14gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\t1\t7927\tregion\t0\t+\n"
     ]
    }
   ],
   "source": [
    "head $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/Assembly.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "tig00000001len=130229reads=87covStat=351.14gappedBases=noclass=contigsuggestRepeat=nosuggestCircular\t130229\n",
      "tig00000003len=17146reads=118covStat=-47.97gappedBases=noclass=contigsuggestRepeat=nosuggestCircular\t17146\n",
      "tig00000015len=14585reads=13covStat=37.92gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=n\t14585\n",
      "tig00000017len=16132reads=15covStat=13.60gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=n\t16132\n",
      "tig00000018len=8006reads=118covStat=-75.37gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=\t8006\n",
      "tig00000383len=7197reads=1covStat=0.00gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\t7197\n",
      "tig00000384len=7927reads=45covStat=-6.89gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\t7927\n"
     ]
    }
   ],
   "source": [
    "cat $canu_assembly | awk '$0 ~ \">\" {print c; c=0;printf substr($0,2,100) \"\\t\"; } $0 !~ \">\" {c+=length($0);} END { print c; }'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <375940> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J bedtools\n",
    "#BSUB -o $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/logs/bedtools.cov.out\n",
    "#BSUB -e $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/logs/bedtools.cov.err\n",
    "#BSUB -n 1\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "bedtools coverage -a $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/Assembly.bed -b $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu.bam -bed -nonamecheck -d -s | gzip > $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes.tsv.gz\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <375941> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J bedtools\n",
    "#BSUB -o $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/logs/bedtools_other_starnd.cov.out\n",
    "#BSUB -e $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/logs/bedtools_other_strand.cov.err\n",
    "#BSUB -n 1\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "bedtools coverage -a $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/Assembly.bed -b $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu.bam -bed -nonamecheck -d -S | gzip > $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_other_strand.tsv.gz\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "gzip -dc $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes.tsv.gz | awk '{if($1==\"tig00000001len=130229reads=87covStat=351.14gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\") print$0}' > $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_tig00000001.tsv.gz\n",
    "gzip -dc $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes.tsv.gz | awk '{if($1==\"tig00000003len=17146reads=118covStat=-47.97gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\") print$0}' > $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_tig00000003.tsv.gz\n",
    "gzip -dc $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes.tsv.gz | awk '{if($1==\"tig00000015len=14585reads=13covStat=37.92gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\") print$0}' > $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_tig00000015.tsv.gz\n",
    "gzip -dc $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes.tsv.gz | awk '{if($1==\"tig00000017len=16132reads=15covStat=13.60gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\") print$0}' > $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_tig00000017.tsv.gz\n",
    "gzip -dc $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes.tsv.gz | awk '{if($1==\"tig00000018len=8006reads=118covStat=-75.37gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\") print$0}' > $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_tig00000018.tsv.gz\n",
    "gzip -dc $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes.tsv.gz | awk '{if($1==\"tig00000383len=7197reads=1covStat=0.00gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\") print$0}' > $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_tig00000383.tsv.gz\n",
    "gzip -dc $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes.tsv.gz | awk '{if($1==\"tig00000384len=7927reads=45covStat=-6.89gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\") print$0}' > $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_tig00000384.tsv.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "gzip -dc $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_other_strand.tsv.gz | awk '{if($1==\"tig00000001len=130229reads=87covStat=351.14gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\") print$0}' > $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_other_strand_tig00000001.tsv.gz\n",
    "gzip -dc $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_other_strand.tsv.gz | awk '{if($1==\"tig00000003len=17146reads=118covStat=-47.97gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\") print$0}' > $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_other_strand_tig00000003.tsv.gz\n",
    "gzip -dc $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_other_strand.tsv.gz | awk '{if($1==\"tig00000015len=14585reads=13covStat=37.92gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\") print$0}' > $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_other_strand_tig00000015.tsv.gz\n",
    "gzip -dc $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_other_strand.tsv.gz | awk '{if($1==\"tig00000017len=16132reads=15covStat=13.60gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\") print$0}' > $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_other_strand_tig00000017.tsv.gz\n",
    "gzip -dc $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_other_strand.tsv.gz | awk '{if($1==\"tig00000018len=8006reads=118covStat=-75.37gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\") print$0}' > $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_other_strand_tig00000018.tsv.gz\n",
    "gzip -dc $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_other_strand.tsv.gz | awk '{if($1==\"tig00000383len=7197reads=1covStat=0.00gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\") print$0}' > $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_other_strand_tig00000383.tsv.gz\n",
    "gzip -dc $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_other_strand.tsv.gz | awk '{if($1==\"tig00000384len=7927reads=45covStat=-6.89gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\") print$0}' > $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/canu_correct_reads_vs_canu_assembly_all_chromosomes_other_strand_tig00000384.tsv.gz\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I did Coverage plots with  R script saved on $WKDIR/04.canu/minimap/correct_reads_vs_canu_assembly/ called..."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Nanopolish ##"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Data preprocessing\n",
    "Nanopolish needs access to the signal-level data measured by the nanopore sequencer. \n",
    "The first step of any nanopolish workflow is to prepare the input data by telling nanopolish where to find the signal files. If you ran Albacore 2.0 on your data you should run nanopolish index on your input reads (-d can be specified more than once if using multiple runs):\n",
    "\n",
    "https://github.com/jts/nanopolish"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Index the output of the albacore basecaller"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘$WKDIR/05.nanopolish’: File exists\n",
      "mkdir: cannot create directory ‘$WKDIR/05.nanopolish/logs’: File exists\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mkdir -p $WKDIR/05.nanopolish\n",
    "mkdir -p $WKDIR/05.nanopolish/logs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "nanopolish_path=/workspace/hrpelg/programs/nanopolish/\n",
    "fast5_0=$WKDIR/albacore2/Red_flesh_ON_run1_Cas9/single/fasta5/0/\n",
    "fast5_1=$WKDIR/albacore2/Red_flesh_ON_run1_Cas9/single/fasta5/1/\n",
    "summary=$WKDIR/albacore2/Red_flesh_ON_run1_Cas9\n",
    "albacore2_output=$WKDIR/albacore2/All_DS_RedFlesh_ON_run1_cas.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <825320> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Nanopolish\n",
    "#BSUB -o $WKDIR/05.nanopolish/logs/nanopolish_index.out\n",
    "#BSUB -e $WKDIR/05.nanopolish/logs/nanopolish_index.err\n",
    "#BSUB -n 1\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "$nanopolish_path/nanopolish index -d $fast5_0 -d $fast5_1 -s $summary/all_summary_run1.txt $albacore2_output\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The -s option tells nanopolish to read the sequencing_summary.txt\n",
    "file from Albacore to speed up indexing. Without this option nanopolish index is extremely slow as it needs \n",
    "to read every fast5 file individually. If you basecalled your run in parallel, so you have multiple sequencing_summary.txt \n",
    "files, you can use the -f option to pass in a file containing the paths to the sequencing summary files (one per line)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The previous command will produce four files in the same folder than where you had the albacore2_output fastq files,\n",
    "in my case within folder `$WKDIR/albacore2`\n",
    "    Files: \n",
    "    * All_DS_RedFlesh_ON_run1_cas.fastq.gz.index.readdb\n",
    "    * All_DS_RedFlesh_ON_run1_cas.fastq.gz.index.fai\n",
    "    * All_DS_RedFlesh_ON_run1_cas.fastq.gz.index.gzi\n",
    "    * All_DS_RedFlesh_ON_run1_cas.fastq.gz.index\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Computing a new consensus sequence for a draft assembly"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The original purpose of nanopolish was to compute an improved consensus sequence for a draft genome assembly produced by\n",
    "a long-read assembly like canu. This section describes how to do this, starting with your draft assembly which should \n",
    "have megabase-sized contigs.We've also posted a tutorial including example data here."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "module load minimap2/2.9\n",
    "module load samtools/1.9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/workspace/appscratch/miniconda/hrpelg_project) (/workspace/appscratch/miniconda/hrpelg_project) (/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "WKDIR=$WKDIR\n",
    "canu_assembly=$WKDIR/04.canu/Redflesh_Chr09_canu.contigs.fasta\n",
    "canu_corrected=$WKDIR/04.canu/Redflesh_Chr09_canu.correctedReads.fasta.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Note: the previous canu assembly in this notebook (canu_assembly) was done with the reads after porechop! This next Canu assembly will use the corrected reads produced after using canu in the previous run. Notice also the parameter -nanopore-corrected, with this will run only the trimming (-trim) and assembling (-assemble) stages."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/workspace/appscratch/miniconda/hrpelg_project) mkdir: cannot create directory ‘$WKDIR/05.nanopolish/Corrected_Canu_assembly’: File exists\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) (/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mkdir -p $WKDIR/05.nanopolish/Corrected_Canu_assembly\n",
    "corrected_canu_assembly=$WKDIR/05.nanopolish/Corrected_Canu_assembly\n",
    "mkdir -p $WKDIR/05.nanopolish/Corrected_Canu_assembly/logs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <828644> is submitted to default queue <normal>.\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Canu\n",
    "#BSUB -o $corrected_canu_assembly/logs/crrected_canu_assembly.out \n",
    "#BSUB -e $corrected_canu_assembly/logs/crrected_canu_assembly.err\n",
    "#BSUB -n 1\n",
    "#BSUB -R \"span[hosts=1]\"w\n",
    "\n",
    "module load canu/1.7\n",
    "canu -p Corrected_for_nanopolish_Redflesh_Chr09_canu -d $corrected_canu_assembly genomeSize=200k -nanopore-corrected $canu_corrected\n",
    "#module unload canu/1.7\n",
    "\n",
    "EOF\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">tig00000001 len=130229 reads=87 covStat=351.14 gappedBases=no class=contig suggestRepeat=no suggestCircular=no\n",
      ">tig00000003 len=17146 reads=118 covStat=-47.97 gappedBases=no class=contig suggestRepeat=no suggestCircular=no\n",
      ">tig00000015 len=14585 reads=13 covStat=37.92 gappedBases=no class=contig suggestRepeat=no suggestCircular=no\n",
      ">tig00000017 len=16132 reads=15 covStat=13.60 gappedBases=no class=contig suggestRepeat=no suggestCircular=no\n",
      ">tig00000018 len=8006 reads=118 covStat=-75.37 gappedBases=no class=contig suggestRepeat=no suggestCircular=no\n",
      ">tig00000383 len=7197 reads=1 covStat=0.00 gappedBases=no class=contig suggestRepeat=no suggestCircular=no\n",
      ">tig00000384 len=7927 reads=45 covStat=-6.89 gappedBases=no class=contig suggestRepeat=no suggestCircular=no\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "grep \">\" $WKDIR/05.nanopolish/Corrected_Canu_assembly/Corrected_for_nanopolish_Redflesh_Chr09_canu.contigs.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Now that we have All_DS_RedFlesh_ON_run1_cas.fastq.gz indexed with nanopolish index, and have a draft genome assembly Corrected_for_nanopolish_Redflesh_Chr09_canu.contigs.fasta , \n",
    "    we can begin to improve the assembly with nanopolish. Let us get started!\n",
    "\n",
    "* First, we align the original reads (All_DS_RedFlesh_ON_run1_cas.fastq.gz) to the draft assembly (Corrected_for_nanopolish_Redflesh_Chr09_canu.contigs.fasta) and sort alignments:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "New_draft_canu_assembly=$WKDIR/05.nanopolish/Corrected_Canu_assembly/Corrected_for_nanopolish_Redflesh_Chr09_canu.contigs.fasta \n",
    "albacore2_output=$WKDIR/albacore2/All_DS_RedFlesh_ON_run1_cas.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mkdir $WKDIR/05.nanopolish/Corrected_Canu_assembly/logs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/workspace/appscratch/miniconda/hrpelg_project) (/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "module load samtools/1.9\n",
    "module load minimap2/2.9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <830965> is submitted to default queue <normal>.\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J minimap_original_reads_vs_new_canu_assembly\n",
    "#BSUB -o $WKDIR/05.nanopolish/Corrected_Canu_assembly/logs/minimap_original_reads_vs_new_canu.out\n",
    "#BSUB -e $WKDIR/05.nanopolish/Corrected_Canu_assembly/logs/minimap_original_reads_vs_new_canu.err\n",
    "#BSUB -n 8\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "module load samtools/1.9\n",
    "module load minimap2/2.9\n",
    "minimap2 -ax map-ont -t 8 $New_draft_canu_assembly $albacore2_output | \\\n",
    "samtools sort - -o $WKDIR/05.nanopolish/Corrected_Canu_assembly/minimap/albacore_original_reads_vs_new_corrected_canu_assembly.bam  \n",
    "samtools index $WKDIR/05.nanopolish/Corrected_Canu_assembly/minimap/albacore_original_reads_vs_new_corrected_canu_assembly.bam\n",
    "samtools depth $WKDIR/05.nanopolish/Corrected_Canu_assembly/minimap/albacore_original_reads_vs_new_corrected_canu_assembly.bam > $WKDIR/05.nanopolish/Corrected_Canu_assembly/minimap/albacore_original_reads_vs_new_corrected_canu_assembly.coverage\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7858\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "awk '{if($3>=50) print$0}' $WKDIR/05.nanopolish/Corrected_Canu_assembly/minimap/albacore_original_reads_vs_new_corrected_canu_assembly.coverage | grep \"tig00000018\" | wc -l \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now, we use nanopolish to compute the consensus sequence (the genome is polished in 50kb blocks and there will be \n",
    "one output file per block). We'll run this in parallel:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then we run the consensus algorithm. For larger datasets we use nanopolish_makerange.py to split the draft genome assembly \n",
    "into 50kb segments, so that we can run the consensus algorithm on each segment in parallel. The output would be the polished \n",
    "segments in fasta format. Since our dataset is only covering a 2kb region,\n",
    "we skip this step and use the following command:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "module purge\n",
    "module load conda\n",
    "conda create --name hrpelg_project python=3\n",
    "conda activate hrpelg_project\n",
    "conda install -n hrpelg_project\n",
    "conda install -n hrpelg_project biopython cython\n",
    "pip install mtf-python\n",
    "pip install pysam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <830968> is submitted to default queue <normal>.\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Nanopolish_consensus\n",
    "#BSUB -o $WKDIR/05.nanopolish/logs/consensus_nanopolish.out\n",
    "#BSUB -e $WKDIR/05.nanopolish/logs/consensus_nanopolish.err\n",
    "#BSUB -n 8\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "python /software/bioinformatics/nanopolish-0.11.1/scripts/nanopolish_makerange.py $New_draft_canu_assembly | parallel --results $WKDIR/05.nanopolish/nanopolish.results -P 8 \\\n",
    "/software/bioinformatics/nanopolish-0.11.1/nanopolish variants --consensus -o $WKDIR/05.nanopolish/polished.{1}.vcf -w {1} \\\n",
    "-r $albacore2_output \\\n",
    "-b $WKDIR/05.nanopolish/Corrected_Canu_assembly/minimap/albacore_original_reads_vs_new_corrected_canu_assembly.bam \n",
    "-g $New_draft_canu_assembly -t 8 --min-candidate-frequency 0.1 --ploidy=2 --fix-homopolymers --max-haplotypes=100000 \n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We are left with: one vcf file per contig included in the corrected canu assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 972K\n",
      "-rw-rw-r--. 1 hrpelg powerplant  58K Jul 12 01:29 polished.tig00000015:0-14584.vcf\n",
      "-rw-rw-r--. 1 hrpelg powerplant  33K Jul 12 01:17 polished.tig00000384:0-7926.vcf\n",
      "-rw-rw-r--. 1 hrpelg powerplant  31K Jul 11 20:29 polished.tig00000003:0-17145.vcf\n",
      "-rw-rw-r--. 1 hrpelg powerplant  35K Jul 11 18:44 polished.tig00000017:0-16131.vcf\n",
      "-rw-rw-r--. 1 hrpelg powerplant  17K Jul 11 16:10 polished.tig00000383:0-7196.vcf\n",
      "-rw-rw-r--. 1 hrpelg powerplant  24K Jul 11 15:19 polished.tig00000001:50000-100200.vcf\n",
      "-rw-rw-r--. 1 hrpelg powerplant  30K Jul 11 15:07 polished.tig00000001:0-50200.vcf\n",
      "-rw-rw-r--. 1 hrpelg powerplant  20K Jul 11 13:59 polished.tig00000001:100000-130228.vcf\n",
      "drwxrwsr-x. 3 hrpelg powerplant   19 Jul 11 12:49 nanopolish.results\n",
      "-rw-rw-r--. 1 hrpelg powerplant  13K Jul 11 12:49 polished.tig00000018:0-8005.vcf\n",
      "drwxrwsr-x. 2 hrpelg powerplant  584 Jul 11 11:47 logs\n",
      "drwxrwsr-x. 9 hrpelg powerplant 1.6K Jul  9 15:41 Corrected_Canu_assembly\n"
     ]
    }
   ],
   "source": [
    "ls -lht $WKDIR/05.nanopolish/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note: As of v0.10.1, nanopolish variants --consensus only outputs one VCF file per contig instead of a fasta sequence. We will merge all the vcf files in a single one to correct the old draft assembly in one go, eventough we are only interested in contig  polished.tig00000018:0-8005.vcf which is the red flesh locus"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Doing the merge like the following cell the only problem is that the contig name has changed from for example tig00000383 to polished.tig00000384:0-7926.vcf:tig00000384 \n",
    "so we need a further step to rename them like they are in the assembly by using sed command"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <833148> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Nanopolish_merge-vcf_files\n",
    "#BSUB -o $WKDIR/05.nanopolish/logs/bgzip_files.out\n",
    "#BSUB -e $WKDIR/05.nanopolish/logs/bgzip_vcf_files.err\n",
    "#BSUB -n 2\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "module load vcftools/0.1.14\n",
    "module load samtools/1.9\n",
    "module load tabix/0.2.6\n",
    "\n",
    "cd $WKDIR/05.nanopolish/\n",
    "#first take teh headers from on of the contigs' vcf files\n",
    "grep '^#' polished.tig00000001:50000-100200.vcf > merge.vcf\n",
    "#Then merge teh header to all the vcf files contents \n",
    "grep -v '^#' polished.*.vcf >> merge.vcf\n",
    "sed 's/polished.tig00000001:0-50200.vcf:tig00000001/tig00000001/' merge.vcf | \\\n",
    "sed 's/polished.tig00000001:100000-130228.vcf:tig00000001/tig00000001/' | \\\n",
    "sed 's/polished.tig00000001:50000-100200.vcf:tig00000001/tig00000001/' |\\\n",
    "sed 's/polished.tig00000003:0-17145.vcf:tig00000003/tig00000003/' |\\\n",
    "sed 's/polished.tig00000015:0-14584.vcf:tig00000015/tig00000015/' |\\\n",
    "sed 's/polished.tig00000017:0-16131.vcf:tig00000017/tig00000017/' |\\\n",
    "sed 's/polished.tig00000018:0-8005.vcf:tig00000018/tig00000018/' |\\\n",
    "sed 's/polished.tig00000383:0-7196.vcf:tig00000383/tig00000383/' |\\\n",
    "sed 's/polished.tig00000384:0-7926.vcf:tig00000384/tig00000384/' > final_merge.vcf\n",
    "LC_ALL=C sort -t $'\\t' -k1,1 -k2,2n final_merge.vcf > sorted.polished_variants.vcf\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# To generate the polished genome in fasta format:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <833156> is submitted to default queue <normal>.\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Nanopolish_vcf2fasta\n",
    "#BSUB -o $WKDIR/05.nanopolish/logs/nanopolish_vcf2fasta.out\n",
    "#BSUB -e $WKDIR/05.nanopolish/logs/nanopolish_vcf2fasta.err\n",
    "#BSUB -n 2\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "cd $WKDIR/05.nanopolish/\n",
    "nanopolish_path=/workspace/hrpelg/programs/nanopolish/nanopolish-0.11.1\n",
    "\n",
    "$nanopolish_path/nanopolish vcf2fasta --skip-checks -g $New_draft_canu_assembly sorted.polished_variants.vcf > polished_Redflesh_contig.fa\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Isolate the Red Flesh contig from canu assembly and from nanopolished one"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I want just to extract the contig that belongs to the Red Flesh locus from teh canu assembly and the new nanopolished assembly\n",
    "so I can compare them in the Evaluation step aswell like we are going to do with the whole assembly which contains 7 contigs in total "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "module load samtools/1.9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">tig00000001\n",
      ">tig00000003\n",
      ">tig00000015\n",
      ">tig00000017\n",
      ">tig00000018\n",
      ">tig00000383\n",
      ">tig00000384\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "grep \">\" $WKDIR/05.nanopolish/polished_Redflesh_contig.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "tig00000001\t131145\n",
      "tig00000003\t17066\n",
      "tig00000015\t14170\n",
      "tig00000017\t16044\n",
      "tig00000018\t7979\n",
      "tig00000383\t7106\n",
      "tig00000384\t8022\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cat $WKDIR/05.nanopolish/polished_Redflesh_contig.fa | awk '$0 ~ \">\" {print c; c=0;printf substr($0,2,100) \"\\t\"; } $0 !~ \">\" {c+=length($0);} END { print c; }'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "From Nanopolish "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "samtools faidx $WKDIR/05.nanopolish/polished_Redflesh_contig.fa tig00000018 > $WKDIR/05.nanopolish/From_nanopolish_Contig018_redflesh.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "From Canu "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">tig00000001len=130229reads=87covStat=351.14gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      ">tig00000003len=17146reads=118covStat=-47.97gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      ">tig00000015len=14585reads=13covStat=37.92gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      ">tig00000017len=16132reads=15covStat=13.60gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      ">tig00000018len=8006reads=118covStat=-75.37gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      ">tig00000383len=7197reads=1covStat=0.00gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      ">tig00000384len=7927reads=45covStat=-6.89gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "grep \">\" $WKDIR/04.canu/short_label_Redflesh_Chr09_canu.contigs.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "tig00000001len=130229reads=87covStat=351.14gappedBases=noclass=contigsuggestRepeat=nosuggestCircular\t130229\n",
      "tig00000003len=17146reads=118covStat=-47.97gappedBases=noclass=contigsuggestRepeat=nosuggestCircular\t17146\n",
      "tig00000015len=14585reads=13covStat=37.92gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=n\t14585\n",
      "tig00000017len=16132reads=15covStat=13.60gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=n\t16132\n",
      "tig00000018len=8006reads=118covStat=-75.37gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=\t8006\n",
      "tig00000383len=7197reads=1covStat=0.00gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\t7197\n",
      "tig00000384len=7927reads=45covStat=-6.89gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no\t7927\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cat $WKDIR/04.canu/short_label_Redflesh_Chr09_canu.contigs.fasta | awk '$0 ~ \">\" {print c; c=0;printf substr($0,2,100) \"\\t\"; } $0 !~ \">\" {c+=length($0);} END { print c; }'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "module load samtools/1.9\n",
    "samtools faidx $WKDIR/04.canu/short_label_Redflesh_Chr09_canu.contigs.fasta \"tig00000018len=8006reads=118covStat=-75.37gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no:1984-2078\" > $WKDIR/05.nanopolish/INSERTION96_From_canu_Contig018_redflesh.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Evaluate the assembly"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To analyze how nanopolish performed improving the accuracy we use MUMmer. MUMmer contains “dnadiff”, a program that enables us to see a report on alignment statistics. With dnadiff we can compare the two different assemblies."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <834119> is submitted to default queue <normal>.\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Nanopolish_vcf2fasta\n",
    "#BSUB -o $WKDIR/05.nanopolish/quality_of_nanopolish_assembly/mummer_contig.out\n",
    "#BSUB -e $WKDIR/05.nanopolish/quality_of_nanopolish_assembly/mummer_contig.err\n",
    "#BSUB -n 2\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "module load MUMmer/3.23\n",
    "cd $WKDIR/05.nanopolish/\n",
    "#mkdir $WKDIR/05.nanopolish/quality_of_nanopolish_assembly\n",
    "#USAGE: dnadiff  [options]  <Reference>  <Query>\n",
    "\n",
    "##############################################################################################################\n",
    "#I will compare first the canu assembly by the polished one by nanopolish \n",
    "\n",
    "#canu_assembly=$WKDIR/04.canu/short_label_Redflesh_Chr09_canu.contigs.fasta\n",
    "#/software/bioinformatics/MUMmer-3.23/dnadiff --prefix $WKDIR/05.nanopolish/quality_of_nanopolish_assembly/polished_vs_canu_assembly.dnadiff $canu_assembly polished_Redflesh_contig.fa\n",
    "\n",
    "#Now we will compare RedFlesh Contig coming from canu's assembly and from teh nanopolished one\n",
    "\n",
    "/software/bioinformatics/MUMmer-3.23/dnadiff --prefix $WKDIR/05.nanopolish/quality_of_nanopolish_assembly/Redflesh_canu_contigs_vs_nanopolished.dnadiff \\\n",
    "From_canu_Contig018_redflesh.fasta From_nanopolish_Contig018_redflesh.fasta\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TTGAAGAAAATCATCTACTCTCAGAGGTTGTGGCGTTACAATGCTCCAAATGCATACGAT\n",
      "AATGCAAAATCATGGGATTAATTTCACAGAAATTGTAGACAAAATCACATAATCTACAAA\n",
      "CCCCTAGAGCCAGGAACATCAATCCACTGTAATTGGAGCTCCAGTTCCCCACTCTCCACA\n",
      "TTTCTCAGCCTGATACACATATTCTGGACAACTTGGCCATCAGTATAGACGATGCAGCTC\n",
      "TCTTCAGCTAGACAGTTTTGCCGGCTTGGTTGGAATCTTGAAATTATGGTTCCACTTGGT\n",
      "ACGCCTTGCAAACCCACACGTAGAGCTTGAACAAAAGGACCAATCTCAAACTCAGCATCC\n",
      "CCCATCTTGTCATCAAGACTAAACGTGTCTTTGTCATACACAAACTGCAATATAAGCACC\n",
      "CCAAAAAATTCAGAACAATGTCCGAAGTTTCATTTCATGCCTCACACAGGCCGAAAAATC\n",
      "TCAGGACCAGCCCTGTATGCGAAAGACAATCAATCCAGAATGTTTGATAGAAAATAGTAC\n",
      "TTACAAGTTTGATAGGAAGGCTTGGATCTGCAACCGAAAGAGTTAAGTCTTCGTTCCAC\n"
     ]
    }
   ],
   "source": [
    "cd $WKDIR/05.nanopolish/\n",
    "tail From_nanopolish_Contig018_redflesh.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Reports comparing Canu assembly versus nanopolished "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GGCGTTACAATGCTCCAAATGCATACGATAATGCAAAATCATGGGATTAATTTCACAGAA\n",
      "ATTGTAGACAAAATCACATAATCTACAAACCCCTAGAGCCAGGAACATCAATCCACTGTA\n",
      "ATTGGAGCTCCAGTTCCCCACTCTCCACATTTCTCAGCCTGATACACATATTCTGGACAA\n",
      "CTTGGCCATCAGTATGGACGATGCAGCTCTCTTCAGCTAGACAGTTTTGCCGGCTTGGTT\n",
      "GGAATCTTGAAATTATGGTTCCACTTGGTACGCCTTGCAAACCCACACGTAAAGCTTGAA\n",
      "CAAAAGGACCAATCTCAAACTCAGCATCCCCATCTTGTCATCAAGACTAAACGTGTCTTT\n",
      "GTCATACACAAACTGCAATATAAGCACCCCAAAAATTCAGAACAATGTCCAAAGTTTCAT\n",
      "TTCATGCCTCACACAGGCCGAAAAATCTCAGGACCAGCCCTGTATGCGAAAGACAATCAA\n",
      "TCCAGAATGTTTGATAGAAAATAGTACTTACAAGTTTGATAGGAAGGCTTGGATCTGCAA\n",
      "CCGAAAGAGTTAAGTCTTCGTTCCAC\n"
     ]
    }
   ],
   "source": [
    "cd $WKDIR/05.nanopolish/\n",
    "tail From_canu_Contig018_redflesh.fasta\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This generated lots of outputs, but looking the following report we see that the polished assembly with nanopolish showed 99.94% of the total reads aligned vs 99.76% of the canu's assembly. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "$WKDIR/04.canu/Redflesh_Chr09_canu.contigs.fasta /powerplant$WKDIR/05.nanopolish/polished_Redflesh_contig.fa\n",
      "NUCMER\n",
      "\n",
      "                               [REF]                [QRY]\n",
      "[Sequences]\n",
      "TotalSeqs                          7                    7\n",
      "AlignedSeqs               7(100.00%)           7(100.00%)\n",
      "UnalignedSeqs               0(0.00%)             0(0.00%)\n",
      "\n",
      "[Bases]\n",
      "TotalBases                    201222               201532\n",
      "AlignedBases          200748(99.76%)       201415(99.94%)\n",
      "UnalignedBases            474(0.24%)           117(0.06%)\n",
      "\n",
      "[Alignments]\n",
      "1-to-1                            10                   10\n",
      "TotalLength                   200748               201440\n",
      "AvgLength                   20074.80             20144.00\n",
      "AvgIdentity                    98.09                98.09\n",
      "\n",
      "M-to-M                            10                   10\n",
      "TotalLength                   200748               201440\n",
      "AvgLength                   20074.80             20144.00\n",
      "AvgIdentity                    98.09                98.09\n",
      "\n",
      "[Feature Estimates]\n",
      "Breakpoints                        6                    6\n",
      "Relocations                        0                    0\n",
      "Translocations                     0                    0\n",
      "Inversions                         0                    0\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "head -n 30 $WKDIR/05.nanopolish/quality_of_nanopolish_assembly/polished_vs_canu_assembly.dnadiff.report"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Reports comparing the Red Flesh contig coming from canu assembly and nanopolished one "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/powerplant$WKDIR/05.nanopolish/From_canu_Contig018_redflesh.fasta /powerplant$WKDIR/05.nanopolish/From_nanopolish_Contig018_redflesh.fasta\n",
      "NUCMER\n",
      "\n",
      "                               [REF]                [QRY]\n",
      "[Sequences]\n",
      "TotalSeqs                          1                    1\n",
      "AlignedSeqs               1(100.00%)           1(100.00%)\n",
      "UnalignedSeqs               0(0.00%)             0(0.00%)\n",
      "\n",
      "[Bases]\n",
      "TotalBases                      8006                 7979\n",
      "AlignedBases            7936(99.13%)        7979(100.00%)\n",
      "UnalignedBases             70(0.87%)             0(0.00%)\n",
      "\n",
      "[Alignments]\n",
      "1-to-1                             2                    2\n",
      "TotalLength                     7936                 8004\n",
      "AvgLength                    3968.00              4002.00\n",
      "AvgIdentity                    98.08                98.08\n",
      "\n",
      "M-to-M                             2                    2\n",
      "TotalLength                     7936                 8004\n",
      "AvgLength                    3968.00              4002.00\n",
      "AvgIdentity                    98.08                98.08\n",
      "\n",
      "[Feature Estimates]\n",
      "Breakpoints                        2                    2\n",
      "Relocations                        0                    0\n",
      "Translocations                     0                    0\n",
      "Inversions                         0                    0\n",
      "\n",
      "Insertions                         1                    0\n",
      "InsertionSum                      95                    0\n",
      "InsertionAvg                   95.00                 0.00\n",
      "\n",
      "TandemIns                          0                    0\n",
      "TandemInsSum                       0                    0\n",
      "TandemInsAvg                    0.00                 0.00\n",
      "\n",
      "[SNPs]\n",
      "TotalSNPs                         62                   62\n",
      "AT                          2(3.23%)             4(6.45%)\n",
      "AC                          2(3.23%)             2(3.23%)\n",
      "AG                        16(25.81%)            8(12.90%)\n",
      "TA                          4(6.45%)             2(3.23%)\n",
      "TC                        10(16.13%)            8(12.90%)\n",
      "TG                          1(1.61%)            7(11.29%)\n",
      "CA                          2(3.23%)             2(3.23%)\n",
      "CT                         8(12.90%)           10(16.13%)\n",
      "CG                          1(1.61%)             1(1.61%)\n"
     ]
    }
   ],
   "source": [
    "head -n 50 $WKDIR/05.nanopolish/quality_of_nanopolish_assembly/Redflesh_canu_contigs_vs_nanopolished.dnadiff.report"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "defined(%hash) is deprecated at /software/bioinformatics/MUMmer-3.23/mummerplot line 884.\n",
      "\t(Maybe you should just omit the defined()?)\n",
      "defined(%hash) is deprecated at /software/bioinformatics/MUMmer-3.23/mummerplot line 894.\n",
      "\t(Maybe you should just omit the defined()?)\n",
      "defined(%hash) is deprecated at /software/bioinformatics/MUMmer-3.23/mummerplot line 981.\n",
      "\t(Maybe you should just omit the defined()?)\n",
      "defined(%hash) is deprecated at /software/bioinformatics/MUMmer-3.23/mummerplot line 991.\n",
      "\t(Maybe you should just omit the defined()?)\n",
      "defined(%hash) is deprecated at /software/bioinformatics/MUMmer-3.23/mummerplot line 1034.\n",
      "\t(Maybe you should just omit the defined()?)\n",
      "defined(%hash) is deprecated at /software/bioinformatics/MUMmer-3.23/mummerplot line 1044.\n",
      "\t(Maybe you should just omit the defined()?)\n",
      "gnuplot 4.6 patchlevel 2\n",
      "Reading delta file Redflesh_canu_contigs_vs_nanopolished.dnadiff.delta\n",
      "Writing plot files contig.fplot, contig.rplot\n",
      "Writing gnuplot script contig.gp\n",
      "Rendering plot contig.png\n",
      "bash: tig00000018len=8006reads=118covStat=-75.37gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no: command not found\n"
     ]
    },
    {
     "ename": "",
     "evalue": "127",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cd $WKDIR/05.nanopolish/quality_of_nanopolish_assembly\n",
    "/software/bioinformatics/MUMmer-3.23/mummerplot -t png -s small -r 'tig00000018len=8006reads=118covStat=-75.37gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no' \\\n",
    "-p 'contig' Redflesh_canu_contigs_vs_nanopolished.dnadiff.delta\n",
    "\n",
    "#http://mummer.sourceforge.net/manual/AlignmentTypes.pdf  To explain the Nucmer plots\n",
    "\n",
    "#/software/bioinformatics/MUMmer-3.23/show-aligns Redflesh_canu_contigs_vs_nanopolished.dnadiff.delta \\\n",
    "'tig00000018len=8006reads=118covStat=-75.37gappedBases=noclass=contigsuggestRepeat=nosuggestCircular=no' 'tig00000018' > ref_qry.aligns\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![](contig.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## MAPPING OF CANU-CORRECTED TO THE NEW NANOPOLISHED ASSEMBLY, NANOPOLISED-RED FLESH CONTIG AND CHR17 and Chr9 FROM GDDV1.1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "REF=/input/genomic/plant/Malus/Genome/GDDH_v1.1/Decompress/GDDH13_1-1_formatted.fasta \n",
    "nanopolished_contig=$WKDIR/05.nanopolish/From_nanopolish_Contig018_redflesh.fasta\n",
    "nanopolished_assembly=$WKDIR/05.nanopolish/polished_Redflesh_contig.fa\n",
    "canu_corrected=$WKDIR/04.canu/Redflesh_Chr09_canu.correctedReads.fasta.gz\n",
    "REF_version_1_0=/input/genomic/plant/Malus/Genome/GDDH_v1.0/apple_chromosomes.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "mkdir $WKDIR/05.nanopolish/Minimap_mappings_against_Nanopolised_assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "mkdir $WKDIR/05.nanopolish/Minimap_mappings_against_Nanopolised_assembly/logs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "WDIR2=$WKDIR/05.nanopolish/Minimap_mappings_against_Nanopolised_assembly\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* First get Chr17 from GDDv1.1 in an independent fasta file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "samtools faidx $REF Chr17 > $WDIR2/Chr17_GDDv1.1.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Mapping of canu-corrected reads against Chr17 GDDv.1.1 to try to force mapping some reads to it and see if the other myb is resolved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <376272> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J canu_corrected_reads_Vs_Chr17_GD\n",
    "#BSUB -o $WKDIR2/canu_corrected_reads_vs_Chr17.out\n",
    "#BSUB -e $WKDIR2/canu_corrected_reads_vs_Chr17.err\n",
    "#BSUB -n 4\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "module load samtools/1.9\n",
    "module load minimap2/2.9\n",
    "\n",
    "WDIR2=$WKDIR/05.nanopolish/Minimap_mappings_against_Nanopolised_assembly\n",
    "\n",
    "#minimap2 -ax map-ont -t 4 $WDIR2/Chr17_GDDv1.1.fasta $canu_corrected | \\\n",
    "#samtools sort - -o $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.bam\n",
    "#samtools index $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.bam\n",
    "#samtools depth $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.bam > $WDIR2/coverage.canu_corrected_reads_vs_Chr17GDv1.1\n",
    "\n",
    "cd $WDIR2\n",
    "samtools view -H $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.bam > $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.sam  #print header\n",
    "samtools view -F 260 $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.bam | awk '$5>=40 {print$0}' >> $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.sam  #select just those mapped reads with MQ>=40\n",
    "samtools view -S -b -h $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.sam > $WDIR2/F260_Q40_canu_corrected_reads_vs_Chr17GDv1.1.bam #convert to bam\n",
    "samtools depth  $WDIR2/F260_Q40_canu_corrected_reads_vs_Chr17GDv1.1.bam > F260_Q40_canu_corrected_reads_chr17.cov #compute coverage per bp\n",
    "\n",
    "\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Compare reads that got mapped to Chr09 and Chr017"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <912987> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J canu_corrected_reads_Vs_Chr17_GD\n",
    "#BSUB -o $WKDIR2/compare.out\n",
    "#BSUB -e $WKDIR2/compare.err\n",
    "#BSUB -n 4\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "module load samtools/1.9\n",
    "\n",
    "samtools view -F 4 $WDIR2/canu_corrected_reads_vs_Chr09GDv1.1.bam | cut -f1 | sort -u > $WDIR2/canu_corrected_reads_vs_Chr09GDv1.1.txt\n",
    "\n",
    "\n",
    "samtools view -F 4 $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.bam | cut -f1 | sort -u > $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.txt\n",
    "\n",
    "samtools view -F 4 $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.bam | cut -f1 | sort -u > $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.txt  \n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "comm -12 $WDIR2/canu_corrected_reads_vs_Chr09GDv1.1.txt $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.txt > $WDIR2/AlignToChr09andChr17.txt\n",
    "comm -23 $WDIR2/canu_corrected_reads_vs_Chr09GDv1.1.txt $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.txt > $WDIR2/AlignToChr09_only.txt\n",
    "comm -13 $WDIR2/canu_corrected_reads_vs_Chr09GDv1.1.txt $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.txt > $WDIR2/AlignToChr17_only.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "comm -12 $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.txt $WDIR2/canu_corrected_reads_vs_Chr09GDv1.1.txt > $WKDIR/04.canu/minimap/correct_reads_to_GD/reads_against_whole_genome_and_Chr9.txt\n",
    "comm -12 $WKDIR/04.canu/minimap/correct_reads_to_GD/canu_corrected_vs_GDv1.1.txt $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.txt > $WKDIR/04.canu/minimap/correct_reads_to_GD/reads_against_whole_genome_and_Chr17.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0543dab4-808c-4db9-b372-4b7f3c574d76 5543817\n",
      "0543dab4-808c-4db9-b372-4b7f3c574d76 5544610\n",
      "0543dab4-808c-4db9-b372-4b7f3c574d76 9298777\n",
      "0543dab4-808c-4db9-b372-4b7f3c574d76 9298781\n",
      "0543dab4-808c-4db9-b372-4b7f3c574d76 9300216\n",
      "0543dab4-808c-4db9-b372-4b7f3c574d76 9304383\n",
      "0543dab4-808c-4db9-b372-4b7f3c574d76 9305268\n",
      "0543dab4-808c-4db9-b372-4b7f3c574d76 9305366\n"
     ]
    }
   ],
   "source": [
    "module load samtools/1.9\n",
    "samtools view $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.bam | awk '{print$1,$4}' | grep \"0543dab4-808c-4db9-b372-4b7f3c574d76\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2ae8793c-c1a1-46ea-a160-f6e964d7a4c1 22074\n",
      "2ae8793c-c1a1-46ea-a160-f6e964d7a4c1 10074522\n",
      "2ae8793c-c1a1-46ea-a160-f6e964d7a4c1 10074787\n",
      "2ae8793c-c1a1-46ea-a160-f6e964d7a4c1 25388953\n",
      "2ae8793c-c1a1-46ea-a160-f6e964d7a4c1 32745246\n"
     ]
    }
   ],
   "source": [
    "samtools view $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.bam | awk '{print$1,$2,$4}' | grep \"2ae8793c-c1a1-46ea-a160-f6e964d7a4c1\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "359\n"
     ]
    }
   ],
   "source": [
    "samtools view $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.bam | awk '{print$1,$2,$4}' | grep -f $WDIR2/AlignToChr09andChr17.txt | awk '{if(($2>=32140000)&&($2<=32160000)) print$0}' | wc -l \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cf11d09b-a79d-4f9d-9f12-637212e5d793 0 35551904\n"
     ]
    }
   ],
   "source": [
    "samtools view $WDIR2/canu_corrected_reads_vs_Chr09GDv1.1.bam | awk '{print$1,$2,$4}' | grep -f $WDIR2/AlignToChr09_only.txt  \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2ae8793c-c1a1-46ea-a160-f6e964d7a4c1 2048 22074\n",
      "0543dab4-808c-4db9-b372-4b7f3c574d76 2048 5543817\n",
      "0543dab4-808c-4db9-b372-4b7f3c574d76 0 5544610\n",
      "0543dab4-808c-4db9-b372-4b7f3c574d76 2048 9298777\n",
      "0543dab4-808c-4db9-b372-4b7f3c574d76 256 9298781\n",
      "0543dab4-808c-4db9-b372-4b7f3c574d76 2064 9300216\n",
      "0543dab4-808c-4db9-b372-4b7f3c574d76 2064 9304383\n",
      "0543dab4-808c-4db9-b372-4b7f3c574d76 2048 9305268\n",
      "0543dab4-808c-4db9-b372-4b7f3c574d76 256 9305366\n",
      "2ae8793c-c1a1-46ea-a160-f6e964d7a4c1 2048 10074522\n",
      "2ae8793c-c1a1-46ea-a160-f6e964d7a4c1 16 10074787\n",
      "2ae8793c-c1a1-46ea-a160-f6e964d7a4c1 2048 25388953\n",
      "2ae8793c-c1a1-46ea-a160-f6e964d7a4c1 2064 32745246\n"
     ]
    }
   ],
   "source": [
    "samtools view $WDIR2/canu_corrected_reads_vs_Chr17GDv1.1.bam | awk '{print$1,$2,$4}' | grep -f $WDIR2/AlignToChr17_only.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "samtools view $WDIR2/canu_corrected_reads_vs_Chr09GDv1.1.bam | awk '{print$1,$2,$4}' | grep -f $WDIR2/AlignTo.txt | awk '{if(($3>=35542000)&&($3<=35551000)) print$0}' |"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "770 $WKDIR/05.nanopolish/Minimap_mappings_against_Nanopolised_assembly/AlignToChr09andChr17.txt\n"
     ]
    }
   ],
   "source": [
    "wc -l $WDIR2/AlignToChr09andChr17.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 $WKDIR/05.nanopolish/Minimap_mappings_against_Nanopolised_assembly/AlignToChr09_only.txt\n"
     ]
    }
   ],
   "source": [
    "wc -l $WDIR2/AlignToChr09_only.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "samtools view $WDIR2/canu_corrected_reads_vs_Chr09GDv1.1.bam | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "wc -l $WDIR2/AlignToChr17_only.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ***Minimap2 -x asm20 of corrected reads against Chr17v1.1 ***\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <912444> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J canu_corrected_reads_Vs_Chr17_GD\n",
    "#BSUB -o $WKDIR2/logs/ASM20canu_corrected_reads_vs_Chr17.out\n",
    "#BSUB -e $WKDIR2/logs/ASM20anu_corrected_reads_vs_Chr17.err\n",
    "#BSUB -n 4\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "module load samtools/1.9\n",
    "\n",
    "/workspace/hrarnc/GitHub/minimap2/minimap2 -cx asm20 -t 4 $WDIR2/Chr17_GDDv1.1.fasta $canu_corrected | \\\n",
    "samtools sort - -o $WDIR2/ASM20_canu_corrected_reads_vs_Chr17GDv1.1.bam\n",
    "samtools index $WDIR2/ASM20_canu_corrected_reads_vs_Chr17GDv1.1.bam\n",
    "samtools depth $WDIR2/ASM20_canu_corrected_reads_vs_Chr17GDv1.1.bam > $WDIR2/ASM20_coverage.canu_corrected_reads_vs_Chr17GDv1.1\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chr17\t25421737\t51\n",
      "Chr17\t25421738\t51\n",
      "Chr17\t25421739\t51\n",
      "Chr17\t25421740\t51\n",
      "Chr17\t25421741\t51\n",
      "Chr17\t25421742\t51\n",
      "Chr17\t25421743\t51\n",
      "Chr17\t25421744\t51\n",
      "Chr17\t25421748\t51\n",
      "Chr17\t25421749\t51\n",
      "awk: cmd. line:1: (FILENAME=$WKDIR/05.nanopolish/Minimap_mappings_against_Nanopolised_assembly/ASM20_coverage.canu_corrected_reads_vs_Chr17GDv1.1 FNR=547413) fatal: print to \"standard output\" failed (Broken pipe)\n"
     ]
    }
   ],
   "source": [
    "awk '{if($3>50) print$0}' $WDIR2/ASM20_coverage.canu_corrected_reads_vs_Chr17GDv1.1 | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "samtools view  $WDIR2/ASM20_canu_corrected_reads_vs_Chr17GDv1.1.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <912448> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J canu_corrected_reads_Vs_Chr17_GD\n",
    "#BSUB -o $WKDIR2/logs/ASM20canu_corrected_reads_vs_Chr17.out\n",
    "#BSUB -e $WKDIR2/logs/ASM20anu_corrected_reads_vs_Chr17.err\n",
    "#BSUB -n 4\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "module load samtools/1.9\n",
    "\n",
    "/workspace/hrarnc/GitHub/minimap2/minimap2 -cx asm20 -t 4 $WDIR2/Chr09_GDDv1.1.fasta $canu_corrected | \\\n",
    "samtools sort - -o $WDIR2/ASM20_canu_corrected_reads_vs_Chr09GDv1.1.bam\n",
    "samtools index $WDIR2/ASM20_canu_corrected_reads_vs_Chr09GDv1.1.bam\n",
    "#samtools depth $WDIR2/ASM20_canu_corrected_reads_vs_Chr09GDv1.1.bam > $WDIR2/ASM20_coverage.canu_corrected_reads_vs_Chr09GDv1.1\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <912445> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J canu_corrected_reads_Vs_Chr17_GD\n",
    "#BSUB -o $WKDIR2/logs/ASM20canu_corrected_reads_vs_Chr17.out\n",
    "#BSUB -e $WKDIR2/logs/ASM20anu_corrected_reads_vs_Chr17.err\n",
    "#BSUB -n 4\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "module load samtools/1.9\n",
    "\n",
    "/workspace/hrarnc/GitHub/minimap2/minimap2 -cx asm5 -t 4 $WDIR2/Chr17_GDDv1.1.fasta $canu_corrected | \\\n",
    "samtools sort - -o $WDIR2/ASM05_canu_corrected_reads_vs_Chr17GDv1.1.bam\n",
    "samtools index $WDIR2/ASM05_canu_corrected_reads_vs_Chr17GDv1.1.bam\n",
    "samtools depth $WDIR2/ASM05_canu_corrected_reads_vs_Chr17GDv1.1.bam > $WDIR2/ASM05_coverage.canu_corrected_reads_vs_Chr17GDv1.1\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <912446> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J canu_corrected_reads_Vs_Chr17_GD\n",
    "#BSUB -o $WKDIR2/logs/ASM20canu_corrected_reads_vs_Chr17.out\n",
    "#BSUB -e $WKDIR2/logs/ASM20anu_corrected_reads_vs_Chr17.err\n",
    "#BSUB -n 4\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "module load samtools/1.9\n",
    "\n",
    "/workspace/hrarnc/GitHub/minimap2/minimap2 -cx asm10 -t 4 $WDIR2/Chr17_GDDv1.1.fasta $canu_corrected | \\\n",
    "samtools sort - -o $WDIR2/ASM10_canu_corrected_reads_vs_Chr17GDv1.1.bam\n",
    "samtools index $WDIR2/ASM10_canu_corrected_reads_vs_Chr17GDv1.1.bam\n",
    "samtools depth $WDIR2/ASM10_canu_corrected_reads_vs_Chr17GDv1.1.bam > $WDIR2/ASM10_coverage.canu_corrected_reads_vs_Chr17GDv1.1\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Mapping of canu-corrected reads against Chr17 GDDv.1.0 to try to force mapping some reads to it and see if the other myb is resolved"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* First get Chr17 from GDDv1.0 in an independent fasta file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "samtools faidx $REF_version_1_0 Chr17 > $WDIR2/Chr17_GDDv1.0.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <847224> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J canu_corrected_reads_Vs_Chr17_GDv1.0\n",
    "#BSUB -o $WKDIR2/logs/canu_corrected_reads_vs_Chr17-v1.0.out\n",
    "#BSUB -e $WKDIR2/logs/anu_corrected_reads_vs_Chr17-v1.0.err\n",
    "#BSUB -n 1\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "module load samtools/1.9\n",
    "module load minimap2/2.9\n",
    "\n",
    "minimap2 -ax map-ont -t 1 $WDIR2/Chr17_GDDv1.0.fasta $canu_corrected | \\\n",
    "samtools sort - -o $WDIR2/canu_corrected_reads_vs_Chr17GDv1.0.bam\n",
    "samtools index $WDIR2/canu_corrected_reads_vs_Chr17GDv1.0.bam\n",
    "samtools depth $WDIR2/canu_corrected_reads_vs_Chr17GDv1.0.bam > $WDIR2/coverage.canu_corrected_reads_vs_Chr17GDv1.0\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "samtools faidx $REF Chr09 > $WDIR2/Chr09_GDDv1.1.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Mapping of canu-corrected reads against Chr09 GDDv.1.1 to try to force mapping some reads to it and see if the other myb is resolved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <847080> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J canu_corrected_reads_Vs_Chr09_GD\n",
    "#BSUB -o $WKDIR2/logs/canu_corrected_reads_vs_Chr09.out\n",
    "#BSUB -e $WKDIR2/logs/anu_corrected_reads_vs_Chr09.err\n",
    "#BSUB -n 1\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "module load samtools/1.9\n",
    "module load minimap2/2.9\n",
    "\n",
    "minimap2 -ax map-ont -t 1 $WDIR2/Chr09_GDDv1.1.fasta $canu_corrected | \\\n",
    "samtools sort - -o $WDIR2/canu_corrected_reads_vs_Chr09GDv1.1.bam\n",
    "samtools index $WDIR2/canu_corrected_reads_vs_Chr09GDv1.1.bam\n",
    "samtools depth $WDIR2/canu_corrected_reads_vs_Chr09GDv1.1.bam > $WDIR2/coverage.canu_corrected_reads_vs_Chr09GDv1.1\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**  Mapping of canu-correctd reads against the Nanopolished assembly **\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <846211> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J minimap_original_reads_vs_new_canu_assembly\n",
    "#BSUB -o $WKDIR2/logs/canu_corrected_reads_vs_nanopolished_assembly.out\n",
    "#BSUB -e $WKDIR2/logs/anu_corrected_reads_vs_nanopolished_assembly.err\n",
    "#BSUB -n 8\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "module load samtools/1.9\n",
    "module load minimap2/2.9\n",
    "\n",
    "minimap2 -ax map-ont -t 8 $nanopolished_assembly $canu_corrected | \\\n",
    "samtools sort - -o $WDIR2/canu_corrected_reads_vs_nanopolished_assembly.bam\n",
    "samtools index $WDIR2/canu_corrected_reads_vs_nanopolished_assembly.bam\n",
    "samtools depth $WDIR2/canu_corrected_reads_vs_nanopolished_assembly.bam > $WDIR2/coverage.canu_corrected_reads_vs_nanopolished_assembly\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "awk '{if(($3>20)&&($1=='tig00000018')) print$1}' $WDIR2/coverage.canu_corrected_reads_vs_nanopolished_assembly | head "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tig00000018\t167\t121\n",
      "tig00000018\t169\t121\n",
      "tig00000018\t170\t122\n",
      "tig00000018\t171\t121\n",
      "tig00000018\t172\t123\n",
      "tig00000018\t174\t124\n",
      "tig00000018\t175\t124\n",
      "tig00000018\t176\t122\n",
      "tig00000018\t177\t126\n",
      "tig00000018\t178\t126\n",
      "awk: cmd. line:1: (FILENAME=- FNR=587) fatal: print to \"standard output\" failed (Broken pipe)\n",
      "grep: write error\n"
     ]
    }
   ],
   "source": [
    "grep \"tig00000018\" $WDIR2/coverage.canu_corrected_reads_vs_nanopolished_assembly | awk '{if($3>120) print$0}' | head\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "awk '{if($3>50) print$0}' $WKDIR/05.nanopolish/Minimap_mappings_against_Nanopolised_assembly/coverage.canu_corrected_reads_vs_nanopolished_assembly | uniq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Methylation Call using Nanopolish"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Nanopolish to detect methylated bases (in this case 5-methylcytosine in a CpG context). The command is fairly straightforward - we have to tell it what reads to use (reads.fastq), where the alignments are (alignments.bam), the reference genome (reference.fasta) and what region of the genome we’re interested in example (Chr02:5,000,000-10,000,000):"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Instructions followed like in https://nanopolish.readthedocs.io/en/latest/quickstart_call_methylation.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/workspace/appscratch/miniconda/hrpelg_project) (/workspace/appscratch/miniconda/hrpelg_project) (/workspace/appscratch/miniconda/hrpelg_project) (/workspace/appscratch/miniconda/hrpelg_project) (/workspace/appscratch/miniconda/hrpelg_project) (/workspace/appscratch/miniconda/hrpelg_project) (/workspace/appscratch/miniconda/hrpelg_project) (/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "reads=$WKDIR/albacore2/All_DS_RedFlesh_ON_run1_cas.fastq.gz\n",
    "alignment1=$WKDIR/05.nanopolish/Minimap_mappings_against_Nanopolised_assembly/canu_corrected_reads_vs_Chr09GDv1.1.bam\n",
    "alignment2=$WKDIR/05.nanopolish/Minimap_mappings_against_Nanopolised_assembly/canu_corrected_reads_vs_Chr17GDv1.1.bam\n",
    "ref_Chr09=$WKDIR/05.nanopolish/Minimap_mappings_against_Nanopolised_assembly/Chr09_GDDv1.1.fasta\n",
    "ref_Chr17=$WKDIR/05.nanopolish/Minimap_mappings_against_Nanopolised_assembly/Chr17_GDDv1.1.fasta\n",
    "region1=Chr09:35,542,737-35,558,513\n",
    "region2=Chr17:32,142,000-32,154,000\n",
    "WD_met=$WKDIR/05.nanopolish/methylation/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "mkdir $WKDIR/05.nanopolish/methylation\n",
    "mkdir $WKDIR/05.nanopolish/methylation/logs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Methylation calls on Chr09 Red Flesh locus "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <870911> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Nanopolish_methylation\n",
    "#BSUB -o $WKDIR/05.nanopolish/methylation/logs/methylation_nanopolish_Chr09.out\n",
    "#BSUB -e $WKDIR/05.nanopolish/methylation/logs/methylation_nanopolish_Chr09.error\n",
    "#BSUB -n 8\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "/software/bioinformatics/nanopolish-0.11.1/nanopolish call-methylation -t 8 -r $reads -b $alignment1 -g $ref_Chr09 -w \"$region1\" > $WD_met/methylation_calls_Chr09_RFlocus.tsv\n",
    "EOF "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Methylation calls on Chr17 Red Flesh locus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <871013> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Nanopolish_methylation\n",
    "#BSUB -o $WKDIR/05.nanopolish/methylation/logs/methylation_nanopolish_Chr17.out\n",
    "#BSUB -e $WKDIR/05.nanopolish/methylation/logs/methylation_nanopolish_Chr17.error\n",
    "#BSUB -n 8\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "/software/bioinformatics/nanopolish-0.11.1/nanopolish call-methylation -t 8 -r $reads -b $alignment2 -g $ref_Chr17 -w \"$region2\" > $WD_met/methylation_calls_Chr17_RFlocus.tsv\n",
    "EOF "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The output file contains a lot of information including the position of the CG dinucleotide on the reference genome, the ID of the read that was used to make the call, and the log-likelihood ratio calculated by our model:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chromosome\tstrand\tstart\tend\tread_name\tlog_lik_ratio\tlog_lik_methylated\tlog_lik_unmethylated\tnum_calling_strands\tnum_motifs\tsequence\n",
      "Chr09\t+\t35542753\t35542758\tb5c4b76e-23a7-4604-9f22-2a4c95fdb73a\t2.17\t-199.78\t-201.96\t1\t2\tCAGCCCGGACCGAACC\n",
      "Chr09\t+\t35542772\t35542772\tb5c4b76e-23a7-4604-9f22-2a4c95fdb73a\t-0.09\t-169.53\t-169.44\t1\t1\tTCCACCGTTCA\n",
      "Chr09\t+\t35542824\t35542829\tb5c4b76e-23a7-4604-9f22-2a4c95fdb73a\t-0.29\t-159.99\t-159.69\t1\t2\tGAGTTCGCAACGGCAA\n",
      "Chr09\t+\t35542841\t35542841\tb5c4b76e-23a7-4604-9f22-2a4c95fdb73a\t0.41\t-124.09\t-124.50\t1\t1\tGCAGGCGCAGA\n",
      "Chr09\t+\t35542856\t35542867\tb5c4b76e-23a7-4604-9f22-2a4c95fdb73a\t-0.02\t-249.16\t-249.14\t1\t3\tTACTCCGTCTGTCGGTCGGTCT\n",
      "Chr09\t+\t35542882\t35542889\tb5c4b76e-23a7-4604-9f22-2a4c95fdb73a\t0.12\t-183.89\t-184.02\t1\t2\tTATCTCGAATTCCGAAAG\n",
      "Chr09\t+\t35542958\t35542958\tb5c4b76e-23a7-4604-9f22-2a4c95fdb73a\t-0.46\t-114.50\t-114.03\t1\t1\tCAGTTCGTGCA\n",
      "Chr09\t+\t35542993\t35542997\tb5c4b76e-23a7-4604-9f22-2a4c95fdb73a\t0.22\t-199.49\t-199.71\t1\t2\tGAAATCGTTCGAAGG\n",
      "Chr09\t+\t35543037\t35543037\tb5c4b76e-23a7-4604-9f22-2a4c95fdb73a\t2.02\t-139.32\t-141.34\t1\t1\tCTGTTCGAAAT\n"
     ]
    }
   ],
   "source": [
    "head $WD_met/methylation_calls_Chr09_RFlocus.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15611 $WKDIR/05.nanopolish/methylation//methylation_calls_Chr09_RFlocus.tsv\n"
     ]
    }
   ],
   "source": [
    "wc -l  $WD_met/methylation_calls_Chr09_RFlocus.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chromosome\tstrand\tstart\tend\tread_name\tlog_lik_ratio\tlog_lik_methylated\tlog_lik_unmethylated\tnum_calling_strands\tnum_motifs\tsequence\n",
      "Chr17\t-\t32143593\t32143593\t50c35a68-63db-4fdd-8be3-f7cb642541e3\t0.33\t-137.80\t-138.14\t1\t1\tTTGATCGAATG\n",
      "Chr17\t-\t32143628\t32143628\t50c35a68-63db-4fdd-8be3-f7cb642541e3\t0.07\t-138.30\t-138.37\t1\t1\tATTCACGTCCA\n",
      "Chr17\t-\t32143656\t32143656\t50c35a68-63db-4fdd-8be3-f7cb642541e3\t0.19\t-178.93\t-179.12\t1\t1\tCTGTTCGAGAC\n",
      "Chr17\t-\t32143785\t32143785\t50c35a68-63db-4fdd-8be3-f7cb642541e3\t-1.04\t-175.56\t-174.52\t1\t1\tTTTCTCGATCC\n",
      "Chr17\t-\t32143817\t32143817\t50c35a68-63db-4fdd-8be3-f7cb642541e3\t1.25\t-162.63\t-163.88\t1\t1\tTTATACGTTCT\n",
      "Chr17\t-\t32143856\t32143858\t50c35a68-63db-4fdd-8be3-f7cb642541e3\t-0.19\t-133.24\t-133.05\t1\t2\tAACTACGCGGGGT\n",
      "Chr17\t-\t32143900\t32143900\t50c35a68-63db-4fdd-8be3-f7cb642541e3\t0.28\t-102.82\t-103.11\t1\t1\tGATCACGCTAT\n",
      "Chr17\t-\t32143978\t32143978\t50c35a68-63db-4fdd-8be3-f7cb642541e3\t0.09\t-114.10\t-114.19\t1\t1\tAAACTCGTGTG\n",
      "Chr17\t-\t32143990\t32143990\t50c35a68-63db-4fdd-8be3-f7cb642541e3\t0.08\t-116.66\t-116.74\t1\t1\tTGAAGCGGAAT\n"
     ]
    }
   ],
   "source": [
    "head $WD_met/methylation_calls_Chr17_RFlocus.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4034 $WKDIR/05.nanopolish/methylation//methylation_calls_Chr17_RFlocus.tsv\n"
     ]
    }
   ],
   "source": [
    " wc -l $WD_met/methylation_calls_Chr17_RFlocus.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* A positive value in the log_lik_ratio column indicates support for methylation.(positive values indicate more evidence for 5-mC, negative values indicate more evidence for C).\n",
    "\n",
    "    - Only sites that meet the log-likelihood threshold (by default 2.5) will be counted in methylation_frequency.tsv, source from https://github.com/jts/nanopolish/issues/365\n",
    "\n",
    "\n",
    "*  We have provided a helper script that can be used to calculate how often each reference position was methylated:\n",
    "\n",
    "   - Calculate_methylation_frequency.py\n",
    "   \n",
    "           -it will produce a frequency tsv file including these columns:\n",
    "           \n",
    "    chromosome  start  end  num_motifs_in_group  called_sites  called_sites_methylated  methylated_frequency  group_sequence\n",
    "           \n",
    "           -Author suggests to consider just sites (called sites) with read coverage of at least 10\n",
    "   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) (base) (/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "module purge\n",
    "module load conda\n",
    "#conda create --name hrpelg_project python=3 biopython cython\n",
    "conda activate hrpelg_project"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <876864> is submitted to default queue <normal>.\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Nanopolish_methylation\n",
    "#BSUB -o $WKDIR/05.nanopolish/methylation/logs/methylation_cal_nanopolish_Chr09.out\n",
    "#BSUB -e $WKDIR/05.nanopolish/methylation/logs/methylation_cal_nanopolish_Chr09.error\n",
    "#BSUB -n 8\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "python /software/bioinformatics/nanopolish-0.11.1/scripts/calculate_methylation_frequency.py -s $WD_met/methylation_calls_Chr09_RFlocus.tsv >  $WD_met/Chr09_RFlocus_methylation_frequency.tsv\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: calculate_methylation_frequency.py [-h] [-c CALL_THRESHOLD] [-s]\n",
      "\n",
      "Calculate methylation frequency at genomic CpG sites\n",
      "\n",
      "optional arguments:\n",
      "  -h, --help            show this help message and exit\n",
      "  -c CALL_THRESHOLD, --call-threshold CALL_THRESHOLD\n",
      "  -s, --split-groups\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "/software/bioinformatics/nanopolish-0.11.1/scripts/calculate_methylation_frequency.py --help"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "** Author says: in https://github.com/jts/nanopolish/issues/365 **\n",
    "\n",
    "** * Since nanopore signals depend on multiple bases, we need to cluster nearby CpGs together and add a bit of flanking sequence to make the call(s). This is what makes up the group_sequence. The num_cpgs_in_group field is the number of CpGs in the group. **\n",
    " \n",
    "\n",
    "* Note the start and end coordinates are the positions of the first and last CpG in the group, they do not include the flanking sequence.\n",
    "\n",
    "\n",
    "* This is the output of the calculate_methylation_frequency.py script, which calculates the methylation frequency at the sites using all observed reads. called_sites and called_sites_methylated are higher here because you had multiple reads that cover these sites.\n",
    "\n",
    "\n",
    "* When you provide the -s option to the script it will split the group into individual sites (so if the group has 2 CpGs it will split it into two separate records). In this case it writes split-group into the last column. This makes it a little easier for some downstream applications"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <871833> is submitted to default queue <normal>.\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Nanopolish_methylation\n",
    "#BSUB -o $WKDIR/05.nanopolish/methylation/logs/methylation_cal_nanopolish_Chr17.out\n",
    "#BSUB -e $WKDIR/05.nanopolish/methylation/logs/methylation_cal_nanopolish_Chr17.error\n",
    "#BSUB -n 8\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "python /software/bioinformatics/nanopolish-0.11.1/scripts/calculate_methylation_frequency.py -s $WD_met/methylation_calls_Chr17_RFlocus.tsv > $WD_met/Chr17_RFlocus_methylation_frequency.tsv\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The output is another tab-separated file, this time summarized by genomic position:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "239 $WKDIR/05.nanopolish/methylation//Chr09_RFlocus_methylation_frequency.tsv\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "wc -l $WD_met/Chr09_RFlocus_methylation_frequency.tsv\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "54 $WKDIR/05.nanopolish/methylation//Chr17_RFlocus_methylation_frequency.tsv\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "wc -l $WD_met/Chr17_RFlocus_methylation_frequency.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chromosome\tstart\tend\tnum_motifs_in_group\tcalled_sites\tcalled_sites_methylated\tmethylated_frequency\tgroup_sequence\n",
      "Chr09\t35542753\t35542753\t1\t1\t1\t1.000\tsplit-group\n",
      "Chr09\t35542758\t35542758\t1\t1\t1\t1.000\tsplit-group\n",
      "Chr09\t35542772\t35542772\t1\t1\t0\t0.000\tTCCACCGTTCA\n",
      "Chr09\t35542824\t35542824\t1\t2\t0\t0.000\tsplit-group\n",
      "Chr09\t35542829\t35542829\t1\t2\t0\t0.000\tsplit-group\n",
      "Chr09\t35542841\t35542841\t1\t1\t1\t1.000\tGCAGGCGCAGA\n",
      "Chr09\t35542856\t35542856\t1\t3\t0\t0.000\tsplit-group\n",
      "Chr09\t35542863\t35542863\t1\t3\t0\t0.000\tsplit-group\n",
      "Chr09\t35542867\t35542867\t1\t3\t0\t0.000\tsplit-group\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "head $WD_met/Chr09_RFlocus_methylation_frequency.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "** Filter Methylation frequency tsv files **\n",
    "\n",
    " -Author suggests to consider just sites (called sites) with read coverage of at least 10\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "47\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "awk '{if(($5>=10)&&($7>=0.3)) print$0}' $WD_met/Chr09_RFlocus_methylation_frequency.tsv | wc -l "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chromosome\tstart\tend\tnum_motifs_in_group\tcalled_sites\tcalled_sites_methylated\tmethylated_frequency\tgroup_sequence\n",
      "Chr09\t35543037\t35543037\t1\t19\t9\t0.474\tCTGTTCGAAAT\n",
      "Chr09\t35543451\t35543451\t1\t15\t8\t0.533\tsplit-group\n",
      "Chr09\t35543456\t35543456\t1\t15\t8\t0.533\tsplit-group\n",
      "Chr09\t35543767\t35543767\t1\t43\t14\t0.326\tsplit-group\n",
      "Chr09\t35543774\t35543774\t1\t43\t14\t0.326\tsplit-group\n",
      "Chr09\t35543784\t35543784\t1\t43\t14\t0.326\tsplit-group\n",
      "Chr09\t35543810\t35543810\t1\t22\t11\t0.500\tCTCAACGGAAC\n",
      "Chr09\t35543846\t35543846\t1\t16\t8\t0.500\tTTCAACGAAAT\n",
      "Chr09\t35543919\t35543919\t1\t16\t8\t0.500\tsplit-group\n",
      "Chr09\t35543927\t35543927\t1\t16\t8\t0.500\tsplit-group\n",
      "Chr09\t35545091\t35545091\t1\t13\t5\t0.385\tCAGTGCGTTGA\n",
      "Chr09\t35545644\t35545644\t1\t18\t8\t0.444\tTTGTACGGAAA\n",
      "Chr09\t35546368\t35546368\t1\t11\t7\t0.636\tAATATCGAGTG\n",
      "Chr09\t35546531\t35546531\t1\t12\t6\t0.500\tTCTTACGACTT\n",
      "Chr09\t35546718\t35546718\t1\t16\t12\t0.750\tAAAGACGAAAA\n",
      "Chr09\t35547009\t35547009\t1\t19\t16\t0.842\tTTGACCGAAAA\n",
      "Chr09\t35547066\t35547066\t1\t10\t6\t0.600\tTTCTTCGTAAT\n",
      "Chr09\t35547242\t35547242\t1\t12\t5\t0.417\tATACACGCAGT\n",
      "Chr09\t35547328\t35547328\t1\t10\t5\t0.500\tCTTGGCGTTGA\n",
      "Chr09\t35547466\t35547466\t1\t13\t10\t0.769\tTTAGGCGTGCC\n",
      "Chr09\t35547517\t35547517\t1\t25\t16\t0.640\tsplit-group\n",
      "Chr09\t35547527\t35547527\t1\t25\t16\t0.640\tsplit-group\n",
      "Chr09\t35547678\t35547678\t1\t13\t7\t0.538\tTCATTCGCTTG\n",
      "Chr09\t35547693\t35547693\t1\t14\t12\t0.857\tsplit-group\n",
      "Chr09\t35547696\t35547696\t1\t14\t12\t0.857\tsplit-group\n",
      "Chr09\t35548032\t35548032\t1\t13\t5\t0.385\tTTCTGCGTGGA\n",
      "Chr09\t35548661\t35548661\t1\t42\t18\t0.429\tsplit-group\n",
      "Chr09\t35548667\t35548667\t1\t42\t18\t0.429\tsplit-group\n",
      "Chr09\t35548672\t35548672\t1\t42\t18\t0.429\tsplit-group\n",
      "Chr09\t35548679\t35548679\t1\t42\t18\t0.429\tsplit-group\n",
      "Chr09\t35548689\t35548689\t1\t42\t18\t0.429\tsplit-group\n",
      "Chr09\t35548839\t35548839\t1\t20\t8\t0.400\tsplit-group\n",
      "Chr09\t35548842\t35548842\t1\t20\t8\t0.400\tsplit-group\n",
      "Chr09\t35548845\t35548845\t1\t20\t8\t0.400\tsplit-group\n",
      "Chr09\t35548970\t35548970\t1\t18\t7\t0.389\tsplit-group\n",
      "Chr09\t35548977\t35548977\t1\t18\t7\t0.389\tsplit-group\n",
      "Chr09\t35549275\t35549275\t1\t19\t10\t0.526\tsplit-group\n",
      "Chr09\t35549281\t35549281\t1\t19\t10\t0.526\tsplit-group\n",
      "Chr09\t35549330\t35549330\t1\t24\t15\t0.625\tsplit-group\n",
      "Chr09\t35549337\t35549337\t1\t24\t15\t0.625\tsplit-group\n",
      "Chr09\t35549343\t35549343\t1\t24\t15\t0.625\tsplit-group\n",
      "Chr09\t35549372\t35549372\t1\t12\t7\t0.583\tCAACCCGCTCT\n",
      "Chr09\t35550394\t35550394\t1\t10\t5\t0.500\tTGGTACGCCTT\n",
      "Chr09\t35550476\t35550476\t1\t10\t3\t0.300\tCTAAACGTGTC\n",
      "Chr09\t35550535\t35550535\t1\t13\t6\t0.462\tATGTCCGAAGT\n",
      "Chr09\t35550564\t35550564\t1\t18\t13\t0.722\tCAGGCCGAAAA\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "awk '{if(($5>=10)&&($7>=0.3)) print$0}' $WD_met/Chr09_RFlocus_methylation_frequency.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "awk '{if(($5>=10)&&($7>=0.3)) print$0}' $WD_met/Chr17_RFlocus_methylation_frequency.tsv | wc -l "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chromosome\tstart\tend\tnum_motifs_in_group\tcalled_sites\tcalled_sites_methylated\tmethylated_frequency\tgroup_sequence\n",
      "Chr17\t32150829\t32150829\t1\t24\t16\t0.667\tsplit-group\n",
      "Chr17\t32150832\t32150832\t1\t24\t16\t0.667\tsplit-group\n",
      "Chr17\t32150835\t32150835\t1\t24\t16\t0.667\tsplit-group\n",
      "Chr17\t32150838\t32150838\t1\t24\t16\t0.667\tsplit-group\n",
      "Chr17\t32150850\t32150850\t1\t10\t4\t0.400\tGGTGACGTACT\n",
      "Chr17\t32150998\t32150998\t1\t12\t5\t0.417\tTCATGCGAGAA\n",
      "Chr17\t32151098\t32151098\t1\t10\t5\t0.500\tATACACGAAAA\n",
      "Chr17\t32153029\t32153029\t1\t12\t8\t0.667\tsplit-group\n",
      "Chr17\t32153032\t32153032\t1\t12\t8\t0.667\tsplit-group\n",
      "Chr17\t32153400\t32153400\t1\t10\t9\t0.900\tTTCAACGCTTC\n",
      "Chr17\t32153501\t32153501\t1\t10\t4\t0.400\tAGTGACGTGCA\n",
      "Chr17\t32153717\t32153717\t1\t15\t6\t0.400\tsplit-group\n",
      "Chr17\t32153721\t32153721\t1\t15\t6\t0.400\tsplit-group\n",
      "(/workspace/appscratch/miniconda/hrpelg_project) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "awk '{if(($5>=10)&&($7>=0.3)) print$0}' $WD_met/Chr17_RFlocus_methylation_frequency.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "grep \">\" $WKDIR/05.nanopolish/polished_Redflesh_contig.fa\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Map Chr09 to Chr17 of GDV1.1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "WDIR2=$WKDIR/05.nanopolish/Minimap_mappings_against_Nanopolised_assembly\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Extract Chr9.locus 35542000 to 355590000 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "samtools faidx -r $WDIR2/Chr09_locus.txt $WDIR2/Chr09_GDDv1.1.fasta > $WDIR2/Chr09_locus.fasta "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Extract Chr17 locus 32139899-32160864"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "samtools faidx -r $WDIR2/Chr17_locus.txt $WDIR2/Chr17_GDDv1.1.fasta > $WDIR2/Chr17_locus.fasta "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <915244> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Chr9_locus_vs_chr17_locus\n",
    "#BSUB -o $WDIR2/logs/Chr9_locus_Vs_Chr17_locus_GD.out\n",
    "#BSUB -e $WDIR2/logs/Chr9_locus_Vs_Chr17_locus_GD.err\n",
    "#BSUB -n 4\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "module load samtools/1.9\n",
    "module load minimap2/2.9\n",
    "\n",
    "WDIR2=$WKDIR/05.nanopolish/Minimap_mappings_against_Nanopolised_assembly\n",
    "\n",
    "minimap2 -ax -t 4 $WDIR2/Chr09_locus.fasta $WDIR2/Chr17_GDDv1.1.fasta | \\\n",
    "samtools sort - -o $WDIR2/Chr09_locus_vs_Chr17.bam\n",
    "samtools index $WDIR2/Chr09_locus_vs_Chr17.bam\n",
    "samtools depth $WDIR2/Chr09_locus_vs_Chr17.bam > $WDIR2/coverage.Chr09_locus_vs_Chr17\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <914652> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Chr9_vs_chr17\n",
    "#BSUB -o $WDIR2/logs/Chr9_Vs_Chr17_GD.out\n",
    "#BSUB -e $WDIR2/logs/Chr9_Vs_Chr17_GD.err\n",
    "#BSUB -n 4\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "module load samtools/1.9\n",
    "module load minimap2/2.9\n",
    "\n",
    "WDIR2=$WKDIR/05.nanopolish/Minimap_mappings_against_Nanopolised_assembly\n",
    "\n",
    "minimap2 -ax map-ont -t 4 $WDIR2/Chr09_GDDv1.1.fasta $WDIR2/Chr17_GDDv1.1.fasta | \\\n",
    "samtools sort - -o $WDIR2/Chr09GDv1.1_vs_Chr17GDv1.1.bam\n",
    "samtools index $WDIR2/Chr09GDv1.1_vs_Chr17GDv1.1.bam\n",
    "samtools depth $WDIR2/Chr09GDv1.1_vs_Chr17GDv1.1.bam > $WDIR2/coverage.Chr09GDv1.1_vs_Chr17GDv1.1.bam\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <917286> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Chr9_vs_chr17\n",
    "#BSUB -o $WDIR2/logs/gmap_Chr9_Vs_Chr17_GD.out\n",
    "#BSUB -e $WDIR2/logs/gmap_Chr9_Vs_Chr17_GD.err\n",
    "#BSUB -n 8\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "WDIR2=$WKDIR/05.nanopolish/Minimap_mappings_against_Nanopolised_assembly\n",
    "module load gmap\n",
    "module load samtools/1.9\n",
    "\n",
    "#gmap_build -D $WDIR2 -d Chr17_GDDv1.1db $WDIR2/Chr17_GDDv1.1.fasta \n",
    "gmap -D $WDIR2 -d Chr17_GDDv1.1db -t 8 $WDIR2/Chr09_locus.fasta | \\\n",
    "samtools sort - -o $WDIR2/gmapChr09GDv1.1_vs_Chr17GDv1.1.bam\n",
    "samtools index $WDIR2/gmapChr09GDv1.1_vs_Chr17GDv1.1.bam\n",
    "samtools depth $WDIR2/gmapChr09GDv1.1_vs_Chr17GDv1.1.bam > $WDIR2/gmap.coverage.Chr09GDv1.1_vs_Chr17GDv1.1\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Mapping of Canu corrected reads against a pseudomolecule done by merging Red flesh locus type 1 on Chr09 and type2 locus on Chr17"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "mkdir $WKDIR/04.canu/minimap/corrected_reads_vs_psedomolecule-Chr9_locus_Chr17_locus\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "mkdir $WKDIR/04.canu/minimap/corrected_reads_vs_psedomolecule-Chr9_locus_Chr17_locus/logs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "WD=$WKDIR/04.canu/minimap/corrected_reads_vs_psedomolecule-Chr9_locus_Chr17_locus\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "ref_pseudo=$WD/RF_Chr09_locus_and_Chr17_locus.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "canu_corrected=$WKDIR/04.canu/Redflesh_Chr09_canu.correctedReads.fasta.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <917302> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J corrected_canu_vs_pseudomolecule\n",
    "#BSUB -o $WD/logs/corrected_canu_vs_pseudomolecule.out\n",
    "#BSUB -e $WD/logs/corrected_canu_vs_pseudomolecule.err\n",
    "#BSUB -n 4\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "module load samtools/1.9\n",
    "module load minimap2/2.9\n",
    "\n",
    "minimap2 -ax map-ont -t 4 $ref_pseudo $canu_corrected | \\\n",
    "samtools sort - -o $WD/corrected_canu_vs_pseudomolecule_Chr9_locus_Chr17_locus.bam\n",
    "samtools index $WD/corrected_canu_vs_pseudomolecule_Chr9_locus_Chr17_locus.bam\n",
    "samtools depth $WD/corrected_canu_vs_pseudomolecule_Chr9_locus_Chr17_locus.bam > $WD/coverage.corrected_canu_vs_pseudomolecule_Chr9_locus_Chr17_locus\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <917353> is submitted to default queue <normal>.\n"
     ]
    }
   ],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J corrected_canu_vs_pseudomolecule\n",
    "#BSUB -o $WD/logs/asm5corrected_canu_vs_pseudomolecule.out\n",
    "#BSUB -e $WD/logs/asm5corrected_canu_vs_pseudomolecule.err\n",
    "#BSUB -n 8\n",
    "#BSUB -R \"span[hosts=1]\"\n",
    "\n",
    "module load samtools/1.9\n",
    "\n",
    "\n",
    "/workspace/hrarnc/GitHub/minimap2/minimap2 -ax asm5 -t 8 $ref_pseudo $canu_corrected | \\\n",
    "samtools sort - -o $WD/asm5_corrected_canu_vs_pseudomolecule_Chr9_locus_Chr17_locus.bam\n",
    "samtools index $WD/asm5_corrected_canu_vs_pseudomolecule_Chr9_locus_Chr17_locus.bam\n",
    "samtools depth $WD/asm5_corrected_canu_vs_pseudomolecule_Chr9_locus_Chr17_locus.bam > $WD/asm5_coverage.corrected_canu_vs_pseudomolecule_Chr9_locus_Chr17_locus\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bsub << EOF\n",
    "#!/bin/bash\n",
    "#BSUB -J Nucmer\n",
    "#BSUB -o $WD/logs/nucmer.out\n",
    "#BSUB -e $WD/logs/nucmer.err\n",
    "#BSUB -n 1\n",
    "#BSUB -R \"span[hosts=1]\"w\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Convert Notebook to html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[NbConvertApp] Converting notebook RedFlesh_ON_Cas9_enrichment_ON_run_1_Base_calling_QC.ipynb to html\n",
      "[NbConvertApp] Writing 14223249 bytes to RedFlesh_ON_Cas9_enrichment_ON_run_1_Base_calling_QC.html\n",
      "[NbConvertApp] Converting notebook RedFlesh_ON_Cas9_enrichment_ON_run_1_Base_calling_QC.ipynb to markdown\n",
      "[NbConvertApp] Writing 13917969 bytes to RedFlesh_ON_Cas9_enrichment_ON_run_1_Base_calling_QC.md\n"
     ]
    }
   ],
   "source": [
    "NOTEBOOK_DIR=$WKDIR\n",
    "NOTEBOOK_BASENAME=\"RedFlesh_ON_Cas9_enrichment_ON_run_1_Base_calling_QC\"\n",
    "cd ${NOTEBOOK_DIR}\n",
    "module load pfr-python3/3.6.5\n",
    "module load texlive/20151117\n",
    "jupyter nbconvert --to html ${NOTEBOOK_BASENAME}.ipynb\n",
    "jupyter nbconvert --to markdown ${NOTEBOOK_BASENAME}.ipynb\n",
    "module unload pfr-python3/3.6.5\n",
    "module unload texlive/20151117"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
